<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interface Package for the EpiGraphDB Platform • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Interface Package for the EpiGraphDB Platform">
<meta property="og:description" content="The interface package to access data from the
    EpiGraphDB &lt;https://epigraphdb.org&gt; platform.
    It provides easy access to the EpiGraphDB platform with functions that
    query the corresponding REST endpoints on the API &lt;https://api.epigraphdb.org&gt;
    and return the response data in the tibble data frame format.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/index.html">Articles</a>
    </li>
    <li>
      <a href="articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="epigraphdb-r-package-epigraphdb" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#epigraphdb-r-package-epigraphdb" class="anchor"></a>EpiGraphDB R package <code>epigraphdb</code>
</h1></div>
<p><a href="http://epigraphdb.org"><img src="reference/figures/epigraphdb-logo.svg" alt="" height="60" style="padding:10px"></a> <span class="pull-right"> <a href="https://www.bristol.ac.uk"><img src="reference/figures/uob.svg" alt="" height="60" style="padding:10px"></a> <a href="http://www.bristol.ac.uk/integrative-epidemiology/"><img src="reference/figures/mrc-ieu-logo.png" alt="" height="60" style="padding:10px"></a> </span></p>
<!-- badges: start -->

<!-- badges: end -->
<p><a href="https://epigraphdb.org">EpiGraphDB</a> is an analytical platform and database to support data mining in epidemiology. The platform incorporates a graph of causal estimates generated by systematically applying Mendelian randomization to a wide array of phenotypes, and augments this with a wealth of additional data from other bioinformatic sources. EpiGraphDB aims to support appropriate application and interpretation of causal inference in systematic automated analyses of many phenotypes.</p>
<p><a href="https://github.com/MRCIEU/epigraphdb-r"><code>epigraphdb</code></a> is an R package to provide ease of access to EpiGraphDB services. We will refer to <code>epigraphdb</code> as the name of the R package whereas <code>"EpiGraphDB"</code> as the overall platform.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install the latest development version from github ( <a href="https://devtools.r-lib.org/"><code>devtools</code></a> is required ):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"MRCIEU/epigraphdb-r"</span><span class="op">)</span></code></pre></div>
<p>To install a stable version from CRAN:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"epigraphdb"</span><span class="op">)</span></code></pre></div>
<p><strong>NOTE</strong>: while the package repository is “epigraphdb-r”, the R package name is “epigraphdb”.</p>
</div>
<div id="using-epigraphdb" class="section level2">
<h2 class="hasAnchor">
<a href="#using-epigraphdb" class="anchor"></a>Using <code>epigraphdb</code>
</h2>
<p><code>epigraphdb</code> provides a simple and intuitive way to query the API, as:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://mrcieu.github.io/epigraphdb-r/">"epigraphdb"</a></span><span class="op">)</span>
<span class="co">#&gt;</span>
<span class="co">#&gt;   EpiGraphDB v0.3 (API: https://api.epigraphdb.org)</span>
<span class="co">#&gt;</span>
<span class="fu"><a href="reference/mr.html">mr</a></span><span class="op">(</span>outcome_trait <span class="op">=</span> <span class="st">"Body mass index"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 370 x 12</span>
<span class="co">#&gt;    exposure_id exposure_name outcome_id outcome_name estimate      se</span>
<span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;         &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 627         Epiandroster… 785        Body mass i…   0.0950 2.28e-3</span>
<span class="co">#&gt;  2 541         X-11787       835        Body mass i…  -0.0578 1.77e-4</span>
<span class="co">#&gt;  3 971         Ulcerative c… 835        Body mass i…  -0.0111 1.76e-4</span>
<span class="co">#&gt;  4 60          Waist circum… 835        Body mass i…   0.861  2.07e-2</span>
<span class="co">#&gt;  5 UKB-a:426   Eye problems… 94         Body mass i…  -1.12   1.90e-2</span>
<span class="co">#&gt;  6 UKB-a:373   Ever depress… 95         Body mass i…  -0.616  4.80e-4</span>
<span class="co">#&gt;  7 29          Birth length  95         Body mass i…  -0.141  5.67e-4</span>
<span class="co">#&gt;  8 350         Laurate (12:… 974        Body mass i…   0.418  7.10e-3</span>
<span class="co">#&gt;  9 UKB-a:124   Treatment/me… 974        Body mass i…  -5.14   1.08e-1</span>
<span class="co">#&gt; 10 95          Body mass in… 974        Body mass i…   0.981  2.79e-2</span>
<span class="co">#&gt; # … with 360 more rows, and 6 more variables: p &lt;dbl&gt;, ci_upp &lt;dbl&gt;,</span>
<span class="co">#&gt; #   ci_low &lt;dbl&gt;, selection &lt;chr&gt;, method &lt;chr&gt;, moescore &lt;dbl&gt;)</span></code></pre></div>
<p>For more information on how to use the <code>epigraphdb</code> R package and how to use the API in R please check out the following articles:</p>
<table class="table">
<colgroup><col width="100%"></colgroup>
<thead><tr class="header">
<th>Article</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a></td>
</tr>
<tr class="even">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/using-epigraphdb-r-package.html">Using EpiGraphDB R package</a></td>
</tr>
<tr class="odd">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/using-epigraphdb-api.html">Using EpiGraphDB API (from R and command line)</a></td>
</tr>
<tr class="even">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/options.html">Package options</a></td>
</tr>
<tr class="odd">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a></td>
</tr>
<tr class="even">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a></td>
</tr>
<tr class="odd">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a></td>
</tr>
<tr class="even">
<td><a href="https://mrcieu.github.io/epigraphdb-r/articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a></td>
</tr>
</tbody>
</table>
</div>
<div id="package-functionalities" class="section level2">
<h2 class="hasAnchor">
<a href="#package-functionalities" class="anchor"></a>Package functionalities</h2>
<p>Users can use the general query function <a href="https://mrcieu.github.io/epigraphdb-r/reference/query_epigraphdb.html">query_epigraphdb</a> to get data from an API endpoint on EpiGraphDB without having to deal with HTTP requests by themselves. We also provide a list of functions (see the table below) that are equivalent to the upstream endpoints for the ease of use.</p>
<table class="table table">
<thead><tr class="header">
<th>
<a href="https://api.epigraphdb.org/">API</a>
</th>
<th>
<a href="https://github.com/MRCIEU/epigraphdb-r">r package</a>
</th>
</tr></thead>
<tbody>
<tr>
<td>
<strong>General query</strong>
</td>
<td>
</td>
</tr>
<tr>
<td>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/query_epigraphdb.html"><code>query_epigraphdb</code></a>
</td>
</tr>
<tr>
<td>
<strong>Topic queries</strong>
</td>
<td>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-mr"><code>GET /mr</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/mr.html"><code>mr</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-ontologygwas-efo"><code>GET /ontology/gwas-efo</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/ontology_gwas_efo.html"><code>ontology_gwas_efo</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-obs-cor"><code>GET /obs-cor</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/obs_cor.html"><code>obs_cor</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-genetic-cor"><code>GET /genetic-cor</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/genetic_cor.html"><code>genetic_cor</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtl"><code>GET /pqtl/</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/pqtl.html"><code>pqtl</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtlpleio"><code>GET /pqtl/pleio/</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/pqtl_pleio.html"><code>pqtl_pleio</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pqtllist"><code>GET /pqtl/list</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/pqtl_list.html"><code>pqtl_list</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-confounder"><code>GET /confounder</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/confounder.html"><code>confounder</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-pathway"><code>GET /pathway</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/pathway.html"><code>pathway</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-drugsrisk-factors"><code>GET /drugs/risk-factors</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/drugs_risk_factors.html"><code>drugs_risk_factors</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlmulti-snp-mr"><code>GET /xqtl/multi-snp-mr</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/xqtl_multi_snp_mr.html"><code>xqtl_multi_snp_mr</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlsingle-snp-mr"><code>GET /xqtl/single-snp-mr</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/xqtl_single_snp_mr.html"><code>xqtl_single_snp_mr</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-literaturegwas"><code>GET /literature/gwas</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/literature_gwas.html"><code>literature_gwas</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#post-proteinin-pathway"><code>POST /protein/in-pathway</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/protein_in_pathway.html"><code>protein_in_pathway</code></a>
</td>
</tr>
<tr>
<td>
<strong>Utilities</strong>
</td>
<td>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#post-mappingsgene-to-protein"><code>POST /mappings/gene-to-protein</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/mappings_gene_to_protein.html"><code>mappings_gene_to_protein</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodeslist"><code>GET /meta/nodes/list</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/meta_nodes_list.html"><code>meta_nodes_list</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanrelslist"><code>GET /meta/rels/list</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/meta_rels_list.html"><code>meta_rels_list</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodesmeta_nodelist"><code>GET /meta/nodes/{meta_node}/list</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/meta_nodes_list_node.html"><code>meta_nodes_list_node</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metarelsmeta_rellist"><code>GET /meta/rels/{meta_rel}/list</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/meta_rels_list_rel.html"><code>meta_rels_list_rel</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#get-metanodesmeta_nodesearch"><code>GET /meta/nodes/{meta_node}/search</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/meta_nodes_search_node.html"><code>meta_nodes_search_node</code></a>
</td>
</tr>
<tr>
<td>
<a href="https://docs.epigraphdb.org/api/api-endpoints/#post-cypher"><code>POST /cypher</code></a>
</td>
<td>
<a href="https://mrcieu.github.io/epigraphdb-r/reference/cypher.html"><code>cypher</code></a>
</td>
</tr>
</tbody>
</table>
</div>
<div id="epigraphdb-resources" class="section level2">
<h2 class="hasAnchor">
<a href="#epigraphdb-resources" class="anchor"></a>EpiGraphDB resources</h2>
<table class="table">
<thead><tr class="header">
<th>link</th>
<th>screenshot</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><a href="https://docs.epigraphdb.org">docs</a></td>
<td><img src="articles/figures/epigraphdb-docs.png" alt="docs"></td>
</tr>
<tr class="even">
<td><a href="https://api.epigraphdb.org">API</a></td>
<td><img src="articles/figures/epigraphdb-api-swagger.png" alt="api"></td>
</tr>
<tr class="odd">
<td><a href="https://epigraphdb.org">web application</a></td>
<td><img src="articles/figures/epigraphdb-xqtl-view.png" alt="webapp"></td>
</tr>
<tr class="even">
<td><a href="https://github.com/MRCIEU/epigraphdb-r">r package</a></td>
<td><img src="articles/figures/epigraphdb-r.png" alt="epigraphdb-r"></td>
</tr>
</tbody>
</table>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Please cite EpiGraphDB as</p>
<blockquote>
<p>Yi Liu, Benjamin Elsworth, Pau Erola, Valeriia Haberland, Gibran Hemani, Matt Lyon, Jie Zheng, Oliver Lloyd, Marina Vabistsevits, Tom R Gaunt, EpiGraphDB: a database and data mining platform for health data science, Bioinformatics, btaa961, <a href="https://doi.org/10.1093/bioinformatics/btaa961" class="uri">https://doi.org/10.1093/bioinformatics/btaa961</a></p>
</blockquote>
<pre><code>@article{epigraphdb2020bioinformatics,
    author = {Liu, Yi and Elsworth, Benjamin and Erola, Pau and Haberland, Valeriia and Hemani, Gibran and Lyon, Matt and Zheng, Jie and Lloyd, Oliver and Vabistsevits, Marina and Gaunt, Tom R},
    title = {{EpiGraphDB}: a database and data mining platform for health data science},
    journal = {Bioinformatics},
    year = {2020},
    month = {11},
    issn = {1367-4803},
    doi = {10.1093/bioinformatics/btaa961},
    url = {https://doi.org/10.1093/bioinformatics/btaa961},
    note = {btaa961},
    eprint = {https://academic.oup.com/bioinformatics/advance-article-pdf/doi/10.1093/bioinformatics/btaa961/34178613/btaa961.pdf}
}</code></pre>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>Please get in touch with us for issues, comments, suggestions, etc. via the following methods:</p>
<ul>
<li><a href="https://github.com/MRCIEU/epigraphdb/issues">The issue tracker on the <code>epigraphdb</code> repo</a></li>
<li><a href="mailto:feedback@epigraphdb.org">The support email</a></li>
<li><a href="https://twitter.com/epigraphdb">The EpiGraphDB twitter</a></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=epigraphdb">https://​cloud.r-project.org/​package=epigraphdb</a>
</li>
<li>Browse source code at <br><a href="https://github.com/MRCIEU/epigraphdb-r/">https://​github.com/​MRCIEU/​epigraphdb-r/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/MRCIEU/epigraphdb-r/issues">https://​github.com/​MRCIEU/​epigraphdb-r/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://yiliu6240.github.io">Yi Liu</a> <br><small class="roles"> Maintainer, author </small>  </li>
<li>Valeriia Haberland <br><small class="roles"> Author </small>  </li>
<li>
<a href="https://marinalearning.netlify.com">Marina Vabistsevits</a> <br><small class="roles"> Author </small>  </li>
<li>
<a href="http://www.biocompute.org.uk/">Tom Gaunt</a> <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://cran.r-project.org/package=epigraphdb"><img src="https://www.r-pkg.org/badges/version/epigraphdb" alt="CRAN status"></a></li>
<li><a href="https://codecov.io/gh/MRCIEU/epigraphdb-r?branch=master"><img src="https://codecov.io/gh/MRCIEU/epigraphdb-r/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
