---
title: "Case study 2: Identification of potential drug targets"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Case study 2: Identification of potential drug targets}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This RMarkdown document demonstrates how key elements from the
[notebook for case study 2 in the EpiGraphDB paper](
https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-2-alt-drug-target.ipynb
) can be achieved using the R package.
For detailed explanations of the case study please refer to the
[paper](#)
or the
[case study notebook](
https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-2-alt-drug-target.ipynb
).

## Context

Systematic MR of molecular phenotypes such as proteins and expression of transcript levels offer enormous potential to prioritise drug targets for further investigation.
However, many genes and gene products are not easily druggable, so some potentially important causal genes may not offer an obvious route to intervention.

A parallel problem is that current GWASes of molecular phenotypes have limited sample sizes and limited protein coverages.
A potential way to address both these problems is to use  protein-protein interaction information to identify druggable targets which are linked to a non-druggable, but robustly causal target.
Their relationship to the causal target increases our confidence in their potential causal role even if the initial evidence of effect is below our multiple-testing threshold.

Here in case study 2 we demonstrate an approach to use data in EpiGraphDB to
prioritise potential alternative drug targets in the same PPI network, as follows:

- For an existing drug target of interests, we use PPI networks to search for its directly interacting genes that are evidenced to be druggable.
- We then examine the causal evidence of these candidate genes on the disease.
- We also examine the literature evidence of these candidate genes on the disease.

The triangulation of MR evidence and literature evidence as available from EpiGraphDB regarding these candidate genes will greatly enhance our confidence in identifying potential viable drug targets.

```{r}
library("magrittr")
library("dplyr")
library("purrr")
library("glue")
library("epigraphdb")
```

Here we configure the parameters used in the case study example.
We illustrate this approach using IL23R, an established drug target for inflammatory bowel disease (IBD) (Duerr et al., 2006; Momozawa et al., 2011).

While specific IL23R interventions are still undergoing trials, there is a possibility that these therapies may not be effective for all or even the majority of patients.
This case study therefore explores potential alternative drug targets.

```{r}
GENE_NAME <- "IL23R"
OUTCOME_TRAIT <- "Inflammatory bowel disease"
```

## Using PPI networks for alternative drug targets search

The assumption here is that the most likely alternative targets are either directly interacting with IL23R or somewhere in the same PPI network.
In this example, we consider only genes that were found to interact with IL23R via direct protein-protein interactions, and require that those **interacting proteins** should also be **druggable**.

The thousands of genes are classified with regard to their druggability by Finan et al. 2017, where the **Tier 1** category refers to  approved drugs or those in clinical testing while for other tier categories the druggability confidence drops in order **Tier 2** and then **Tier 3**.

Here we use the
[GET /gene/druggability/ppi](https://docs.epigraphdb.org/api/api-endpoints/#get-genedruggabilityppi)
endpoint to get data on the druggable alternative genes.**

```{r}
get_drug_targets_ppi <- function(gene_name) {
  endpoint <- "/gene/druggability/ppi"
  params <- list(gene_name = gene_name)
  df <- query_epigraphdb(route = endpoint, params = params, mode = "table")
  df
}

ppi_df <- get_drug_targets_ppi(gene_name = GENE_NAME)
ppi_df
```

For further analysis we select the gene of interest (IL23R) as well as its interacting genes with Tier 1 druggability.

```{r}
get_gene_list <- function(ppi_df, include_primary_gene = TRUE) {
  if (include_primary_gene) {
    gene_list <- c(
      ppi_df %>% pull(`g1.name`) %>% unique(),
      ppi_df %>% filter(`g2.druggability_tier` == "Tier 1") %>% pull(`g2.name`)
    )
  } else {
    gene_list <- ppi_df %>%
      filter(`g2.druggability_tier` == "Tier 1") %>%
      pull(`g2.name`)
  }
  gene_list
}

gene_list <- get_gene_list(ppi_df)
gene_list
```

## Using Mendelian randomization results for causal effect estimation

The next step is to find out whether any of these genes have a comparable and statistically plausible effect on IBD.

Here we search EpiGraphDB for the Mendelian randomization (MR) results for these genes and IBD from the recent study by Zheng et al, 2019 (https://epigraphdb.org/xqtl/) via the
[GET /xqtl/single-snp-mr](https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlsingle-snp-mr)
endpoint.

```{r}
extract_mr <- function(outcome_trait, gene_list, qtl_type) {
  endpoint <- "/xqtl/single-snp-mr"
  per_gene <- function(gene_name) {
    params <- list(
      exposure_gene = gene_name,
      outcome_trait = outcome_trait,
      qtl_type = qtl_type,
      pval_threshold = 1e-5
    )
    df <- query_epigraphdb(route = endpoint, params = params, mode = "table")
    df
  }
  res_df <- gene_list %>% map_df(per_gene)
  res_df
}

xqtl_df <- c("pQTL", "eQTL") %>% map_df(function(qtl_type) {
  extract_mr(
    outcome_trait = OUTCOME_TRAIT,
    gene_list = gene_list,
    qtl_type = qtl_type
  ) %>%
    mutate(qtl_type = qtl_type)
})
xqtl_df
```

## Using literature evidence for results enrichment and triangulation

Can we find evidence in the literature where these genes are found to be associated with IBD to increase our level of confidence in MR results or to provide alternative evidence where MR results to not exist?

We can use the
[GET /gene/literature](https://docs.epigraphdb.org/api/api-endpoints/#get-geneliterature)
endpoint to get data on the literature evidence for the set of genes.

```{r}
extract_literature <- function(outcome_trait, gene_list) {
  per_gene <- function(gene_name) {
    endpoint <- "/gene/literature"
    params <- list(
      gene_name = gene_name,
      object_name = outcome_trait %>% stringr::str_to_lower()
    )
    df <- query_epigraphdb(route = endpoint, params = params, mode = "table")
    df
  }
  res_df <- gene_list %>% map_df(per_gene)
  res_df %>%
    mutate(literature_count = map_int(pubmed_id, function(x) length(x)))
}

literature_df <- extract_literature(
  outcome_trait = OUTCOME_TRAIT,
  gene_list = gene_list
)
literature_df
```

## `sessionInfo`

```{r}
sessionInfo()
```
