<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="case-1-pleiotropy_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/case-1-pleiotropy.Rmd"><code>vignettes/case-1-pleiotropy.Rmd</code></a></small>
      <div class="hidden name"><code>case-1-pleiotropy.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-1-pleiotropy.ipynb">notebook for case study 1 in the EpiGraphDB paper</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="#">paper</a> or the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-1-pleiotropy.ipynb">case study notebook</a>.</p>
<div id="context" class="section level2">
<h2 class="hasAnchor">
<a href="#context" class="anchor"></a>Context</h2>
<p>Mendelian randomization (MR), a technique to evaluate the causal role of modifiable exposures on a health outcome using a set of genetic instruments, tacitly assumes that a genetic variant (e.g. SNP) is only related to an outcome of interest through an exposure (i.e. the “exclusion restriction criterion”). Horizontal pleiotropy, where a SNP is associated with multiple phenotypes independently of the exposure of interest, potentially violates this assumption delivering misleading conclusions. In contrast, vertical pleiotropy, where a SNP is associated with multiple phenotypes on the same biological pathway, does not violate this assumption.</p>
<p>Here, we use external evidence on biological pathways and protein-protein interactions to assess the pleiotropic profile for a genetic variant based on the genes (and proteins) with which that variant is associated. In a graph representation, we will show that:</p>
<ul>
<li>We have evidence of potential <strong>vertical</strong> pleiotropy when the proteins associated with a variant belong to one connected group,</li>
<li>In contrast, if the proteins associated with a single variant are isolated from each other, this suggests evidence of <strong>horizontal</strong> pleiotropy.</li>
</ul>
<p>This case study goes as follows:</p>
<ul>
<li>For a list of genes that we wish to assess their pleiotropic profiles (against a variant of interests), we retrieve from EpiGraphDB their mapped protein products.</li>
<li>Then respectively from the evidence on shared pathway network and protein-protein interaction network, we can extract the proteins groups formed in these networks as connected communities to assess the pleiotropic profiles of the genes.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://magrittr.tidyverse.org">"magrittr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org">"dplyr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://purrr.tidyverse.org">"purrr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/tidyverse/glue">"glue"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://igraph.org">"igraph"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://mrcieu.github.io/epigraphdb-r/">"epigraphdb"</a></span><span class="op">)</span></code></pre></div>
<p>Here we configure the parameters used in the case study example. In this example we will look at the variant <strong>rs12720356</strong>, a SNP located in chromosome 19 that has been associated with Crohn’s disease and psoriasis.**</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SNP</span> <span class="op">&lt;-</span> <span class="st">"rs12720356"</span>

<span class="va">GENELIST</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ZGLP1"</span>, <span class="st">"FDX1L"</span>, <span class="st">"MRPL4"</span>, <span class="st">"ICAM5"</span>, <span class="st">"TYK2"</span>, <span class="st">"GRAP2"</span>, <span class="st">"KRI1"</span>, <span class="st">"TMED1"</span>, <span class="st">"ICAM1"</span><span class="op">)</span>

<span class="va">PPI_N_INTERMEDIATE_PROTEINS</span> <span class="op">&lt;-</span> <span class="fl">1</span></code></pre></div>
</div>
<div id="mapping-genes-to-proteins" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-genes-to-proteins" class="anchor"></a>Mapping genes to proteins</h2>
<p>The first step of the analysis is to map each of these genes to their protein product.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_gene_protein</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">genelist</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mappings/gene-to-protein"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    gene_name_list <span class="op">=</span> <span class="va">genelist</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
    route <span class="op">=</span> <span class="va">endpoint</span>,
    params <span class="op">=</span> <span class="va">params</span>,
    mode <span class="op">=</span> <span class="st">"table"</span>,
    method <span class="op">=</span> <span class="st">"POST"</span>
  <span class="op">)</span>
  <span class="va">protein_df</span> <span class="op">&lt;-</span> <span class="va">r</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">protein_df</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">protein_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="va">`gene.name`</span>, uniprot_id <span class="op">=</span> <span class="va">`protein.uniprot_id`</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html">set_names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene_name"</span>, <span class="st">"uniprot_id"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">}</span>

  <span class="va">res_df</span>
<span class="op">}</span>

<span class="va">gene_protein_df</span> <span class="op">&lt;-</span> <span class="fu">get_gene_protein</span><span class="op">(</span>genelist <span class="op">=</span> <span class="va">GENELIST</span><span class="op">)</span>
<span class="va">gene_protein_df</span>
<span class="co">#&gt; # A tibble: 9 x 2</span>
<span class="co">#&gt;   gene_name uniprot_id</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;     </span>
<span class="co">#&gt; 1 ZGLP1     P0C6A0    </span>
<span class="co">#&gt; 2 FDX1L     Q6P4F2    </span>
<span class="co">#&gt; 3 MRPL4     Q9BYD3    </span>
<span class="co">#&gt; 4 ICAM5     Q9UMF0    </span>
<span class="co">#&gt; 5 TYK2      P29597    </span>
<span class="co">#&gt; 6 GRAP2     O75791    </span>
<span class="co">#&gt; 7 KRI1      Q8N9T8    </span>
<span class="co">#&gt; 8 TMED1     Q13445    </span>
<span class="co">#&gt; 9 ICAM1     P05362</span></code></pre></div>
</div>
<div id="identifying-the-involvement-in-the-same-biological-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#identifying-the-involvement-in-the-same-biological-pathways" class="anchor"></a>Identifying the involvement in the same biological pathways</h2>
<p>For each protein we retrieve the pathways they are involved in.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_protein_pathway</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gene_protein_df</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/protein/in-pathway"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    uniprot_id_list <span class="op">=</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`uniprot_id`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span>, method <span class="op">=</span> <span class="st">"POST"</span><span class="op">)</span>

  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`uniprot_id`</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">df</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"uniprot_id"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`uniprot_id`</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>pathway_count <span class="op">=</span> <span class="cn">NA_integer_</span>, pathway_reactome_id <span class="op">=</span> <span class="cn">NA_character_</span><span class="op">)</span>
  <span class="op">}</span>
  <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">res_df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
      pathway_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">pathway_count</span><span class="op">)</span>, <span class="fl">0L</span>, <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">pathway_count</span><span class="op">)</span><span class="op">)</span>,
      pathway_reactome_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">pathway_reactome_id</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>, <span class="va">pathway_reactome_id</span><span class="op">)</span>
    <span class="op">)</span>

  <span class="va">res_df</span>
<span class="op">}</span>
<span class="va">pathway_df</span> <span class="op">&lt;-</span> <span class="fu">get_protein_pathway</span><span class="op">(</span>gene_protein_df <span class="op">=</span> <span class="va">gene_protein_df</span><span class="op">)</span>
<span class="va">pathway_df</span>
<span class="co">#&gt; # A tibble: 9 x 3</span>
<span class="co">#&gt;   uniprot_id pathway_count pathway_reactome_id</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt; &lt;list&gt;             </span>
<span class="co">#&gt; 1 P0C6A0                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 2 Q6P4F2                15 &lt;chr [15]&gt;         </span>
<span class="co">#&gt; 3 Q9BYD3                 6 &lt;chr [6]&gt;          </span>
<span class="co">#&gt; 4 Q9UMF0                 5 &lt;chr [5]&gt;          </span>
<span class="co">#&gt; 5 P29597                28 &lt;chr [28]&gt;         </span>
<span class="co">#&gt; 6 O75791                15 &lt;chr [15]&gt;         </span>
<span class="co">#&gt; 7 Q8N9T8                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 8 Q13445                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 9 P05362                11 &lt;chr [11]&gt;</span></code></pre></div>
<p>Now for each pair of proteins we match the pathways they have in common.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_shared_pathway</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pathway_df</span><span class="op">)</span> <span class="op">{</span>
  <span class="co"># For the protein-pathway data</span>
  <span class="co"># Get protein-protein permutations where they share pathways</span>

  <span class="va">per_permutation</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pathway_df</span>, <span class="va">permutation</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">df</span> <span class="op">&lt;-</span> <span class="va">pathway_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">uniprot_id</span> <span class="op">%in%</span> <span class="va">permutation</span><span class="op">)</span>
    <span class="va">primary_pathway</span> <span class="op">&lt;-</span> <span class="va">pathway_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">uniprot_id</span> <span class="op">==</span> <span class="va">permutation</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">pathway_reactome_id</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span>
    <span class="va">assoc_pathway</span> <span class="op">&lt;-</span> <span class="va">pathway_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">uniprot_id</span> <span class="op">==</span> <span class="va">permutation</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">pathway_reactome_id</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html">intersect</a></span><span class="op">(</span><span class="va">primary_pathway</span>, <span class="va">assoc_pathway</span><span class="op">)</span>
  <span class="op">}</span>

  <span class="va">pairwise_permutations</span> <span class="op">&lt;-</span> <span class="va">pathway_df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`uniprot_id`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">gtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gtools/man/combinations.html">permutations</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, r <span class="op">=</span> <span class="fl">2</span>, v <span class="op">=</span> <span class="va">.</span><span class="op">)</span>
  <span class="va">shared_pathway_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
    protein <span class="op">=</span> <span class="va">pairwise_permutations</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>,
    assoc_protein <span class="op">=</span> <span class="va">pairwise_permutations</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
      shared_pathway <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">`protein`</span>, <span class="va">`assoc_protein`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="op">{</span>
        <span class="fu">per_permutation</span><span class="op">(</span>pathway_df <span class="op">=</span> <span class="va">pathway_df</span>, permutation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
      <span class="op">}</span><span class="op">)</span>,
      combination <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_chr</a></span><span class="op">(</span><span class="va">`protein`</span>, <span class="va">`assoc_protein`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">comb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">comb</span>, collapse <span class="op">=</span> <span class="st">","</span><span class="op">)</span>
      <span class="op">}</span><span class="op">)</span>,
      count <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_int</a></span><span class="op">(</span><span class="va">`shared_pathway`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,
      connected <span class="op">=</span> <span class="va">count</span> <span class="op">&gt;</span> <span class="fl">0</span>
    <span class="op">)</span>

  <span class="va">shared_pathway_df</span>
<span class="op">}</span>
<span class="va">shared_pathway_df</span> <span class="op">&lt;-</span> <span class="fu">get_shared_pathway</span><span class="op">(</span><span class="va">pathway_df</span><span class="op">)</span>
<span class="va">n_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">shared_pathway_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">count</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span><span class="op">(</span><span class="st">"Num. shared_pathway pairs: {n_pairs}"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Num. shared_pathway pairs: 6</span>
<span class="va">shared_pathway_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 72 x 6</span>
<span class="co">#&gt;    protein assoc_protein shared_pathway combination   count connected</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;list&gt;         &lt;chr&gt;         &lt;int&gt; &lt;lgl&gt;    </span>
<span class="co">#&gt;  1 P05362  P29597        &lt;chr [6]&gt;      P05362,P29597     6 TRUE     </span>
<span class="co">#&gt;  2 P29597  P05362        &lt;chr [6]&gt;      P05362,P29597     6 TRUE     </span>
<span class="co">#&gt;  3 P05362  Q9UMF0        &lt;chr [5]&gt;      P05362,Q9UMF0     5 TRUE     </span>
<span class="co">#&gt;  4 Q9UMF0  P05362        &lt;chr [5]&gt;      P05362,Q9UMF0     5 TRUE     </span>
<span class="co">#&gt;  5 O75791  P05362        &lt;chr [2]&gt;      O75791,P05362     2 TRUE     </span>
<span class="co">#&gt;  6 O75791  P29597        &lt;chr [2]&gt;      O75791,P29597     2 TRUE     </span>
<span class="co">#&gt;  7 O75791  Q9UMF0        &lt;chr [2]&gt;      O75791,Q9UMF0     2 TRUE     </span>
<span class="co">#&gt;  8 P05362  O75791        &lt;chr [2]&gt;      O75791,P05362     2 TRUE     </span>
<span class="co">#&gt;  9 P29597  O75791        &lt;chr [2]&gt;      O75791,P29597     2 TRUE     </span>
<span class="co">#&gt; 10 Q9UMF0  O75791        &lt;chr [2]&gt;      O75791,Q9UMF0     2 TRUE     </span>
<span class="co">#&gt; # … with 62 more rows</span></code></pre></div>
<p>We can further query EpiGraphDB regarding the detailed pathway information using <code>GET /meta/nodes/Pathway/search</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_pathway_info</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">reactome_id</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Pathway/search"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>id <span class="op">=</span> <span class="va">reactome_id</span><span class="op">)</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="va">df</span>
<span class="op">}</span>

<span class="va">pathway</span> <span class="op">&lt;-</span> <span class="va">shared_pathway_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">shared_pathway</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">pathway_info</span> <span class="op">&lt;-</span> <span class="va">pathway</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="va">get_pathway_info</span><span class="op">)</span>
<span class="va">pathway_info</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 x 6</span>
<span class="co">#&gt;    node._name        node.name        node._source node.id node._id node.url    </span>
<span class="co">#&gt;    &lt;chr&gt;             &lt;chr&gt;            &lt;list&gt;       &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;       </span>
<span class="co">#&gt;  1 Immune System     Immune System    &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt;  2 Adaptive Immune … Adaptive Immune… &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt;  3 Signal Transduct… Signal Transduc… &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt;  4 Interferon Signa… Interferon Sign… &lt;chr [1]&gt;    R-HSA-… R-HSA-9… https://rea…</span>
<span class="co">#&gt;  5 Interleukin-4 an… Interleukin-4 a… &lt;chr [1]&gt;    R-HSA-… R-HSA-6… https://rea…</span>
<span class="co">#&gt;  6 Interleukin-10 s… Interleukin-10 … &lt;chr [1]&gt;    R-HSA-… R-HSA-6… https://rea…</span>
<span class="co">#&gt;  7 Signaling by Int… Signaling by In… &lt;chr [1]&gt;    R-HSA-… R-HSA-4… https://rea…</span>
<span class="co">#&gt;  8 Cytokine Signali… Cytokine Signal… &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt;  9 Integrin cell su… Integrin cell s… &lt;chr [1]&gt;    R-HSA-… R-HSA-2… https://rea…</span>
<span class="co">#&gt; 10 Immunoregulatory… Immunoregulator… &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt; 11 Extracellular ma… Extracellular m… &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span>
<span class="co">#&gt; 12 Disease           Disease          &lt;chr [1]&gt;    R-HSA-… R-HSA-1… https://rea…</span></code></pre></div>
<p>In order to extract protein groups from the shared pathways, the last step for this query is to convert the shared pathway data into a graph where</p>
<ul>
<li>Proteins are represented as nodes.</li>
<li>And shared pathways are represented as edges.</li>
<li>Then the protein groups (via shared pathways) are the connected communities in the graph.</li>
</ul>
<p>We then count the number of nodes in each connected community and plot the graph.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">protein_df_to_graph</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">df_connected</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">connected</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">`combination`</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">nodes</span> <span class="op">&lt;-</span> <span class="va">df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">graph</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span><span class="op">(</span>
    <span class="va">df_connected</span>,
    directed <span class="op">=</span> <span class="cn">FALSE</span>, vertices <span class="op">=</span> <span class="va">nodes</span>
  <span class="op">)</span>
  <span class="va">graph</span><span class="op">$</span><span class="va">layout</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/igraph/man/layout_with_kk.html">layout_with_kk</a></span>
  <span class="va">graph</span>
<span class="op">}</span>

<span class="va">graph_to_protein_groups</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">graph</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">graph</span> <span class="op">%&gt;%</span>
    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/components.html">components</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/groups.html">groups</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>group_member <span class="op">=</span> <span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>group_size <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_int</a></span><span class="op">(</span><span class="va">`group_member`</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">group_size</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">pathway_protein_graph</span> <span class="op">&lt;-</span> <span class="va">shared_pathway_df</span> <span class="op">%&gt;%</span> <span class="fu">protein_df_to_graph</span><span class="op">(</span><span class="op">)</span>
<span class="va">pathway_protein_groups</span> <span class="op">&lt;-</span> <span class="va">pathway_protein_graph</span> <span class="op">%&gt;%</span> <span class="fu">graph_to_protein_groups</span><span class="op">(</span><span class="op">)</span>
<span class="va">pathway_protein_groups</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; tibble [5 × 2] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;  $ group_member:List of 5</span>
<span class="co">#&gt;   ..$ 1: chr [1:5] "O75791" "P05362" "P29597" "Q6P4F2" ...</span>
<span class="co">#&gt;   ..$ 2: chr "P0C6A0"</span>
<span class="co">#&gt;   ..$ 3: chr "Q13445"</span>
<span class="co">#&gt;   ..$ 4: chr "Q8N9T8"</span>
<span class="co">#&gt;   ..$ 5: chr "Q9BYD3"</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int 5</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 1</span>
<span class="co">#&gt;   .. ..$ : chr [1:5] "1" "2" "3" "4" ...</span>
<span class="co">#&gt;  $ group_size  : Named int [1:5] 5 1 1 1 1</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:5] "1" "2" "3" "4" ...</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">pathway_protein_graph</span><span class="op">)</span></code></pre></div>
<p><img src="case-1-pleiotropy_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="retrieving-shared-protein-protein-interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieving-shared-protein-protein-interactions" class="anchor"></a>Retrieving shared protein-protein interactions</h2>
<p>We can complement the information given by pathway ontologies with data of interactions between proteins. For each pair of proteins, we retrieve shared protein-protein interactions, either a direct interaction or an interaction where there’s one mediator protein in the middle (see parameters section).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_ppi</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gene_protein_df</span>, <span class="va">n_intermediate_proteins</span> <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/protein/ppi/pairwise"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    uniprot_id_list <span class="op">=</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="st">"uniprot_id"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="op">)</span>,
    n_intermediate_proteins <span class="op">=</span> <span class="va">n_intermediate_proteins</span>
  <span class="op">)</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
    route <span class="op">=</span> <span class="va">endpoint</span>,
    params <span class="op">=</span> <span class="va">params</span>,
    mode <span class="op">=</span> <span class="st">"table"</span>,
    method <span class="op">=</span> <span class="st">"POST"</span>
  <span class="op">)</span>

  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>protein <span class="op">=</span> <span class="va">uniprot_id</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">df</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"protein"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_protein_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>protein <span class="op">=</span> <span class="va">uniprot_id</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>assoc_protein <span class="op">=</span> <span class="cn">NA_character_</span>, path_size <span class="op">=</span> <span class="cn">NA_integer_</span><span class="op">)</span>
  <span class="op">}</span>
  <span class="va">res_df</span>
<span class="op">}</span>
<span class="va">ppi_df</span> <span class="op">&lt;-</span> <span class="fu">get_ppi</span><span class="op">(</span><span class="va">gene_protein_df</span>, n_intermediate_proteins <span class="op">=</span> <span class="va">PPI_N_INTERMEDIATE_PROTEINS</span><span class="op">)</span>
<span class="va">ppi_df</span>
<span class="co">#&gt; # A tibble: 1,686 x 3</span>
<span class="co">#&gt;    protein assoc_protein path_size</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;             &lt;int&gt;</span>
<span class="co">#&gt;  1 P0C6A0  P0C6A0                2</span>
<span class="co">#&gt;  2 P0C6A0  P0C6A0                2</span>
<span class="co">#&gt;  3 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  4 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  5 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  6 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  7 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  8 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt;  9 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt; 10 Q6P4F2  Q6P4F2                2</span>
<span class="co">#&gt; # … with 1,676 more rows</span></code></pre></div>
<p>Like in the query for pathway data, we convert and analyse these results as a graph.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_shared_ppi</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ppi_df</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">per_row</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ppi_df</span>, <span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="va">ppi_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
        <span class="va">`protein`</span> <span class="op">==</span> <span class="va">x</span>,
        <span class="va">`assoc_protein`</span> <span class="op">==</span> <span class="va">y</span>
      <span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu">`&gt;`</span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>
    <span class="va">res</span>
  <span class="op">}</span>

  <span class="va">pairwise_permutations</span> <span class="op">&lt;-</span> <span class="va">ppi_df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`protein`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">gtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gtools/man/combinations.html">permutations</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, r <span class="op">=</span> <span class="fl">2</span>, v <span class="op">=</span> <span class="va">.</span><span class="op">)</span>
  <span class="va">shared_ppi_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html">tibble</a></span><span class="op">(</span>
    protein <span class="op">=</span> <span class="va">pairwise_permutations</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>,
    assoc_protein <span class="op">=</span> <span class="va">pairwise_permutations</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>
      shared_ppi <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_lgl</a></span><span class="op">(</span><span class="va">`protein`</span>, <span class="va">`assoc_protein`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="op">{</span>
        <span class="fu">per_row</span><span class="op">(</span>ppi_df <span class="op">=</span> <span class="va">ppi_df</span>, x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span><span class="op">)</span>
      <span class="op">}</span><span class="op">)</span>,
      combination <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_chr</a></span><span class="op">(</span><span class="va">`protein`</span>, <span class="va">`assoc_protein`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">comb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">)</span><span class="op">)</span>
        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">comb</span>, collapse <span class="op">=</span> <span class="st">","</span><span class="op">)</span>
      <span class="op">}</span><span class="op">)</span>,
      connected <span class="op">=</span> <span class="va">shared_ppi</span>
    <span class="op">)</span>

  <span class="va">shared_ppi_df</span>
<span class="op">}</span>

<span class="va">shared_ppi_df</span> <span class="op">&lt;-</span> <span class="fu">get_shared_ppi</span><span class="op">(</span><span class="va">ppi_df</span><span class="op">)</span>
<span class="va">n_ppi_pairs</span> <span class="op">&lt;-</span> <span class="va">shared_ppi_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">shared_ppi</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span><span class="op">(</span><span class="st">"Num. shared_ppi pairs: {n_ppi_pairs}"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Num. shared_ppi pairs: 10</span>
<span class="va">shared_ppi_df</span>
<span class="co">#&gt; # A tibble: 72 x 5</span>
<span class="co">#&gt;    protein assoc_protein shared_ppi combination   connected</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;lgl&gt;      &lt;chr&gt;         &lt;lgl&gt;    </span>
<span class="co">#&gt;  1 O75791  P05362        TRUE       O75791,P05362 TRUE     </span>
<span class="co">#&gt;  2 O75791  P0C6A0        FALSE      O75791,P0C6A0 FALSE    </span>
<span class="co">#&gt;  3 O75791  P29597        TRUE       O75791,P29597 TRUE     </span>
<span class="co">#&gt;  4 O75791  Q13445        FALSE      O75791,Q13445 FALSE    </span>
<span class="co">#&gt;  5 O75791  Q6P4F2        FALSE      O75791,Q6P4F2 FALSE    </span>
<span class="co">#&gt;  6 O75791  Q8N9T8        FALSE      O75791,Q8N9T8 FALSE    </span>
<span class="co">#&gt;  7 O75791  Q9BYD3        FALSE      O75791,Q9BYD3 FALSE    </span>
<span class="co">#&gt;  8 O75791  Q9UMF0        FALSE      O75791,Q9UMF0 FALSE    </span>
<span class="co">#&gt;  9 P05362  O75791        TRUE       O75791,P05362 TRUE     </span>
<span class="co">#&gt; 10 P05362  P0C6A0        FALSE      P05362,P0C6A0 FALSE    </span>
<span class="co">#&gt; # … with 62 more rows</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ppi_protein_graph</span> <span class="op">&lt;-</span> <span class="fu">protein_df_to_graph</span><span class="op">(</span><span class="va">shared_ppi_df</span><span class="op">)</span>
<span class="va">ppi_protein_groups</span> <span class="op">&lt;-</span> <span class="fu">graph_to_protein_groups</span><span class="op">(</span><span class="va">ppi_protein_graph</span><span class="op">)</span>
<span class="va">ppi_protein_groups</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; tibble [5 × 2] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;  $ group_member:List of 5</span>
<span class="co">#&gt;   ..$ 1: chr [1:4] "O75791" "P05362" "P29597" "Q9UMF0"</span>
<span class="co">#&gt;   ..$ 5: chr [1:2] "Q8N9T8" "Q9BYD3"</span>
<span class="co">#&gt;   ..$ 2: chr "P0C6A0"</span>
<span class="co">#&gt;   ..$ 3: chr "Q13445"</span>
<span class="co">#&gt;   ..$ 4: chr "Q6P4F2"</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int 5</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 1</span>
<span class="co">#&gt;   .. ..$ : chr [1:5] "1" "5" "2" "3" ...</span>
<span class="co">#&gt;  $ group_size  : Named int [1:5] 4 2 1 1 1</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:5] "1" "5" "2" "3" ...</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">ppi_protein_graph</span><span class="op">)</span></code></pre></div>
<p><img src="case-1-pleiotropy_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.4 (2021-02-15)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2.2 igraph_1.2.6     glue_1.4.2       purrr_0.3.4     </span>
<span class="co">#&gt; [5] dplyr_1.0.5      magrittr_2.0.1  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] highr_0.8         compiler_4.0.4    pillar_1.5.1      tools_4.0.4      </span>
<span class="co">#&gt;  [5] digest_0.6.27     jsonlite_1.7.2    evaluate_0.14     memoise_2.0.0    </span>
<span class="co">#&gt;  [9] lifecycle_1.0.0   tibble_3.1.0      pkgconfig_2.0.3   rlang_0.4.10     </span>
<span class="co">#&gt; [13] cli_2.3.1         rstudioapi_0.13   curl_4.3          yaml_2.2.1       </span>
<span class="co">#&gt; [17] pkgdown_1.6.1     xfun_0.22         fastmap_1.1.0     stringr_1.4.0    </span>
<span class="co">#&gt; [21] httr_1.4.2        knitr_1.31        gtools_3.8.2      generics_0.1.0   </span>
<span class="co">#&gt; [25] desc_1.3.0        fs_1.5.0          vctrs_0.3.6       systemfonts_1.0.1</span>
<span class="co">#&gt; [29] rprojroot_2.0.2   tidyselect_1.1.0  R6_2.5.0          textshaping_0.3.3</span>
<span class="co">#&gt; [33] fansi_0.4.2       rmarkdown_2.7     ps_1.6.0          htmltools_0.5.1.1</span>
<span class="co">#&gt; [37] ellipsis_0.3.1    assertthat_0.2.1  ragg_1.1.2        utf8_1.2.1       </span>
<span class="co">#&gt; [41] stringi_1.5.3     cachem_1.0.4      crayon_1.4.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
