<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study 3: Triangulating causal estimates with literature evidence • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Case study 3: Triangulating causal estimates with literature evidence">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="case-3-literature-triangulation_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study 3: Triangulating causal estimates with literature evidence</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/case-3-literature-triangulation.Rmd"><code>vignettes/case-3-literature-triangulation.Rmd</code></a></small>
      <div class="hidden name"><code>case-3-literature-triangulation.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-3-literature-triangulation.ipynb">notebook for case study 3 in the EpiGraphDB paper</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="#">paper</a> or the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-3-literature-triangulation.ipynb">case study notebook</a>.</p>
<div id="context" class="section level2">
<h2 class="hasAnchor">
<a href="#context" class="anchor"></a>Context</h2>
<p>The biomedical literature contains a wealth of information than far exceeds our capacity for systematic manual extraction. For this reason, there are many existing literature mining methods to extract the key concepts and content. Here we use data from <a href="https://skr3.nlm.nih.gov/SemMedDB/">SemMedDB</a>, a well established database that provides subject-predicate-object triples from all PubMed titles and abstracts. Using a subset of this data we created MELODI-presto (<a href="https://melodi-presto.mrcieu.ac.uk/" class="uri">https://melodi-presto.mrcieu.ac.uk/</a>), a method to assign triples to any given biomedical query via a PubMed search and some basic enrichment, and have applied this systematically to traits represented in EpiGraphDB. This allows us to identify overlapping terms connecting any set of GWAS traits, e.g. exposure and disease outcome. From here we can attempt to triangulate causal estimates, and conversely, check the mechanisms identified from the literature against the causal evidence.</p>
<p>This case study goes as follows:</p>
<ul>
<li>For an exposure trait we identify its causal evidence against other outcome traits, then link the outcomes to diseases</li>
<li>Then for an exposure-outcome pair we look for its literature evidence, and the mechanisms (as SemMed triples) identified from the literature evidence</li>
<li>Finally we look into detail the mechanisms specifically related to an overlapping term and visualise these mechanisms.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://magrittr.tidyverse.org">"magrittr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org">"dplyr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://purrr.tidyverse.org">"purrr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:magrittr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     set_names</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/tidyverse/glue">"glue"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'glue'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://igraph.org">"igraph"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'igraph'</span>
<span class="co">#&gt; The following objects are masked from 'package:purrr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     compose, simplify</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     as_data_frame, groups, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://mrcieu.github.io/epigraphdb-r/">"epigraphdb"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     EpiGraphDB v1.0 (API: https://api.epigraphdb.org)</span>
<span class="co">#&gt; </span></code></pre></div>
<p>Here we set the starting trait, which we will use to explore associated disease traits.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">STARTING_TRAIT</span> <span class="op">&lt;-</span> <span class="st">"Sleep duration"</span></code></pre></div>
</div>
<div id="get-traits-mr-association" class="section level2">
<h2 class="hasAnchor">
<a href="#get-traits-mr-association" class="anchor"></a>Get traits MR association</h2>
<p>Given an exposure trait, find all traits with causal evidence. This method searches the causal evidence data for cases where our exposure trait has a potential casual effect on an outcome trait.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_mr</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">trait</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    exposure_trait <span class="op">=</span> <span class="va">trait</span>,
    pval_threshold <span class="op">=</span> <span class="fl">1e-10</span>
  <span class="op">)</span>
  <span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="va">mr_df</span>
<span class="op">}</span>

<span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu">get_mr</span><span class="op">(</span><span class="va">STARTING_TRAIT</span><span class="op">)</span>
<span class="va">mr_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 1,178</span>
<span class="co">#&gt; Columns: 10</span>
<span class="co">#&gt; $ exposure.id    &lt;chr&gt; "ieu-a-1088", "ieu-a-1088", "ieu-a-1088", "ieu-a-1088",…</span>
<span class="co">#&gt; $ exposure.trait &lt;chr&gt; "Sleep duration", "Sleep duration", "Sleep duration", "…</span>
<span class="co">#&gt; $ outcome.id     &lt;chr&gt; "met-a-353", "met-a-307", "ieu-a-99", "ieu-a-59", "ieu-…</span>
<span class="co">#&gt; $ outcome.trait  &lt;chr&gt; "Cortisol", "Cholesterol", "Hip circumference", "Hip ci…</span>
<span class="co">#&gt; $ mr.b           &lt;dbl&gt; 0.059176931, -0.024282343, -0.288661815, 0.072874008, 0…</span>
<span class="co">#&gt; $ mr.se          &lt;dbl&gt; 1.385651e-03, 5.409778e-05, 5.928478e-03, 1.617032e-04,…</span>
<span class="co">#&gt; $ mr.pval        &lt;dbl&gt; 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0…</span>
<span class="co">#&gt; $ mr.method      &lt;chr&gt; "FE IVW", "FE IVW", "FE IVW", "FE IVW", "FE IVW", "FE I…</span>
<span class="co">#&gt; $ mr.selection   &lt;chr&gt; "DF", "DF", "DF", "DF", "DF", "DF", "DF", "DF", "DF", "…</span>
<span class="co">#&gt; $ mr.moescore    &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…</span></code></pre></div>
</div>
<div id="map-outcome-traits-to-disease" class="section level2">
<h2 class="hasAnchor">
<a href="#map-outcome-traits-to-disease" class="anchor"></a>Map outcome traits to disease</h2>
<p>For this example, we are interested in traits mapped to a disease node. To do this we utilise the mapping from GWAS trait to Disease via EFO term.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># NOTE: this block takes a bit of time, depending on the size of `mr_df` above.</span>
<span class="co">#       We will not run the block here in package building,</span>
<span class="co">#       but readers are encouraged to try it offline!</span>
<span class="va">trait_to_disease</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">trait</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/ontology/gwas-efo-disease"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>trait <span class="op">=</span> <span class="va">trait</span><span class="op">)</span>
  <span class="va">disease_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">disease_df</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="va">disease_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`disease.label`</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">}</span>
  <span class="va">res</span>
<span class="op">}</span>

<span class="va">outcome_disease_df</span> <span class="op">&lt;-</span> <span class="va">mr_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`outcome.trait`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>disease <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">`outcome.trait`</span>, <span class="va">trait_to_disease</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_lgl</a></span><span class="op">(</span><span class="va">`disease`</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">outcome_disease_df</span></code></pre></div>
</div>
<div id="take-one-example-to-look-for-literature-evidence" class="section level2">
<h2 class="hasAnchor">
<a href="#take-one-example-to-look-for-literature-evidence" class="anchor"></a>Take one example to look for literature evidence</h2>
<p>For the multiple <code>exposure -&gt; outcome</code> relationships as reported from the table above, here we look at the literature evidence for one pair in detail:</p>
<ul>
<li>Trait X: “Sleep duration” (<a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-1088/">ieu-a-1088</a>)</li>
<li>Trait Y: “Coronary heart disease” (<a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-6/">ieu-a-6</a>).</li>
</ul>
<p>The following looks for enriched triples of information (Subject-Predicate-Object) associated with our two traits. These have been derived via PubMed searches and corresponding <a href="https://skr3.nlm.nih.gov/SemMedDB/">SemMedDB</a> data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_gwas_pair_literature</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gwas_id</span>, <span class="va">assoc_gwas_id</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/literature/gwas/pairwise"</span>
  <span class="co"># NOTE in this example we blacklist to semmentic types</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    gwas_id <span class="op">=</span> <span class="va">gwas_id</span>,
    assoc_gwas_id <span class="op">=</span> <span class="va">assoc_gwas_id</span>,
    by_gwas_id <span class="op">=</span> <span class="cn">TRUE</span>,
    pval_threshold <span class="op">=</span> <span class="fl">1e-1</span>,
    semmantic_types <span class="op">=</span> <span class="st">"nusq"</span>,
    semmantic_types <span class="op">=</span> <span class="st">"dsyn"</span>,
    blacklist <span class="op">=</span> <span class="cn">TRUE</span>,
    limit <span class="op">=</span> <span class="fl">1000</span>
  <span class="op">)</span>
  <span class="va">lit_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="va">lit_df</span>
<span class="op">}</span>

<span class="va">GWAS_ID_X</span> <span class="op">&lt;-</span> <span class="st">"ieu-a-1088"</span>
<span class="va">GWAS_ID_Y</span> <span class="op">&lt;-</span> <span class="st">"ieu-a-6"</span>
<span class="va">lit_df</span> <span class="op">&lt;-</span> <span class="fu">get_gwas_pair_literature</span><span class="op">(</span><span class="va">GWAS_ID_X</span>, <span class="va">GWAS_ID_Y</span><span class="op">)</span>

<span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="va">lit_df</span><span class="op">)</span>
<span class="co">#&gt; Rows: 1,000</span>
<span class="co">#&gt; Columns: 20</span>
<span class="co">#&gt; $ gwas.id          &lt;chr&gt; "ieu-a-1088", "ieu-a-1088", "ieu-a-1088", "ieu-a-1088…</span>
<span class="co">#&gt; $ gwas.trait       &lt;chr&gt; "Sleep duration", "Sleep duration", "Sleep duration",…</span>
<span class="co">#&gt; $ gs1.pval         &lt;dbl&gt; 9.468989e-05, 2.478516e-06, 2.478516e-06, 1.854114e-0…</span>
<span class="co">#&gt; $ gs1.localCount   &lt;int&gt; 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,…</span>
<span class="co">#&gt; $ st1.name         &lt;chr&gt; "Sulpiride", "Glycine", "Glycine", "gamma-aminobutyri…</span>
<span class="co">#&gt; $ s1.id            &lt;chr&gt; "C0038803:INTERACTS_WITH:C0003596", "C0017890:STIMULA…</span>
<span class="co">#&gt; $ s1.subject_id    &lt;chr&gt; "C0038803", "C0017890", "C0017890", "C0016904", "C001…</span>
<span class="co">#&gt; $ s1.object_id     &lt;chr&gt; "C0003596", "C0033487", "C0033487", "C0033487", "C003…</span>
<span class="co">#&gt; $ s1.predicate     &lt;chr&gt; "INTERACTS_WITH", "STIMULATES", "STIMULATES", "STIMUL…</span>
<span class="co">#&gt; $ st.name          &lt;chr&gt; "Apomorphine", "propofol", "propofol", "propofol", "p…</span>
<span class="co">#&gt; $ st.type          &lt;list&gt; &lt;"orch", "phsu"&gt;, &lt;"orch", "phsu"&gt;, &lt;"orch", "phsu"&gt;…</span>
<span class="co">#&gt; $ s2.id            &lt;chr&gt; "C0003596:TREATS:C0242350", "C0033487:AUGMENTS:C01517…</span>
<span class="co">#&gt; $ s2.subject_id    &lt;chr&gt; "C0003596", "C0033487", "C0033487", "C0033487", "C003…</span>
<span class="co">#&gt; $ s2.object_id     &lt;chr&gt; "C0242350", "C0151744", "C0010054", "C0151744", "C001…</span>
<span class="co">#&gt; $ s2.predicate     &lt;chr&gt; "TREATS", "AUGMENTS", "TREATS", "AUGMENTS", "TREATS",…</span>
<span class="co">#&gt; $ gs2.pval         &lt;dbl&gt; 0.045878175, 0.002682225, 0.001627351, 0.002682225, 0…</span>
<span class="co">#&gt; $ gs2.localCount   &lt;int&gt; 3, 2, 2, 2, 2, 2, 3, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,…</span>
<span class="co">#&gt; $ st2.name         &lt;chr&gt; "Erectile dysfunction", "Myocardial Ischemia", "Coron…</span>
<span class="co">#&gt; $ assoc_gwas.id    &lt;chr&gt; "ieu-a-6", "ieu-a-6", "ieu-a-6", "ieu-a-6", "ieu-a-6"…</span>
<span class="co">#&gt; $ assoc_gwas.trait &lt;chr&gt; "Coronary heart disease", "Coronary heart disease", "…</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Predicate counts for SemMed triples for trait X</span>
<span class="va">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">`s1.predicate`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 2</span>
<span class="co">#&gt;   s1.predicate           n</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt;</span>
<span class="co">#&gt; 1 INTERACTS_WITH       719</span>
<span class="co">#&gt; 2 STIMULATES            96</span>
<span class="co">#&gt; 3 INHIBITS              93</span>
<span class="co">#&gt; 4 NEG_INTERACTS_WITH    51</span>
<span class="co">#&gt; 5 higher_than           23</span>
<span class="co">#&gt; 6 COEXISTS_WITH         15</span>
<span class="co">#&gt; 7 NEG_INHIBITS           3</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Predicate counts for SemMed triples for trait Y</span>
<span class="va">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">`s2.predicate`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 16 x 2</span>
<span class="co">#&gt;    s2.predicate            n</span>
<span class="co">#&gt;    &lt;chr&gt;               &lt;int&gt;</span>
<span class="co">#&gt;  1 TREATS                191</span>
<span class="co">#&gt;  2 STIMULATES            132</span>
<span class="co">#&gt;  3 CAUSES                120</span>
<span class="co">#&gt;  4 PREVENTS              117</span>
<span class="co">#&gt;  5 ASSOCIATED_WITH       103</span>
<span class="co">#&gt;  6 PREDISPOSES            83</span>
<span class="co">#&gt;  7 AFFECTS                71</span>
<span class="co">#&gt;  8 INTERACTS_WITH         62</span>
<span class="co">#&gt;  9 INHIBITS               47</span>
<span class="co">#&gt; 10 COEXISTS_WITH          34</span>
<span class="co">#&gt; 11 DISRUPTS               19</span>
<span class="co">#&gt; 12 NEG_PREVENTS           13</span>
<span class="co">#&gt; 13 higher_than             4</span>
<span class="co">#&gt; 14 AUGMENTS                2</span>
<span class="co">#&gt; 15 NEG_ASSOCIATED_WITH     1</span>
<span class="co">#&gt; 16 same_as                 1</span></code></pre></div>
<div id="filter-the-data-by-predicates" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-the-data-by-predicates" class="anchor"></a>Filter the data by predicates</h3>
<p>Sometimes it is preferable to filter the SemMedDB data, e.g. to remove less informative Predicates, such as COEXISTS_WITH and ASSOCIATED_WITH.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Filter out some predicates that are not informative</span>
<span class="va">pred_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"COEXISTS_WITH"</span>, <span class="st">"ASSOCIATED_WITH"</span><span class="op">)</span>
<span class="va">lit_df_filter</span> <span class="op">&lt;-</span> <span class="va">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span>
    <span class="op">!</span><span class="va">`s1.predicate`</span> <span class="op">%in%</span> <span class="va">pred_filter</span>,
    <span class="op">!</span><span class="va">`s2.predicate`</span> <span class="op">%in%</span> <span class="va">pred_filter</span>
  <span class="op">)</span>
<span class="va">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">`s1.predicate`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 5 x 2</span>
<span class="co">#&gt;   s1.predicate           n</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt;</span>
<span class="co">#&gt; 1 INTERACTS_WITH       649</span>
<span class="co">#&gt; 2 INHIBITS              87</span>
<span class="co">#&gt; 3 STIMULATES            66</span>
<span class="co">#&gt; 4 NEG_INTERACTS_WITH    26</span>
<span class="co">#&gt; 5 higher_than           22</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">`s2.predicate`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 14 x 2</span>
<span class="co">#&gt;    s2.predicate            n</span>
<span class="co">#&gt;    &lt;chr&gt;               &lt;int&gt;</span>
<span class="co">#&gt;  1 TREATS                180</span>
<span class="co">#&gt;  2 STIMULATES            132</span>
<span class="co">#&gt;  3 CAUSES                120</span>
<span class="co">#&gt;  4 PREVENTS              117</span>
<span class="co">#&gt;  5 PREDISPOSES            83</span>
<span class="co">#&gt;  6 AFFECTS                70</span>
<span class="co">#&gt;  7 INTERACTS_WITH         62</span>
<span class="co">#&gt;  8 INHIBITS               46</span>
<span class="co">#&gt;  9 DISRUPTS               19</span>
<span class="co">#&gt; 10 NEG_PREVENTS           13</span>
<span class="co">#&gt; 11 higher_than             4</span>
<span class="co">#&gt; 12 AUGMENTS                2</span>
<span class="co">#&gt; 13 NEG_ASSOCIATED_WITH     1</span>
<span class="co">#&gt; 14 same_as                 1</span></code></pre></div>
</div>
</div>
<div id="literature-results" class="section level2">
<h2 class="hasAnchor">
<a href="#literature-results" class="anchor"></a>Literature results</h2>
<p>If we explore the full table in <code>lit_df_filter</code>, we can see lots of links between the two traits, pinned on specific overlapping terms.</p>
<p>We can summarise the <a href="https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml">SemMedDB semantic type</a> and number of overlapping terms:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lit_counts</span> <span class="op">&lt;-</span> <span class="va">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">`st.type`</span>, <span class="va">`st.name`</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">`st.type`</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">`n`</span><span class="op">)</span><span class="op">)</span>
<span class="va">lit_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 33 x 3</span>
<span class="co">#&gt;    st.type   st.name                                  n</span>
<span class="co">#&gt;    &lt;list&gt;    &lt;chr&gt;                                &lt;int&gt;</span>
<span class="co">#&gt;  1 &lt;chr [2]&gt; ethanol                                571</span>
<span class="co">#&gt;  2 &lt;chr [2]&gt; Oral anticoagulants                     24</span>
<span class="co">#&gt;  3 &lt;chr [2]&gt; Metoprolol                              20</span>
<span class="co">#&gt;  4 &lt;chr [2]&gt; morphine                                 7</span>
<span class="co">#&gt;  5 &lt;chr [2]&gt; Benzodiazepines                          4</span>
<span class="co">#&gt;  6 &lt;chr [2]&gt; propofol                                 4</span>
<span class="co">#&gt;  7 &lt;chr [2]&gt; Diazepam                                 2</span>
<span class="co">#&gt;  8 &lt;chr [2]&gt; Midazolam                                2</span>
<span class="co">#&gt;  9 &lt;chr [2]&gt; Thiopental                               2</span>
<span class="co">#&gt; 10 &lt;chr [2]&gt; Apomorphine                              1</span>
<span class="co">#&gt; 11 &lt;chr [2]&gt; gamma hydroxybutyrate                    1</span>
<span class="co">#&gt; 12 &lt;chr [2]&gt; Luteolin                                 1</span>
<span class="co">#&gt; 13 &lt;chr [2]&gt; Neostigmine                              1</span>
<span class="co">#&gt; 14 &lt;chr [2]&gt; Glycoproteins                            3</span>
<span class="co">#&gt; 15 &lt;chr [2]&gt; glutathione                              1</span>
<span class="co">#&gt; 16 &lt;chr [3]&gt; Cytochrome P450                          6</span>
<span class="co">#&gt; 17 &lt;chr [3]&gt; mitogen-activated protein kinase p38     4</span>
<span class="co">#&gt; 18 &lt;chr [3]&gt; AHSA1                                    2</span>
<span class="co">#&gt; 19 &lt;chr [3]&gt; AIMP2                                    2</span>
<span class="co">#&gt; 20 &lt;chr [3]&gt; CRK                                      2</span>
<span class="co">#&gt; 21 &lt;chr [3]&gt; GRAP2                                    2</span>
<span class="co">#&gt; 22 &lt;chr [3]&gt; MAPK1                                    2</span>
<span class="co">#&gt; 23 &lt;chr [3]&gt; MAPK14                                   2</span>
<span class="co">#&gt; 24 &lt;chr [3]&gt; POLDIP2                                  2</span>
<span class="co">#&gt; 25 &lt;chr [3]&gt; Protein Kinase C                         2</span>
<span class="co">#&gt; 26 &lt;chr [1]&gt; agonists                                31</span>
<span class="co">#&gt; 27 &lt;chr [1]&gt; Antidepressive Agents                   12</span>
<span class="co">#&gt; 28 &lt;chr [1]&gt; Anesthetics                              8</span>
<span class="co">#&gt; 29 &lt;chr [2]&gt; melatonin                               88</span>
<span class="co">#&gt; 30 &lt;chr [2]&gt; Medroxyprogesterone 17-Acetate           5</span>
<span class="co">#&gt; # … with 3 more rows</span></code></pre></div>
<p>Note, the SemMedDB semantic types have been pre-filtered to only include a subset of possibilities.</p>
<p>Further examples of these term IDs and descriptions can be found here - <a href="https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml">https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml</a></p>
<div id="plotting-the-overlapping-term-frequencies" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-the-overlapping-term-frequencies" class="anchor"></a>Plotting the overlapping term frequencies</h3>
<p>We can also visualise the above table as a bar chart. In this case we will remove Ethanol as it is an outlier.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lit_counts</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">n</span> <span class="op">&lt;</span> <span class="fl">100</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="op">{</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">`st.name`</span>, y <span class="op">=</span> <span class="va">`n`</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label <span class="op">=</span> <span class="va">`n`</span><span class="op">)</span>,
        position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span>,
        hjust <span class="op">=</span> <span class="fl">0</span>
      <span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span>
  <span class="op">}</span></code></pre></div>
<p><img src="case-3-literature-triangulation_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div id="look-in-detail-at-one-overlapping-term" class="section level3">
<h3 class="hasAnchor">
<a href="#look-in-detail-at-one-overlapping-term" class="anchor"></a>Look in detail at one overlapping term</h3>
<p>Here we look at cases where <code>leptin</code> is the central overlapping term.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">focus_term</span> <span class="op">&lt;-</span> <span class="st">"leptin"</span>
<span class="va">lit_detail</span> <span class="op">&lt;-</span> <span class="va">lit_df_filter</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">`st.name`</span> <span class="op">==</span> <span class="va">focus_term</span><span class="op">)</span>
<span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 20</span>
<span class="co">#&gt;   gwas.id  gwas.trait   gs1.pval gs1.localCount st1.name s1.id     s1.subject_id</span>
<span class="co">#&gt;   &lt;chr&gt;    &lt;chr&gt;           &lt;dbl&gt;          &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;        </span>
<span class="co">#&gt; 1 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; 2 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; 3 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; 4 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; 5 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; 6 ieu-a-1… Sleep dura…   2.48e-6              2 Cortico… C0010124… C0010124     </span>
<span class="co">#&gt; # … with 13 more variables: s1.object_id &lt;chr&gt;, s1.predicate &lt;chr&gt;,</span>
<span class="co">#&gt; #   st.name &lt;chr&gt;, st.type &lt;list&gt;, s2.id &lt;chr&gt;, s2.subject_id &lt;chr&gt;,</span>
<span class="co">#&gt; #   s2.object_id &lt;chr&gt;, s2.predicate &lt;chr&gt;, gs2.pval &lt;dbl&gt;,</span>
<span class="co">#&gt; #   gs2.localCount &lt;int&gt;, st2.name &lt;chr&gt;, assoc_gwas.id &lt;chr&gt;,</span>
<span class="co">#&gt; #   assoc_gwas.trait &lt;chr&gt;</span></code></pre></div>
<p>We can create a network diagram to visualise these relationships.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lit_detail</span> <span class="op">&lt;-</span> <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/vars.html">vars</a></span><span class="op">(</span><span class="va">`gwas.trait`</span>, <span class="va">`assoc_gwas.trait`</span><span class="op">)</span>, <span class="fu">stringr</span><span class="fu">::</span><span class="va"><a href="https://stringr.tidyverse.org/reference/case.html">str_to_upper</a></span><span class="op">)</span>

<span class="co"># add node types: 1 - selected GWAS, 2 - traits from literature, 3 - current focus term connecting 1 and 2</span>
<span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`gwas.trait`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>node_type <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`assoc_gwas.trait`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>node_type <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`st1.name`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>node_type <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`st2.name`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>node_type <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`st.name`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>node_type <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,
<span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">nodes</span>
<span class="co">#&gt; # A tibble: 20 x 2</span>
<span class="co">#&gt;    node                        node_type</span>
<span class="co">#&gt;    &lt;chr&gt;                           &lt;dbl&gt;</span>
<span class="co">#&gt;  1 SLEEP DURATION                      1</span>
<span class="co">#&gt;  2 CORONARY HEART DISEASE              1</span>
<span class="co">#&gt;  3 Corticosterone                      2</span>
<span class="co">#&gt;  4 Sleep Apnea, Obstructive            2</span>
<span class="co">#&gt;  5 Phosphotransferases                 2</span>
<span class="co">#&gt;  6 JAK2                                2</span>
<span class="co">#&gt;  7 Janus kinase 2                      2</span>
<span class="co">#&gt;  8 Insulin                             2</span>
<span class="co">#&gt;  9 Adiponectin                         2</span>
<span class="co">#&gt; 10 Coronary heart disease              2</span>
<span class="co">#&gt; 11 Proteome                            2</span>
<span class="co">#&gt; 12 Endothelial dysfunction             2</span>
<span class="co">#&gt; 13 Cardiovascular Diseases             2</span>
<span class="co">#&gt; 14 Congestive heart failure            2</span>
<span class="co">#&gt; 15 Acute Coronary Syndrome             2</span>
<span class="co">#&gt; 16 Coronary Arteriosclerosis           2</span>
<span class="co">#&gt; 17 Stable angina                       2</span>
<span class="co">#&gt; 18 Metabolic syndrome                  2</span>
<span class="co">#&gt; 19 Acute myocardial infarction         2</span>
<span class="co">#&gt; 20 leptin                              3</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>
  <span class="co"># exposure -&gt; s1 subject</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`gwas.trait`</span>, assoc_node <span class="op">=</span> <span class="va">`st1.name`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="co"># s2 object -&gt; outcome</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">`st2.name`</span>, assoc_node <span class="op">=</span> <span class="va">`assoc_gwas.trait`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="co"># s1 subject - s1 predicate -&gt; s1 object</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>
      node <span class="op">=</span> <span class="va">`st1.name`</span>, assoc_node <span class="op">=</span> <span class="va">`st.name`</span>,
      label <span class="op">=</span> <span class="va">`s1.predicate`</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="co"># s2 subject - s2 predicate -&gt; s2 object</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>
      node <span class="op">=</span> <span class="va">`st.name`</span>, assoc_node <span class="op">=</span> <span class="va">`st2.name`</span>,
      label <span class="op">=</span> <span class="va">`s2.predicate`</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">edges</span>
<span class="co">#&gt; # A tibble: 40 x 3</span>
<span class="co">#&gt;    node                     assoc_node             label</span>
<span class="co">#&gt;    &lt;chr&gt;                    &lt;chr&gt;                  &lt;chr&gt;</span>
<span class="co">#&gt;  1 SLEEP DURATION           Corticosterone         &lt;NA&gt; </span>
<span class="co">#&gt;  2 Sleep Apnea, Obstructive CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  3 Phosphotransferases      CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  4 JAK2                     CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  5 Janus kinase 2           CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  6 Insulin                  CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  7 Adiponectin              CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  8 Coronary heart disease   CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt;  9 Proteome                 CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt; 10 Endothelial dysfunction  CORONARY HEART DISEASE &lt;NA&gt; </span>
<span class="co">#&gt; # … with 30 more rows</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_network</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">edges</span>, <span class="va">nodes</span><span class="op">)</span> <span class="op">{</span>

  <span class="va">graph</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span><span class="op">(</span><span class="va">edges</span>, directed <span class="op">=</span> <span class="cn">TRUE</span>, vertices <span class="op">=</span> <span class="va">nodes</span><span class="op">)</span>
  <span class="va">graph</span><span class="op">$</span><span class="va">layout</span> <span class="op">&lt;-</span> <span class="va">layout_with_kk</span>

  <span class="co"># generate colors based on node type</span>
  <span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"tomato"</span>, <span class="st">"lightblue"</span>, <span class="st">"gold"</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span><span class="op">$</span><span class="va">color</span> <span class="op">&lt;-</span> <span class="va">colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span><span class="op">$</span><span class="va">node_type</span><span class="op">]</span>

  <span class="co"># Configure canvas</span>
  <span class="va">default_mar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span><span class="st">"mar"</span><span class="op">)</span>
  <span class="va">new_mar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="va">new_mar</span><span class="op">)</span>

  <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/plot.igraph.html">plot.igraph</a></span><span class="op">(</span>
    <span class="va">graph</span>,
    vertex.size <span class="op">=</span> <span class="fl">13</span>,
    vertex.label.color <span class="op">=</span> <span class="st">"black"</span>,
    vertex.label.family <span class="op">=</span> <span class="st">"Helvetica"</span>,
    vertex.label.cex <span class="op">=</span> <span class="fl">0.8</span>,
    edge.arrow.size <span class="op">=</span> <span class="fl">0.4</span>,
    edge.label.color <span class="op">=</span> <span class="st">"black"</span>,
    edge.label.family <span class="op">=</span> <span class="st">"Helvetica"</span>,
    edge.label.cex <span class="op">=</span> <span class="fl">0.5</span>
  <span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="va">default_mar</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">plot_network</span><span class="op">(</span><span class="va">edges</span>, <span class="va">nodes</span><span class="op">)</span></code></pre></div>
<p><img src="case-3-literature-triangulation_files/figure-html/unnamed-chunk-16-1.png" width="2187.5"></p>
</div>
</div>
<div id="checking-the-source-literature" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-the-source-literature" class="anchor"></a>Checking the source literature</h2>
<p>We can refer back to the articles to check the text that was used to derive the SemMedDB data. This is important due to the imperfect nature of the SemRep annotation process (<a href="https://semrep.nlm.nih.gov/" class="uri">https://semrep.nlm.nih.gov/</a>).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_literature</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gwas_id</span>, <span class="va">semmed_triple_id</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/literature/gwas"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    gwas_id <span class="op">=</span> <span class="va">gwas_id</span>,
    semmed_triple_id <span class="op">=</span> <span class="va">semmed_triple_id</span>,
    by_gwas_id <span class="op">=</span> <span class="cn">TRUE</span>,
    pval_threshold <span class="op">=</span> <span class="fl">1e-1</span>
  <span class="op">)</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="va">df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`triple.id`</span>, <span class="va">`triple.name`</span>, <span class="va">`lit.id`</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">pub_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span>
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>gwas_id <span class="op">=</span> <span class="va">`gwas.id`</span>, semmed_triple_id <span class="op">=</span> <span class="va">`s1.id`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>gwas_id <span class="op">=</span> <span class="va">`assoc_gwas.id`</span>, semmed_triple_id <span class="op">=</span> <span class="va">`s2.id`</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/transpose.html">transpose</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">get_literature</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">gwas_id</span>, <span class="va">x</span><span class="op">$</span><span class="va">semmed_triple_id</span><span class="op">)</span><span class="op">)</span>
<span class="va">pub_df</span>
<span class="co">#&gt; # A tibble: 124 x 3</span>
<span class="co">#&gt;    triple.id                       triple.name                           lit.id </span>
<span class="co">#&gt;    &lt;chr&gt;                           &lt;chr&gt;                                 &lt;chr&gt;  </span>
<span class="co">#&gt;  1 C0010124:NEG_INTERACTS_WITH:C0… Corticosterone NEG_INTERACTS_WITH Le… 200515…</span>
<span class="co">#&gt;  2 C0010124:NEG_INTERACTS_WITH:C0… Corticosterone NEG_INTERACTS_WITH Le… 154755…</span>
<span class="co">#&gt;  3 C0299583:TREATS:C0520679        Leptin|LEP TREATS Sleep Apnea, Obstr… 104496…</span>
<span class="co">#&gt;  4 C0299583:TREATS:C0520679        Leptin|LEP TREATS Sleep Apnea, Obstr… 104496…</span>
<span class="co">#&gt;  5 C0299583:STIMULATES:C0031727    Leptin STIMULATES Phosphotransferases 159272…</span>
<span class="co">#&gt;  6 C0299583:STIMULATES:3717        Leptin|LEP STIMULATES Janus kinase 2… 159272…</span>
<span class="co">#&gt;  7 C0299583:STIMULATES:C0169661    Leptin|LEP STIMULATES Janus kinase 2… 159272…</span>
<span class="co">#&gt;  8 C0299583:INTERACTS_WITH:C00216… Leptin INTERACTS_WITH Insulin         156480…</span>
<span class="co">#&gt;  9 C0299583:INTERACTS_WITH:C00216… Leptin INTERACTS_WITH Insulin         127240…</span>
<span class="co">#&gt; 10 C0299583:STIMULATES:C0389071    Leptin STIMULATES Adiponectin         158112…</span>
<span class="co">#&gt; # … with 114 more rows</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.4 (2021-02-15)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2.2 igraph_1.2.6     ggplot2_3.3.3    glue_1.4.2      </span>
<span class="co">#&gt; [5] purrr_0.3.4      dplyr_1.0.5      magrittr_2.0.1  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] highr_0.8         compiler_4.0.4    pillar_1.5.1      tools_4.0.4      </span>
<span class="co">#&gt;  [5] digest_0.6.27     jsonlite_1.7.2    evaluate_0.14     memoise_2.0.0    </span>
<span class="co">#&gt;  [9] lifecycle_1.0.0   tibble_3.1.0      gtable_0.3.0      pkgconfig_2.0.3  </span>
<span class="co">#&gt; [13] rlang_0.4.10      rstudioapi_0.13   cli_2.3.1         curl_4.3         </span>
<span class="co">#&gt; [17] yaml_2.2.1        pkgdown_1.6.1     xfun_0.22         fastmap_1.1.0    </span>
<span class="co">#&gt; [21] httr_1.4.2        withr_2.4.1       stringr_1.4.0     knitr_1.31       </span>
<span class="co">#&gt; [25] generics_0.1.0    desc_1.3.0        fs_1.5.0          vctrs_0.3.6      </span>
<span class="co">#&gt; [29] systemfonts_1.0.1 rprojroot_2.0.2   grid_4.0.4        tidyselect_1.1.0 </span>
<span class="co">#&gt; [33] R6_2.5.0          textshaping_0.3.3 fansi_0.4.2       rmarkdown_2.7    </span>
<span class="co">#&gt; [37] farver_2.1.0      ps_1.6.0          scales_1.1.1      htmltools_0.5.1.1</span>
<span class="co">#&gt; [41] ellipsis_0.3.1    assertthat_0.2.1  colorspace_2.0-0  labeling_0.4.2   </span>
<span class="co">#&gt; [45] ragg_1.1.2        utf8_1.2.1        stringi_1.5.3     munsell_0.5.0    </span>
<span class="co">#&gt; [49] cachem_1.0.4      crayon_1.4.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
