<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations">
<meta property="og:description" content="epigraphdb">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</h1>
            
      
      
      <div class="hidden name"><code>case-1-pleiotropy.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-1-pleiotropy.ipynb">notebook for case study 1 in the EpiGraphDB paper</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="#">paper</a> or the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-1-pleiotropy.ipynb">case study notebook</a>.</p>
<div id="context" class="section level2">
<h2 class="hasAnchor">
<a href="#context" class="anchor"></a>Context</h2>
<p>Mendelian randomization (MR), a technique to evaluate the causal role of modifiable exposures on a health outcome using a set of genetic instruments, tacitly assumes that a genetic variant (e.g. SNP) is only related to an outcome of interest through an exposure (i.e. the “exclusion restriction criterion”). Horizontal pleiotropy, where a SNP is associated with multiple phenotypes independently of the exposure of interest, potentially violates this assumption delivering misleading conclusions. In contrast, vertical pleiotropy, where a SNP is associated with multiple phenotypes on the same biological pathway, does not violate this assumption.</p>
<p>Here, we use external evidence on biological pathways and protein-protein interactions to assess the pleiotropic profile for a genetic variant based on the genes (and proteins) with which that variant is associated. In a graph representation, we will show that:</p>
<ul>
<li>We have evidence of potential <strong>vertical</strong> pleiotropy when the proteins associated with a variant belong to one connected group,</li>
<li>In contrast, if the proteins associated with a single variant are isolated from each other, this suggests evidence of <strong>horizontal</strong> pleiotropy.</li>
</ul>
<p>This case study goes as follows:</p>
<ul>
<li>For a list of genes that we wish to assess their pleiotropic profiles (against a variant of interests), we retrive from EpiGraphDB their mapped protein products.</li>
<li>Then respecively from the evidence on shared pathway network and protein-protein interaction network, we can extract the proteins groups formed in these networks as connected communities to assess the pleiotropic profiles of the genes.</li>
</ul>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"magrittr"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"dplyr"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"purrr"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:magrittr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     set_names</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"glue"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'glue'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"igraph"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'igraph'</span>
<span class="co">#&gt; The following objects are masked from 'package:purrr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     compose, simplify</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     as_data_frame, groups, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"epigraphdb"</span>)
<span class="co">#&gt;   EpiGraphDB v0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   Web API: http://api.epigraphdb.org</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   To turn off this message, use</span>
<span class="co">#&gt;   suppressPackageStartupMessages({library("epigraphdb")})</span></pre></body></html></div>
<p>Here we configure the parameters used in the case study example. In this example we will look at the variant <strong>rs12720356</strong>, a SNP located in chromosome 19 that has been associated with Crohn’s disease and psoriasis.**</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">SNP</span> <span class="kw">&lt;-</span> <span class="st">"rs12720356"</span>

<span class="no">GENELIST</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ZGLP1"</span>, <span class="st">"FDX1L"</span>, <span class="st">"MRPL4"</span>, <span class="st">"ICAM5"</span>, <span class="st">"TYK2"</span>, <span class="st">"GRAP2"</span>, <span class="st">"KRI1"</span>, <span class="st">"TMED1"</span>, <span class="st">"ICAM1"</span>)

<span class="no">PPI_N_INTERMEDIATE_PROTEINS</span> <span class="kw">&lt;-</span> <span class="fl">1</span></pre></body></html></div>
</div>
<div id="mapping-genes-to-proteins" class="section level2">
<h2 class="hasAnchor">
<a href="#mapping-genes-to-proteins" class="anchor"></a>Mapping genes to proteins</h2>
<p>The first step of the analysis is to map each of these genes to their protein product.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">get_gene_protein</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">genelist</span>) {
  <span class="no">endpoint</span> <span class="kw">&lt;-</span> <span class="st">"/mappings/gene-to-protein"</span>
  <span class="no">params</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    <span class="kw">gene_name_list</span> <span class="kw">=</span> <span class="no">genelist</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span>()
  )
  <span class="no">r</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(
    <span class="kw">route</span> <span class="kw">=</span> <span class="no">endpoint</span>,
    <span class="kw">params</span> <span class="kw">=</span> <span class="no">params</span>,
    <span class="kw">mode</span> <span class="kw">=</span> <span class="st">"table"</span>,
    <span class="kw">method</span> <span class="kw">=</span> <span class="st">"POST"</span>
  )
  <span class="no">protein_df</span> <span class="kw">&lt;-</span> <span class="no">r</span>
  <span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">protein_df</span>) <span class="kw">&gt;</span> <span class="fl">0</span>) {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">protein_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">protein_name</span> <span class="kw">=</span> <span class="no">`gene.name`</span>, <span class="kw">uniprot_id</span> <span class="kw">=</span> <span class="no">`protein.uniprot_id`</span>)
  } <span class="kw">else</span> {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tibble</a></span>() <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/magrittr/man/aliases.html">set_names</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"protein_name"</span>, <span class="st">"uniprot_id"</span>))
  }

  <span class="no">res_df</span>
}

<span class="no">gene_protein_df</span> <span class="kw">&lt;-</span> <span class="fu">get_gene_protein</span>(<span class="kw">genelist</span> <span class="kw">=</span> <span class="no">GENELIST</span>)
<span class="no">gene_protein_df</span>
<span class="co">#&gt; # A tibble: 9 x 2</span>
<span class="co">#&gt;   protein_name uniprot_id</span>
<span class="co">#&gt;   &lt;chr&gt;        &lt;chr&gt;     </span>
<span class="co">#&gt; 1 TMED1        Q13445    </span>
<span class="co">#&gt; 2 MRPL4        Q9BYD3    </span>
<span class="co">#&gt; 3 TYK2         P29597    </span>
<span class="co">#&gt; 4 KRI1         Q8N9T8    </span>
<span class="co">#&gt; 5 FDX1L        Q6P4F2    </span>
<span class="co">#&gt; 6 GRAP2        O75791    </span>
<span class="co">#&gt; 7 ICAM1        P05362    </span>
<span class="co">#&gt; 8 ICAM5        Q9UMF0    </span>
<span class="co">#&gt; 9 ZGLP1        P0C6A0</span></pre></body></html></div>
</div>
<div id="identifying-the-involvement-in-the-same-biological-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#identifying-the-involvement-in-the-same-biological-pathways" class="anchor"></a>Identifying the involvement in the same biological pathways</h2>
<p>For each protein we retrieve the pathways they are involved in.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">get_protein_pathway</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">gene_protein_df</span>) {
  <span class="no">endpoint</span> <span class="kw">&lt;-</span> <span class="st">"/protein/in-pathway"</span>
  <span class="no">params</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    <span class="kw">uniprot_id_list</span> <span class="kw">=</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">`uniprot_id`</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span>()
  )
  <span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(<span class="kw">route</span> <span class="kw">=</span> <span class="no">endpoint</span>, <span class="kw">params</span> <span class="kw">=</span> <span class="no">params</span>, <span class="kw">mode</span> <span class="kw">=</span> <span class="st">"table"</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"POST"</span>)

  <span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">df</span>) <span class="kw">&gt;</span> <span class="fl">0</span>) {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">`uniprot_id`</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span>(<span class="no">df</span>, <span class="kw">by</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"uniprot_id"</span>))
  } <span class="kw">else</span> {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">`uniprot_id`</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">pathway_count</span> <span class="kw">=</span> <span class="fl">NA_integer_</span>, <span class="kw">pathway_reactome_id</span> <span class="kw">=</span> <span class="fl">NA_character_</span>)
  }
  <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">res_df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(
      <span class="kw">pathway_count</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">pathway_count</span>), <span class="fl">0L</span>, <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span>(<span class="no">pathway_count</span>)),
      <span class="kw">pathway_reactome_id</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">pathway_reactome_id</span>), <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(), <span class="no">pathway_reactome_id</span>)
    )

  <span class="no">res_df</span>
}
<span class="no">pathway_df</span> <span class="kw">&lt;-</span> <span class="fu">get_protein_pathway</span>(<span class="kw">gene_protein_df</span> <span class="kw">=</span> <span class="no">gene_protein_df</span>)
<span class="no">pathway_df</span>
<span class="co">#&gt; # A tibble: 9 x 3</span>
<span class="co">#&gt;   uniprot_id pathway_count pathway_reactome_id</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt; &lt;list&gt;             </span>
<span class="co">#&gt; 1 Q13445                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 2 Q9BYD3                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 3 P29597                 3 &lt;chr [3]&gt;          </span>
<span class="co">#&gt; 4 Q8N9T8                 0 &lt;NULL&gt;             </span>
<span class="co">#&gt; 5 Q6P4F2                 1 &lt;chr [1]&gt;          </span>
<span class="co">#&gt; 6 O75791                 3 &lt;chr [3]&gt;          </span>
<span class="co">#&gt; 7 P05362                 5 &lt;chr [5]&gt;          </span>
<span class="co">#&gt; 8 Q9UMF0                 2 &lt;chr [2]&gt;          </span>
<span class="co">#&gt; 9 P0C6A0                 0 &lt;NULL&gt;</span></pre></body></html></div>
<p>Now for each pair of proteins we match the pathways they have in common.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">get_shared_pathway</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">pathway_df</span>) {
  <span class="co"># For the protein-pathway data</span>
  <span class="co"># Get protein-protein permutations where they share pathways</span>

  <span class="no">per_permutation</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">pathway_df</span>, <span class="no">permutation</span>) {
    <span class="no">df</span> <span class="kw">&lt;-</span> <span class="no">pathway_df</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">uniprot_id</span> <span class="kw">%in%</span> <span class="no">permutation</span>)
    <span class="no">primary_pathway</span> <span class="kw">&lt;-</span> <span class="no">pathway_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">uniprot_id</span> <span class="kw">==</span> <span class="no">permutation</span>[<span class="fl">1</span>]) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">pathway_reactome_id</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()
    <span class="no">assoc_pathway</span> <span class="kw">&lt;-</span> <span class="no">pathway_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">uniprot_id</span> <span class="kw">==</span> <span class="no">permutation</span>[<span class="fl">2</span>]) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">pathway_reactome_id</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>()
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">intersect</a></span>(<span class="no">primary_pathway</span>, <span class="no">assoc_pathway</span>)
  }

  <span class="no">pairwise_permutations</span> <span class="kw">&lt;-</span> <span class="no">pathway_df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">`uniprot_id`</span>) <span class="kw">%&gt;%</span>
    <span class="kw pkg">gtools</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gtools/man/combinations.html">permutations</a></span>(<span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">.</span>), <span class="kw">r</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">v</span> <span class="kw">=</span> <span class="no">.</span>)
  <span class="no">shared_pathway_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tibble</a></span>(
    <span class="kw">protein</span> <span class="kw">=</span> <span class="no">pairwise_permutations</span>[, <span class="fl">1</span>],
    <span class="kw">assoc_protein</span> <span class="kw">=</span> <span class="no">pairwise_permutations</span>[, <span class="fl">2</span>]
  ) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(
      <span class="kw">shared_pathway</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span>(<span class="no">`protein`</span>, <span class="no">`assoc_protein`</span>, <span class="kw">function</span>(<span class="no">x</span>, <span class="no">y</span>) {
        <span class="fu">per_permutation</span>(<span class="kw">pathway_df</span> <span class="kw">=</span> <span class="no">pathway_df</span>, <span class="kw">permutation</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">x</span>, <span class="no">y</span>))
      }),
      <span class="kw">combination</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_chr</a></span>(<span class="no">`protein`</span>, <span class="no">`assoc_protein`</span>, <span class="kw">function</span>(<span class="no">x</span>, <span class="no">y</span>) {
        <span class="no">comb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">x</span>, <span class="no">y</span>))
        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="no">comb</span>, <span class="kw">collapse</span> <span class="kw">=</span> <span class="st">","</span>)
      }),
      <span class="kw">count</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_int</a></span>(<span class="no">`shared_pathway`</span>, <span class="kw">function</span>(<span class="no">x</span>) <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span>(<span class="no">x</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>()),
      <span class="kw">connected</span> <span class="kw">=</span> <span class="no">count</span> <span class="kw">&gt;</span> <span class="fl">0</span>
    )

  <span class="no">shared_pathway_df</span>
}
<span class="no">shared_pathway_df</span> <span class="kw">&lt;-</span> <span class="fu">get_shared_pathway</span>(<span class="no">pathway_df</span>)
<span class="no">n_pairs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">shared_pathway_df</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">count</span> <span class="kw">&gt;</span> <span class="fl">0</span>))
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">glue</span><span class="kw ns">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span>(<span class="st">"Num. shared_pathway pairs: {n_pairs}"</span>))
<span class="co">#&gt; Num. shared_pathway pairs: 6</span>
<span class="no">shared_pathway_df</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="no">count</span>))
<span class="co">#&gt; # A tibble: 72 x 6</span>
<span class="co">#&gt;    protein assoc_protein shared_pathway combination   count connected</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;list&gt;         &lt;chr&gt;         &lt;int&gt; &lt;lgl&gt;    </span>
<span class="co">#&gt;  1 P05362  Q9UMF0        &lt;chr [2]&gt;      P05362,Q9UMF0     2 TRUE     </span>
<span class="co">#&gt;  2 Q9UMF0  P05362        &lt;chr [2]&gt;      P05362,Q9UMF0     2 TRUE     </span>
<span class="co">#&gt;  3 P05362  P29597        &lt;chr [1]&gt;      P05362,P29597     1 TRUE     </span>
<span class="co">#&gt;  4 P29597  P05362        &lt;chr [1]&gt;      P05362,P29597     1 TRUE     </span>
<span class="co">#&gt;  5 O75791  P05362        &lt;chr [0]&gt;      O75791,P05362     0 FALSE    </span>
<span class="co">#&gt;  6 O75791  P0C6A0        &lt;NULL&gt;         O75791,P0C6A0     0 FALSE    </span>
<span class="co">#&gt;  7 O75791  P29597        &lt;chr [0]&gt;      O75791,P29597     0 FALSE    </span>
<span class="co">#&gt;  8 O75791  Q13445        &lt;NULL&gt;         O75791,Q13445     0 FALSE    </span>
<span class="co">#&gt;  9 O75791  Q6P4F2        &lt;chr [0]&gt;      O75791,Q6P4F2     0 FALSE    </span>
<span class="co">#&gt; 10 O75791  Q8N9T8        &lt;NULL&gt;         O75791,Q8N9T8     0 FALSE    </span>
<span class="co">#&gt; # … with 62 more rows</span></pre></body></html></div>
<p>We can further query EpiGraphDB regarding the detailed pathway information using <code>GET /meta/nodes/Pathway/search</code>.</p>
<pre><code>get_pathway_info = function(reactome_id) {
    endpoint = "/meta/nodes/Pathway/search"
    params = list(id=reactome_id)
    df = query_epigraphdb(route=endpoint, params=params, mode="table")
}

pathway = shared_pathway_df %&gt;% pull(shared_pathway) %&gt;%
    unlist() %&gt;% unique()

pathway_info = pathway %&gt;% map_df(get_pathway_info)
pathway_info %&gt;% print()</code></pre>
<p>In order to extract protein groups from the shared pathways, the last step for this query is to convert the shared pathway data into a graph where</p>
<ul>
<li>Proteins are represented as nodes.</li>
<li>And shared pathways are represented as edges.</li>
<li>Then the protein groups (via shared pathways) are the connected communities in the graph.</li>
</ul>
<p>We then count the number of nodes in each connected community and plot the graph.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">protein_df_to_graph</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">df</span>) {
  <span class="no">df_connected</span> <span class="kw">&lt;-</span> <span class="no">df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">connected</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(<span class="no">`combination`</span>, <span class="kw">.keep_all</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
  <span class="no">nodes</span> <span class="kw">&lt;-</span> <span class="no">df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">protein</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>()
  <span class="no">graph</span> <span class="kw">&lt;-</span> <span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span>(
    <span class="no">df_connected</span>,
    <span class="kw">directed</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">vertices</span> <span class="kw">=</span> <span class="no">nodes</span>
  )
  <span class="no">graph</span>$<span class="no">layout</span> <span class="kw">&lt;-</span> <span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/igraph/man/layout_with_kk.html">layout_with_kk</a></span>
  <span class="no">graph</span>
}

<span class="no">graph_to_protein_groups</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">graph</span>) {
  <span class="no">graph</span> <span class="kw">%&gt;%</span>
    <span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/components.html">components</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/groups.html">groups</a></span>() <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tibble</a></span>(<span class="kw">group_member</span> <span class="kw">=</span> <span class="no">.</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">group_size</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_int</a></span>(<span class="no">`group_member`</span>, <span class="no">length</span>)) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="no">group_size</span>))
}

<span class="no">pathway_protein_graph</span> <span class="kw">&lt;-</span> <span class="no">shared_pathway_df</span> <span class="kw">%&gt;%</span> <span class="fu">protein_df_to_graph</span>()
<span class="no">pathway_protein_groups</span> <span class="kw">&lt;-</span> <span class="no">pathway_protein_graph</span> <span class="kw">%&gt;%</span> <span class="fu">graph_to_protein_groups</span>()
<span class="no">pathway_protein_groups</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>()
<span class="co">#&gt; tibble [7 × 2] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;  $ group_member:List of 7</span>
<span class="co">#&gt;   ..$ 2: chr [1:3] "P05362" "P29597" "Q9UMF0"</span>
<span class="co">#&gt;   ..$ 1: chr "O75791"</span>
<span class="co">#&gt;   ..$ 3: chr "P0C6A0"</span>
<span class="co">#&gt;   ..$ 4: chr "Q13445"</span>
<span class="co">#&gt;   ..$ 5: chr "Q6P4F2"</span>
<span class="co">#&gt;   ..$ 6: chr "Q8N9T8"</span>
<span class="co">#&gt;   ..$ 7: chr "Q9BYD3"</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int 7</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 1</span>
<span class="co">#&gt;   .. ..$ : chr [1:7] "2" "1" "3" "4" ...</span>
<span class="co">#&gt;  $ group_size  : Named int [1:7] 3 1 1 1 1 1 1</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:7] "2" "1" "3" "4" ...</span></pre></body></html></div>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">pathway_protein_graph</span>)</pre></body></html></div>
<p><img src="case-1-pleiotropy_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="retrieving-shared-protein-protein-interactions" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieving-shared-protein-protein-interactions" class="anchor"></a>Retrieving shared protein-protein interactions</h2>
<p>We can complement the information given by pathway ontologies with data of interactions between proteins. For each pair of proteins, we retrieve shared protein-protein interactions, either a direct interaction or an interaction where there’s one mediator protein in the middle (see paramaters section).</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">get_ppi</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">gene_protein_df</span>, <span class="no">n_intermediate_proteins</span> <span class="kw">=</span> <span class="fl">0</span>) {
  <span class="no">endpoint</span> <span class="kw">&lt;-</span> <span class="st">"/protein/ppi/pairwise"</span>
  <span class="no">params</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    <span class="kw">uniprot_id_list</span> <span class="kw">=</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="st">"uniprot_id"</span>) <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span>(),
    <span class="kw">n_intermediate_proteins</span> <span class="kw">=</span> <span class="no">n_intermediate_proteins</span>
  )
  <span class="no">df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(
    <span class="kw">route</span> <span class="kw">=</span> <span class="no">endpoint</span>,
    <span class="kw">params</span> <span class="kw">=</span> <span class="no">params</span>,
    <span class="kw">mode</span> <span class="kw">=</span> <span class="st">"table"</span>,
    <span class="kw">method</span> <span class="kw">=</span> <span class="st">"POST"</span>
  )

  <span class="kw">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">df</span>) <span class="kw">&gt;</span> <span class="fl">0</span>) {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">protein</span> <span class="kw">=</span> <span class="no">uniprot_id</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span>(<span class="no">df</span>, <span class="kw">by</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"protein"</span>))
  } <span class="kw">else</span> {
    <span class="no">res_df</span> <span class="kw">&lt;-</span> <span class="no">gene_protein_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">protein</span> <span class="kw">=</span> <span class="no">uniprot_id</span>) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">assoc_protein</span> <span class="kw">=</span> <span class="fl">NA_character_</span>, <span class="kw">path_size</span> <span class="kw">=</span> <span class="fl">NA_integer_</span>)
  }
  <span class="no">res_df</span>
}
<span class="no">ppi_df</span> <span class="kw">&lt;-</span> <span class="fu">get_ppi</span>(<span class="no">gene_protein_df</span>, <span class="kw">n_intermediate_proteins</span> <span class="kw">=</span> <span class="no">PPI_N_INTERMEDIATE_PROTEINS</span>)
<span class="no">ppi_df</span>
<span class="co">#&gt; # A tibble: 77 x 3</span>
<span class="co">#&gt;    protein assoc_protein path_size</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;             &lt;int&gt;</span>
<span class="co">#&gt;  1 Q13445  &lt;NA&gt;                 NA</span>
<span class="co">#&gt;  2 Q9BYD3  Q8N9T8                2</span>
<span class="co">#&gt;  3 Q9BYD3  Q8N9T8                2</span>
<span class="co">#&gt;  4 Q9BYD3  Q8N9T8                2</span>
<span class="co">#&gt;  5 Q9BYD3  Q8N9T8                2</span>
<span class="co">#&gt;  6 P29597  P05362                2</span>
<span class="co">#&gt;  7 P29597  P05362                2</span>
<span class="co">#&gt;  8 P29597  P05362                2</span>
<span class="co">#&gt;  9 P29597  P05362                2</span>
<span class="co">#&gt; 10 P29597  O75791                2</span>
<span class="co">#&gt; # … with 67 more rows</span></pre></body></html></div>
<p>Like in the query for pathway data, we convert and analyse these results as a graph.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">get_shared_ppi</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">ppi_df</span>) {
  <span class="no">per_row</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">ppi_df</span>, <span class="no">x</span>, <span class="no">y</span>) {
    <span class="no">res</span> <span class="kw">&lt;-</span> <span class="no">ppi_df</span> <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(
        <span class="no">`protein`</span> <span class="kw">==</span> <span class="no">x</span>,
        <span class="no">`assoc_protein`</span> <span class="kw">==</span> <span class="no">y</span>
      ) <span class="kw">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>() <span class="kw">%&gt;%</span>
      <span class="fu">`&gt;`</span>(<span class="fl">0</span>)
    <span class="no">res</span>
  }

  <span class="no">pairwise_permutations</span> <span class="kw">&lt;-</span> <span class="no">ppi_df</span> <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="no">`protein`</span>) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>() <span class="kw">%&gt;%</span>
    <span class="kw pkg">gtools</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gtools/man/combinations.html">permutations</a></span>(<span class="kw">n</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">.</span>), <span class="kw">r</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">v</span> <span class="kw">=</span> <span class="no">.</span>)
  <span class="no">shared_ppi_df</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">tibble</a></span>(
    <span class="kw">protein</span> <span class="kw">=</span> <span class="no">pairwise_permutations</span>[, <span class="fl">1</span>],
    <span class="kw">assoc_protein</span> <span class="kw">=</span> <span class="no">pairwise_permutations</span>[, <span class="fl">2</span>]
  ) <span class="kw">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(
      <span class="kw">shared_ppi</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_lgl</a></span>(<span class="no">`protein`</span>, <span class="no">`assoc_protein`</span>, <span class="kw">function</span>(<span class="no">x</span>, <span class="no">y</span>) {
        <span class="fu">per_row</span>(<span class="kw">ppi_df</span> <span class="kw">=</span> <span class="no">ppi_df</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="no">x</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">y</span>)
      }),
      <span class="kw">combination</span> <span class="kw">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2_chr</a></span>(<span class="no">`protein`</span>, <span class="no">`assoc_protein`</span>, <span class="kw">function</span>(<span class="no">x</span>, <span class="no">y</span>) {
        <span class="no">comb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">x</span>, <span class="no">y</span>))
        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="no">comb</span>, <span class="kw">collapse</span> <span class="kw">=</span> <span class="st">","</span>)
      }),
      <span class="kw">connected</span> <span class="kw">=</span> <span class="no">shared_ppi</span>
    )

  <span class="no">shared_ppi_df</span>
}

<span class="no">shared_ppi_df</span> <span class="kw">&lt;-</span> <span class="fu">get_shared_ppi</span>(<span class="no">ppi_df</span>)
<span class="no">n_ppi_pairs</span> <span class="kw">&lt;-</span> <span class="no">shared_ppi_df</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">shared_ppi</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>()
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">glue</span><span class="kw ns">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span>(<span class="st">"Num. shared_ppi pairs: {n_ppi_pairs}"</span>))
<span class="co">#&gt; Num. shared_ppi pairs: 10</span>
<span class="no">shared_ppi_df</span>
<span class="co">#&gt; # A tibble: 72 x 5</span>
<span class="co">#&gt;    protein assoc_protein shared_ppi combination   connected</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;lgl&gt;      &lt;chr&gt;         &lt;lgl&gt;    </span>
<span class="co">#&gt;  1 O75791  P05362        TRUE       O75791,P05362 TRUE     </span>
<span class="co">#&gt;  2 O75791  P0C6A0        FALSE      O75791,P0C6A0 FALSE    </span>
<span class="co">#&gt;  3 O75791  P29597        TRUE       O75791,P29597 TRUE     </span>
<span class="co">#&gt;  4 O75791  Q13445        FALSE      O75791,Q13445 FALSE    </span>
<span class="co">#&gt;  5 O75791  Q6P4F2        FALSE      O75791,Q6P4F2 FALSE    </span>
<span class="co">#&gt;  6 O75791  Q8N9T8        FALSE      O75791,Q8N9T8 FALSE    </span>
<span class="co">#&gt;  7 O75791  Q9BYD3        FALSE      O75791,Q9BYD3 FALSE    </span>
<span class="co">#&gt;  8 O75791  Q9UMF0        FALSE      O75791,Q9UMF0 FALSE    </span>
<span class="co">#&gt;  9 P05362  O75791        TRUE       O75791,P05362 TRUE     </span>
<span class="co">#&gt; 10 P05362  P0C6A0        FALSE      P05362,P0C6A0 FALSE    </span>
<span class="co">#&gt; # … with 62 more rows</span></pre></body></html></div>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">ppi_protein_graph</span> <span class="kw">&lt;-</span> <span class="fu">protein_df_to_graph</span>(<span class="no">shared_ppi_df</span>)
<span class="no">ppi_protein_groups</span> <span class="kw">&lt;-</span> <span class="fu">graph_to_protein_groups</span>(<span class="no">ppi_protein_graph</span>)
<span class="no">ppi_protein_groups</span> <span class="kw">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>()
<span class="co">#&gt; tibble [5 × 2] (S3: tbl_df/tbl/data.frame)</span>
<span class="co">#&gt;  $ group_member:List of 5</span>
<span class="co">#&gt;   ..$ 1: chr [1:4] "O75791" "P05362" "P29597" "Q9UMF0"</span>
<span class="co">#&gt;   ..$ 5: chr [1:2] "Q8N9T8" "Q9BYD3"</span>
<span class="co">#&gt;   ..$ 2: chr "P0C6A0"</span>
<span class="co">#&gt;   ..$ 3: chr "Q13445"</span>
<span class="co">#&gt;   ..$ 4: chr "Q6P4F2"</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int 5</span>
<span class="co">#&gt;   ..- attr(*, "dimnames")=List of 1</span>
<span class="co">#&gt;   .. ..$ : chr [1:5] "1" "5" "2" "3" ...</span>
<span class="co">#&gt;  $ group_size  : Named int [1:5] 4 2 1 1 1</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:5] "1" "5" "2" "3" ...</span></pre></body></html></div>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">ppi_protein_graph</span>)</pre></body></html></div>
<p><img src="case-1-pleiotropy_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.0 (2020-04-24)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 16.04.6 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</span>
<span class="co">#&gt; LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2 igraph_1.2.5   glue_1.4.1     purrr_0.3.4    dplyr_1.0.0   </span>
<span class="co">#&gt; [6] magrittr_1.5  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.4.6       knitr_1.28         MASS_7.3-51.5      tidyselect_1.1.0  </span>
<span class="co">#&gt;  [5] R6_2.4.1           rlang_0.4.6        fansi_0.4.1        httr_1.4.1        </span>
<span class="co">#&gt;  [9] stringr_1.4.0      tools_4.0.0        xfun_0.14          utf8_1.1.4        </span>
<span class="co">#&gt; [13] cli_2.0.2          gtools_3.8.2       ellipsis_0.3.1     htmltools_0.4.0   </span>
<span class="co">#&gt; [17] yaml_2.2.1         assertthat_0.2.1   digest_0.6.25      rprojroot_1.3-2   </span>
<span class="co">#&gt; [21] tibble_3.0.1       lifecycle_0.2.0    pkgdown_1.5.1.9000 crayon_1.3.4      </span>
<span class="co">#&gt; [25] vctrs_0.3.0        fs_1.4.1           curl_4.3           memoise_1.1.0     </span>
<span class="co">#&gt; [29] evaluate_0.14      rmarkdown_2.2      stringi_1.4.6      pillar_1.4.4      </span>
<span class="co">#&gt; [33] compiler_4.0.0     desc_1.2.0         generics_0.0.2     backports_1.1.7   </span>
<span class="co">#&gt; [37] jsonlite_1.6.1     pkgconfig_2.0.3</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Yi Liu, Valeriia Haberland, Pau Erola, Benjamin Elsworth, Tom Gaunt.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
