<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study 2: Identification of potential drug targets • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Case study 2: Identification of potential drug targets">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="case-2-alt-drug-target_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study 2: Identification of potential drug targets</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/case-2-alt-drug-target.Rmd"><code>vignettes/case-2-alt-drug-target.Rmd</code></a></small>
      <div class="hidden name"><code>case-2-alt-drug-target.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-2-alt-drug-target.ipynb">notebook for case study 2 in the EpiGraphDB paper</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="#">paper</a> or the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-2-alt-drug-target.ipynb">case study notebook</a>.</p>
<div id="context" class="section level2">
<h2 class="hasAnchor">
<a href="#context" class="anchor"></a>Context</h2>
<p>Systematic MR of molecular phenotypes such as proteins and expression of transcript levels offer enormous potential to prioritise drug targets for further investigation. However, many genes and gene products are not easily druggable, so some potentially important causal genes may not offer an obvious route to intervention.</p>
<p>A parallel problem is that current GWASes of molecular phenotypes have limited sample sizes and limited protein coverages. A potential way to address both these problems is to use protein-protein interaction information to identify druggable targets which are linked to a non-druggable, but robustly causal target. Their relationship to the causal target increases our confidence in their potential causal role even if the initial evidence of effect is below our multiple-testing threshold.</p>
<p>Here in case study 2 we demonstrate an approach to use data in EpiGraphDB to prioritise potential alternative drug targets in the same PPI network, as follows:</p>
<ul>
<li>For an existing drug target of interests, we use PPI networks to search for its directly interacting genes that are evidenced to be druggable.</li>
<li>We then examine the causal evidence of these candidate genes on the disease.</li>
<li>We also examine the literature evidence of these candidate genes on the disease.</li>
</ul>
<p>The triangulation of MR evidence and literature evidence as available from EpiGraphDB regarding these candidate genes will greatly enhance our confidence in identifying potential viable drug targets.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://magrittr.tidyverse.org">"magrittr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org">"dplyr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://purrr.tidyverse.org">"purrr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:magrittr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     set_names</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/tidyverse/glue">"glue"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'glue'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://mrcieu.github.io/epigraphdb-r/">"epigraphdb"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     EpiGraphDB v1.0 (API: https://api.epigraphdb.org)</span>
<span class="co">#&gt; </span></code></pre></div>
<p>Here we configure the parameters used in the case study example. We illustrate this approach using IL23R, an established drug target for inflammatory bowel disease (IBD) (Duerr et al., 2006; Momozawa et al., 2011).</p>
<p>While specific IL23R interventions are still undergoing trials, there is a possibility that these therapies may not be effective for all or even the majority of patients. This case study therefore explores potential alternative drug targets.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GENE_NAME</span> <span class="op">&lt;-</span> <span class="st">"IL23R"</span>
<span class="va">OUTCOME_TRAIT</span> <span class="op">&lt;-</span> <span class="st">"Inflammatory bowel disease"</span></code></pre></div>
</div>
<div id="using-ppi-networks-for-alternative-drug-targets-search" class="section level2">
<h2 class="hasAnchor">
<a href="#using-ppi-networks-for-alternative-drug-targets-search" class="anchor"></a>Using PPI networks for alternative drug targets search</h2>
<p>The assumption here is that the most likely alternative targets are either directly interacting with IL23R or somewhere in the same PPI network. In this example, we consider only genes that were found to interact with IL23R via direct protein-protein interactions, and require that those <strong>interacting proteins</strong> should also be <strong>druggable</strong>.</p>
<p>The thousands of genes are classified with regard to their druggability by Finan et al. 2017, where the <strong>Tier 1</strong> category refers to approved drugs or those in clinical testing while for other tier categories the druggability confidence drops in order <strong>Tier 2</strong> and then <strong>Tier 3</strong>.</p>
<p>Here we use the <a href="https://docs.epigraphdb.org/api/api-endpoints/#get-genedruggabilityppi">GET /gene/druggability/ppi</a> endpoint to get data on the druggable alternative genes.**</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_drug_targets_ppi</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/gene/druggability/ppi"</span>
  <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="va">gene_name</span><span class="op">)</span>
  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
  <span class="va">df</span>
<span class="op">}</span>

<span class="va">ppi_df</span> <span class="op">&lt;-</span> <span class="fu">get_drug_targets_ppi</span><span class="op">(</span>gene_name <span class="op">=</span> <span class="va">GENE_NAME</span><span class="op">)</span>
<span class="va">ppi_df</span>
<span class="co">#&gt; # A tibble: 42 x 5</span>
<span class="co">#&gt;    g1.name p1.uniprot_id p2.uniprot_id g2.name g2.druggability_tier</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;   &lt;chr&gt;               </span>
<span class="co">#&gt;  1 IL23R   Q5VWK5        P04141        CSF2    Tier 1              </span>
<span class="co">#&gt;  2 IL23R   Q5VWK5        P01562        IFNA1   Tier 1              </span>
<span class="co">#&gt;  3 IL23R   Q5VWK5        P01579        IFNG    Tier 1              </span>
<span class="co">#&gt;  4 IL23R   Q5VWK5        P22301        IL10    Tier 1              </span>
<span class="co">#&gt;  5 IL23R   Q5VWK5        P29460        IL12B   Tier 1              </span>
<span class="co">#&gt;  6 IL23R   Q5VWK5        P42701        IL12RB1 Tier 1              </span>
<span class="co">#&gt;  7 IL23R   Q5VWK5        P35225        IL13    Tier 1              </span>
<span class="co">#&gt;  8 IL23R   Q5VWK5        P40933        IL15    Tier 1              </span>
<span class="co">#&gt;  9 IL23R   Q5VWK5        Q16552        IL17A   Tier 1              </span>
<span class="co">#&gt; 10 IL23R   Q5VWK5        Q96PD4        IL17F   Tier 1              </span>
<span class="co">#&gt; # … with 32 more rows</span></code></pre></div>
<p>For further analysis we select the gene of interest (IL23R) as well as its interacting genes with Tier 1 druggability.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">get_gene_list</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ppi_df</span>, <span class="va">include_primary_gene</span> <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="va">include_primary_gene</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>
      <span class="va">ppi_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`g1.name`</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span>,
      <span class="va">ppi_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">`g2.druggability_tier`</span> <span class="op">==</span> <span class="st">"Tier 1"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`g2.name`</span><span class="op">)</span>
    <span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="va">ppi_df</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">`g2.druggability_tier`</span> <span class="op">==</span> <span class="st">"Tier 1"</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">`g2.name`</span><span class="op">)</span>
  <span class="op">}</span>
  <span class="va">gene_list</span>
<span class="op">}</span>

<span class="va">gene_list</span> <span class="op">&lt;-</span> <span class="fu">get_gene_list</span><span class="op">(</span><span class="va">ppi_df</span><span class="op">)</span>
<span class="va">gene_list</span>
<span class="co">#&gt;  [1] "IL23R"   "CSF2"    "IFNA1"   "IFNG"    "IL10"    "IL12B"   "IL12RB1"</span>
<span class="co">#&gt;  [8] "IL13"    "IL15"    "IL17A"   "IL17F"   "IL2"     "IL22"    "IL23A"  </span>
<span class="co">#&gt; [15] "IL4"     "IL5"     "IL6"     "IL9"     "JAK1"    "JAK2"    "NFKB1"  </span>
<span class="co">#&gt; [22] "PIK3CA"  "RORC"    "STAT3"   "TSLP"    "TYK2"</span></code></pre></div>
</div>
<div id="using-mendelian-randomization-results-for-causal-effect-estimation" class="section level2">
<h2 class="hasAnchor">
<a href="#using-mendelian-randomization-results-for-causal-effect-estimation" class="anchor"></a>Using Mendelian randomization results for causal effect estimation</h2>
<p>The next step is to find out whether any of these genes have a comparable and statistically plausible effect on IBD.</p>
<p>Here we search EpiGraphDB for the Mendelian randomization (MR) results for these genes and IBD from the recent study by Zheng et al, 2019 (<a href="https://epigraphdb.org/xqtl/" class="uri">https://epigraphdb.org/xqtl/</a>) via the <a href="https://docs.epigraphdb.org/api/api-endpoints/#get-xqtlsingle-snp-mr">GET /xqtl/single-snp-mr</a> endpoint.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">extract_mr</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">outcome_trait</span>, <span class="va">gene_list</span>, <span class="va">qtl_type</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/xqtl/single-snp-mr"</span>
  <span class="va">per_gene</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
      exposure_gene <span class="op">=</span> <span class="va">gene_name</span>,
      outcome_trait <span class="op">=</span> <span class="va">outcome_trait</span>,
      qtl_type <span class="op">=</span> <span class="va">qtl_type</span>,
      pval_threshold <span class="op">=</span> <span class="fl">1e-5</span>
    <span class="op">)</span>
    <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
    <span class="va">df</span>
  <span class="op">}</span>
  <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_list</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="va">per_gene</span><span class="op">)</span>
  <span class="va">res_df</span>
<span class="op">}</span>

<span class="va">xqtl_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"pQTL"</span>, <span class="st">"eQTL"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">qtl_type</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">extract_mr</span><span class="op">(</span>
    outcome_trait <span class="op">=</span> <span class="va">OUTCOME_TRAIT</span>,
    gene_list <span class="op">=</span> <span class="va">gene_list</span>,
    qtl_type <span class="op">=</span> <span class="va">qtl_type</span>
  <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>qtl_type <span class="op">=</span> <span class="va">qtl_type</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="va">xqtl_df</span>
<span class="co">#&gt; # A tibble: 9 x 9</span>
<span class="co">#&gt;   gene.ensembl_id gene.name gwas.id  gwas.trait   r.beta   r.se       r.p r.rsid</span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;    &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt; </span>
<span class="co">#&gt; 1 ENSG00000162594 IL23R     ieu-a-2… Inflammator…  1.50  0.0546 2.21e-166 rs115…</span>
<span class="co">#&gt; 2 ENSG00000113302 IL12B     ieu-a-2… Inflammator…  0.418 0.0345 9.59e- 34 rs492…</span>
<span class="co">#&gt; 3 ENSG00000162594 IL23R     ieu-a-2… Inflammator…  0.887 0.0644 4.17e- 43 rs206…</span>
<span class="co">#&gt; 4 ENSG00000164136 IL15      ieu-a-2… Inflammator… -1.42  0.197  5.53e- 13 rs753…</span>
<span class="co">#&gt; 5 ENSG00000113520 IL4       ieu-a-2… Inflammator…  0.460 0.0840 4.47e-  8 rs207…</span>
<span class="co">#&gt; 6 ENSG00000096968 JAK2      ieu-a-2… Inflammator… -1.90  0.204  1.32e- 20 rs478…</span>
<span class="co">#&gt; 7 ENSG00000109320 NFKB1     ieu-a-2… Inflammator…  0.974 0.174  2.16e-  8 rs476…</span>
<span class="co">#&gt; 8 ENSG00000143365 RORC      ieu-a-2… Inflammator… -0.995 0.116  1.21e- 17 rs484…</span>
<span class="co">#&gt; 9 ENSG00000168610 STAT3     ieu-a-2… Inflammator…  0.597 0.0757 2.96e- 15 rs105…</span>
<span class="co">#&gt; # … with 1 more variable: qtl_type &lt;chr&gt;</span></code></pre></div>
</div>
<div id="using-literature-evidence-for-results-enrichment-and-triangulation" class="section level2">
<h2 class="hasAnchor">
<a href="#using-literature-evidence-for-results-enrichment-and-triangulation" class="anchor"></a>Using literature evidence for results enrichment and triangulation</h2>
<p>Can we find evidence in the literature where these genes are found to be associated with IBD to increase our level of confidence in MR results or to provide alternative evidence where MR results to not exist?</p>
<p>We can use the <a href="https://docs.epigraphdb.org/api/api-endpoints/#get-geneliterature">GET /gene/literature</a> endpoint to get data on the literature evidence for the set of genes.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">extract_literature</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">outcome_trait</span>, <span class="va">gene_list</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">per_gene</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/gene/literature"</span>
    <span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
      gene_name <span class="op">=</span> <span class="va">gene_name</span>,
      object_name <span class="op">=</span> <span class="va">outcome_trait</span> <span class="op">%&gt;%</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/case.html">str_to_lower</a></span><span class="op">(</span><span class="op">)</span>
    <span class="op">)</span>
    <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
    <span class="va">df</span>
  <span class="op">}</span>
  <span class="va">res_df</span> <span class="op">&lt;-</span> <span class="va">gene_list</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="va">per_gene</span><span class="op">)</span>
  <span class="va">res_df</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>literature_count <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_int</a></span><span class="op">(</span><span class="va">pubmed_id</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">literature_df</span> <span class="op">&lt;-</span> <span class="fu">extract_literature</span><span class="op">(</span>
  outcome_trait <span class="op">=</span> <span class="va">OUTCOME_TRAIT</span>,
  gene_list <span class="op">=</span> <span class="va">gene_list</span>
<span class="op">)</span>
<span class="va">literature_df</span>
<span class="co">#&gt; # A tibble: 49 x 7</span>
<span class="co">#&gt;    pubmed_id  gene.name lt.id  lt.name    lt.type st.predicate  literature_count</span>
<span class="co">#&gt;    &lt;list&gt;     &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;      &lt;list&gt;  &lt;chr&gt;                    &lt;int&gt;</span>
<span class="co">#&gt;  1 &lt;chr [1]&gt;  IL23R     C0021… Inflammat… &lt;chr [… PREDISPOSES                  1</span>
<span class="co">#&gt;  2 &lt;chr [2]&gt;  IL23R     C0021… Inflammat… &lt;chr [… NEG_ASSOCIAT…                2</span>
<span class="co">#&gt;  3 &lt;chr [1]&gt;  IL23R     C0021… Inflammat… &lt;chr [… CAUSES                       1</span>
<span class="co">#&gt;  4 &lt;chr [21]&gt; IL23R     C0021… Inflammat… &lt;chr [… ASSOCIATED_W…               21</span>
<span class="co">#&gt;  5 &lt;chr [1]&gt;  IL23R     C0021… Inflammat… &lt;chr [… AFFECTS                      1</span>
<span class="co">#&gt;  6 &lt;chr [2]&gt;  CSF2      C0021… Inflammat… &lt;chr [… ASSOCIATED_W…                2</span>
<span class="co">#&gt;  7 &lt;chr [1]&gt;  CSF2      C0021… Inflammat… &lt;chr [… AFFECTS                      1</span>
<span class="co">#&gt;  8 &lt;chr [1]&gt;  IFNA1     C0021… Inflammat… &lt;chr [… TREATS                       1</span>
<span class="co">#&gt;  9 &lt;chr [1]&gt;  IFNA1     C0021… Inflammat… &lt;chr [… PREVENTS                     1</span>
<span class="co">#&gt; 10 &lt;chr [3]&gt;  IFNA1     C0021… Inflammat… &lt;chr [… ASSOCIATED_W…                3</span>
<span class="co">#&gt; # … with 39 more rows</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.4 (2021-02-15)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2.2 glue_1.4.2       purrr_0.3.4      dplyr_1.0.5     </span>
<span class="co">#&gt; [5] magrittr_2.0.1  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] rstudioapi_0.13   knitr_1.31        tidyselect_1.1.0  R6_2.5.0         </span>
<span class="co">#&gt;  [5] ragg_1.1.2        rlang_0.4.10      fastmap_1.1.0     fansi_0.4.2      </span>
<span class="co">#&gt;  [9] httr_1.4.2        stringr_1.4.0     tools_4.0.4       xfun_0.22        </span>
<span class="co">#&gt; [13] utf8_1.2.1        cli_2.3.1         ellipsis_0.3.1    htmltools_0.5.1.1</span>
<span class="co">#&gt; [17] systemfonts_1.0.1 assertthat_0.2.1  yaml_2.2.1        digest_0.6.27    </span>
<span class="co">#&gt; [21] rprojroot_2.0.2   tibble_3.1.0      lifecycle_1.0.0   pkgdown_1.6.1    </span>
<span class="co">#&gt; [25] crayon_1.4.1      textshaping_0.3.3 ps_1.6.0          vctrs_0.3.6      </span>
<span class="co">#&gt; [29] fs_1.5.0          curl_4.3          memoise_2.0.0     cachem_1.0.4     </span>
<span class="co">#&gt; [33] evaluate_0.14     rmarkdown_2.7     stringi_1.5.3     pillar_1.5.1     </span>
<span class="co">#&gt; [37] compiler_4.0.4    desc_1.3.0        generics_0.1.0    jsonlite_1.7.2   </span>
<span class="co">#&gt; [41] pkgconfig_2.0.3</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
