<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Case study 3: Triangulating causal estimates with literature evidence • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Case study 3: Triangulating causal estimates with literature evidence">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Case study 3: Triangulating causal estimates with literature evidence</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/case-3-literature-triangulation.Rmd"><code>vignettes/case-3-literature-triangulation.Rmd</code></a></small>
      <div class="hidden name"><code>case-3-literature-triangulation.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-3-literature-triangulation.ipynb">notebook for case study 3 in the EpiGraphDB paper</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="#">paper</a> or the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/paper-case-studies/case-3-literature-triangulation.ipynb">case study notebook</a>.</p>
<div id="context" class="section level2">
<h2 class="hasAnchor">
<a href="#context" class="anchor"></a>Context</h2>
<p>The biomedical literature contains a wealth of information than far exceeds our capacity for systematic manual extraction. For this reason, there are many existing literature mining methods to extract the key concepts and content. Here we use data from <a href="https://skr3.nlm.nih.gov/SemMedDB/">SemMedDB</a>, a well established database that provides subject-predicate-object triples from all PubMed titles and abstracts. Using a subset of this data we created MELODI-presto (<a href="https://melodi-presto.mrcieu.ac.uk/" class="uri">https://melodi-presto.mrcieu.ac.uk/</a>), a method to assign triples to any given biomedical query via a PubMed search and some basic enrichment, and have applied this systematically to traits represented in EpiGraphDB. This allows us to identify overlapping terms connecting any set of GWAS traits, e.g. exposure and disease outcome. From here we can attempt to triangulate causal estimates, and conversely, check the mechanisms identified from the literature against the causal evidence.</p>
<p>This case study goes as follows:</p>
<ul>
<li>For an exposure trait we identify its causal evidence against other outcome traits, then link the outcomes to diseases</li>
<li>Then for an exposure-outcome pair we look for its literature evidence, and the mechanisms (as SemMed triples) identified from the literature evidence</li>
<li>Finally we look into detail the mechanisms specifically related to an overlapping term and visualise these mechanisms.</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"magrittr"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://dplyr.tidyverse.org">"dplyr"</a></span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="http://purrr.tidyverse.org">"purrr"</a></span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:magrittr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     set_names</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/tidyverse/glue">"glue"</a></span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'glue'</span>
<span class="co">#&gt; The following object is masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="http://ggplot2.tidyverse.org">"ggplot2"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="http://igraph.org">"igraph"</a></span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'igraph'</span>
<span class="co">#&gt; The following objects are masked from 'package:purrr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     compose, simplify</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     as_data_frame, groups, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://mrcieu.github.io/epigraphdb-r">"epigraphdb"</a></span>)
<span class="co">#&gt; </span>
<span class="co">#&gt;     EpiGraphDB v0.3 (API: https://api.epigraphdb.org)</span>
<span class="co">#&gt; </span>
</pre></div>
<p>Here we set the starting trait, which we will use to explore associated disease traits.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw">STARTING_TRAIT</span> <span class="op">&lt;-</span> <span class="st">"Sleep duration"</span>
</pre></div>
</div>
<div id="get-traits-mr-association" class="section level2">
<h2 class="hasAnchor">
<a href="#get-traits-mr-association" class="anchor"></a>Get traits MR association</h2>
<p>Given an exposure trait, find all traits with causal evidence. This method searches the causal evidence data for cases where our exposure trait has a potential casual effect on an outcome trait.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">get_mr</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">trait</span>) {
  <span class="kw">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>
  <span class="kw">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    exposure_trait = <span class="kw">trait</span>,
    pval_threshold = <span class="fl">1e-10</span>
  )
  <span class="kw">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(route = <span class="kw">endpoint</span>, params = <span class="kw">params</span>, mode = <span class="st">"table"</span>)
  <span class="kw">mr_df</span>
}

<span class="kw">mr_df</span> <span class="op">&lt;-</span> <span class="fu">get_mr</span>(<span class="kw">STARTING_TRAIT</span>)
<span class="kw">mr_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span>()
<span class="co">#&gt; Rows: 29</span>
<span class="co">#&gt; Columns: 10</span>
<span class="co">#&gt; $ exposure.id    &lt;chr&gt; "ieu-a-1088", "ieu-a-1088", "ieu-a-1088", "ieu-a-1088"…</span>
<span class="co">#&gt; $ exposure.trait &lt;chr&gt; "Sleep duration", "Sleep duration", "Sleep duration", …</span>
<span class="co">#&gt; $ outcome.id     &lt;chr&gt; "ukb-a-460", "ieu-a-118", "ieu-a-115", "ieu-a-1073", "…</span>
<span class="co">#&gt; $ outcome.trait  &lt;chr&gt; "Vitamin and mineral supplements: Vitamin B", "Neuroti…</span>
<span class="co">#&gt; $ mr.b           &lt;dbl&gt; -0.0100669162, -0.1818909943, 2.3293509483, 0.25011813…</span>
<span class="co">#&gt; $ mr.se          &lt;dbl&gt; 0.0001310998, 0.0114868227, 0.1428329498, 0.0063949227…</span>
<span class="co">#&gt; $ mr.pval        &lt;dbl&gt; 0.000000e+00, 0.000000e+00, 0.000000e+00, 0.000000e+00…</span>
<span class="co">#&gt; $ mr.method      &lt;chr&gt; "FE IVW", "FE IVW", "FE IVW", "FE IVW", "FE IVW", "FE …</span>
<span class="co">#&gt; $ mr.selection   &lt;chr&gt; "DF", "DF", "DF", "DF", "DF", "DF", "DF", "HF", "DF", …</span>
<span class="co">#&gt; $ mr.moescore    &lt;dbl&gt; 1.00, 1.00, 1.00, 1.00, 1.00, 1.00, 1.00, 0.94, 1.00, …</span>
</pre></div>
</div>
<div id="map-outcome-traits-to-disease" class="section level2">
<h2 class="hasAnchor">
<a href="#map-outcome-traits-to-disease" class="anchor"></a>Map outcome traits to disease</h2>
<p>For this example, we are interested in traits mapped to a disease node. To do this we utilise the mapping from GWAS trait to Disease via EFO term.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="kw">trait_to_disease</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">trait</span>) {
  <span class="kw">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/ontology/gwas-efo-disease"</span>
  <span class="kw">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(trait = <span class="kw">trait</span>)
  <span class="kw">disease_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(route = <span class="kw">endpoint</span>, params = <span class="kw">params</span>, mode = <span class="st">"table"</span>)
  <span class="co">if</span> (<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">disease_df</span>) <span class="op">&gt;</span> <span class="fl">0</span>) {
    <span class="kw">res</span> <span class="op">&lt;-</span> <span class="kw">disease_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span>(<span class="kw">`disease.label`</span>)
  } <span class="co">else</span> {
    <span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>()
  }
  <span class="kw">res</span>
}

<span class="kw">disease_df</span> <span class="op">&lt;-</span> <span class="kw">mr_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(disease = <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span>(<span class="kw">`outcome.trait`</span>, <span class="kw">trait_to_disease</span>)) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_lgl</a></span>(<span class="kw">`disease`</span>, <span class="fu">function</span>(<span class="kw">x</span>) <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span>(<span class="kw">x</span>)))
<span class="kw">disease_df</span>
<span class="co">#&gt; # A tibble: 5 x 11</span>
<span class="co">#&gt;   exposure.id exposure.trait outcome.id outcome.trait     mr.b   mr.se  mr.pval</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ieu-a-1088  Sleep duration ukb-a-107  Non-cancer i… -0.00257 2.53e-4 3.84e-24</span>
<span class="co">#&gt; 2 ieu-a-1088  Sleep duration ieu-a-6    Coronary hea… -1.04    1.10e-1 2.32e-21</span>
<span class="co">#&gt; 3 ieu-a-1088  Sleep duration ukb-a-548  Diagnoses - … -0.00671 8.64e-4 8.01e-15</span>
<span class="co">#&gt; 4 ieu-a-1088  Sleep duration ukb-a-54   Cancer code … -0.00191 2.47e-4 1.07e-14</span>
<span class="co">#&gt; 5 ukb-a-9     Sleep duration ukb-a-13   Sleeplessnes… -0.322   3.45e-2 1.10e-11</span>
<span class="co">#&gt; # … with 4 more variables: mr.method &lt;chr&gt;, mr.selection &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.moescore &lt;dbl&gt;, disease &lt;list&gt;</span>
</pre></div>
</div>
<div id="take-one-example-to-look-for-literature-evidence" class="section level2">
<h2 class="hasAnchor">
<a href="#take-one-example-to-look-for-literature-evidence" class="anchor"></a>Take one example to look for literature evidence</h2>
<p>For the multiple <code>exposure -&gt; outcome</code> relationships as reported from the table above, here we look at the literature evidence for one pair in detail:</p>
<ul>
<li>Trait X: “Sleep duration” (<a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-1088/">ieu-a-1088</a>)</li>
<li>Trait Y: “Coronary heart disease” (<a href="https://gwas.mrcieu.ac.uk/datasets/ieu-a-6/">ieu-a-6</a>).</li>
</ul>
<p>The following looks for enriched triples of information (Subject-Predicate-Object) associated with our two traits. These have been derived via PubMed searches and corresponding <a href="https://skr3.nlm.nih.gov/SemMedDB/">SemMedDB</a> data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="kw">get_gwas_pair_literature</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">gwas_id</span>, <span class="kw">assoc_gwas_id</span>) {
  <span class="kw">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/literature/gwas/pairwise"</span>
  <span class="co"># NOTE in this example we blacklist to semmentic types</span>
  <span class="kw">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    gwas_id = <span class="kw">gwas_id</span>,
    assoc_gwas_id = <span class="kw">assoc_gwas_id</span>,
    by_gwas_id = <span class="fl">TRUE</span>,
    pval_threshold = <span class="fl">1e-1</span>,
    semmantic_types = <span class="st">"nusq"</span>,
    semmantic_types = <span class="st">"dsyn"</span>,
    blacklist = <span class="fl">TRUE</span>,
    limit = <span class="fl">1000</span>
  )
  <span class="kw">lit_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(route = <span class="kw">endpoint</span>, params = <span class="kw">params</span>, mode = <span class="st">"table"</span>)
  <span class="kw">lit_df</span>
}

<span class="kw">GWAS_ID_X</span> <span class="op">&lt;-</span> <span class="st">"ieu-a-1088"</span>
<span class="kw">GWAS_ID_Y</span> <span class="op">&lt;-</span> <span class="st">"ieu-a-6"</span>
<span class="kw">lit_df</span> <span class="op">&lt;-</span> <span class="fu">get_gwas_pair_literature</span>(<span class="kw">GWAS_ID_X</span>, <span class="kw">GWAS_ID_Y</span>)

<span class="fu"><a href="https://tibble.tidyverse.org/reference/glimpse.html">glimpse</a></span>(<span class="kw">lit_df</span>)
<span class="co">#&gt; Rows: 839</span>
<span class="co">#&gt; Columns: 18</span>
<span class="co">#&gt; $ gwas.trait       &lt;chr&gt; "Sleep duration", "Sleep duration", "Sleep duration"…</span>
<span class="co">#&gt; $ gwas.id          &lt;chr&gt; "ieu-a-1088", "ieu-a-1088", "ieu-a-1088", "ieu-a-108…</span>
<span class="co">#&gt; $ gs1.localCount   &lt;int&gt; 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2…</span>
<span class="co">#&gt; $ gs1.pval         &lt;dbl&gt; 7.851119e-06, 7.851119e-06, 7.851119e-06, 7.851119e-…</span>
<span class="co">#&gt; $ s1.subject_name  &lt;chr&gt; "Caffeine", "Caffeine", "Caffeine", "Caffeine", "Caf…</span>
<span class="co">#&gt; $ s1.predicate     &lt;chr&gt; "INTERACTS_WITH", "INTERACTS_WITH", "INTERACTS_WITH"…</span>
<span class="co">#&gt; $ s1.id            &lt;chr&gt; "Caffeine:INTERACTS_WITH:Melatonin", "Caffeine:INTER…</span>
<span class="co">#&gt; $ s1.object_name   &lt;chr&gt; "Melatonin", "Melatonin", "Melatonin", "Melatonin", …</span>
<span class="co">#&gt; $ st.name          &lt;chr&gt; "Melatonin", "Melatonin", "Melatonin", "Melatonin", …</span>
<span class="co">#&gt; $ st.type          &lt;chr&gt; "horm", "horm", "horm", "horm", "horm", "horm", "hor…</span>
<span class="co">#&gt; $ s2.subject_name  &lt;chr&gt; "Melatonin", "Melatonin", "Melatonin", "Melatonin", …</span>
<span class="co">#&gt; $ s2.predicate     &lt;chr&gt; "ASSOCIATED_WITH", "ASSOCIATED_WITH", "ASSOCIATED_WI…</span>
<span class="co">#&gt; $ s2.id            &lt;chr&gt; "Melatonin:ASSOCIATED_WITH:Acute coronary syndrome",…</span>
<span class="co">#&gt; $ s2.object_name   &lt;chr&gt; "Acute coronary syndrome", "Myocardial Infarction", …</span>
<span class="co">#&gt; $ gs2.localCount   &lt;int&gt; 3, 2, 3, 9, 2, 6, 4, 2, 2, 2, 3, 3, 2, 2, 2, 5, 2, 4…</span>
<span class="co">#&gt; $ gs2.pval         &lt;dbl&gt; 8.329046e-05, 1.094386e-02, 8.329046e-05, 3.435014e-…</span>
<span class="co">#&gt; $ assoc_gwas.trait &lt;chr&gt; "Coronary heart disease", "Coronary heart disease", …</span>
<span class="co">#&gt; $ assoc_gwas.id    &lt;chr&gt; "ieu-a-6", "ieu-a-6", "ieu-a-6", "ieu-a-6", "ieu-a-6…</span>
</pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># Predicate counts for SemMed triples for trait X</span>
<span class="kw">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="kw">`s1.predicate`</span>) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="kw">n</span>))
<span class="co">#&gt; # A tibble: 8 x 2</span>
<span class="co">#&gt;   s1.predicate           n</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt;</span>
<span class="co">#&gt; 1 INTERACTS_WITH       643</span>
<span class="co">#&gt; 2 INHIBITS              80</span>
<span class="co">#&gt; 3 NEG_INTERACTS_WITH    49</span>
<span class="co">#&gt; 4 STIMULATES            47</span>
<span class="co">#&gt; 5 higher_than           10</span>
<span class="co">#&gt; 6 COEXISTS_WITH          7</span>
<span class="co">#&gt; 7 NEG_INHIBITS           2</span>
<span class="co">#&gt; 8 CONVERTS_TO            1</span>
</pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># Predicate counts for SemMed triples for trait Y</span>
<span class="kw">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="kw">`s2.predicate`</span>) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="kw">n</span>))
<span class="co">#&gt; # A tibble: 15 x 2</span>
<span class="co">#&gt;    s2.predicate        n</span>
<span class="co">#&gt;    &lt;chr&gt;           &lt;int&gt;</span>
<span class="co">#&gt;  1 CAUSES            154</span>
<span class="co">#&gt;  2 STIMULATES        132</span>
<span class="co">#&gt;  3 PREVENTS          123</span>
<span class="co">#&gt;  4 PREDISPOSES       107</span>
<span class="co">#&gt;  5 AFFECTS            95</span>
<span class="co">#&gt;  6 ASSOCIATED_WITH    50</span>
<span class="co">#&gt;  7 INTERACTS_WITH     50</span>
<span class="co">#&gt;  8 TREATS             31</span>
<span class="co">#&gt;  9 DISRUPTS           30</span>
<span class="co">#&gt; 10 INHIBITS           24</span>
<span class="co">#&gt; 11 COEXISTS_WITH      21</span>
<span class="co">#&gt; 12 NEG_PREVENTS       16</span>
<span class="co">#&gt; 13 higher_than         3</span>
<span class="co">#&gt; 14 AUGMENTS            2</span>
<span class="co">#&gt; 15 same_as             1</span>
</pre></div>
<div id="filter-the-data-by-predicates" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-the-data-by-predicates" class="anchor"></a>Filter the data by predicates</h3>
<p>Sometimes it is preferable to filter the SemMedDB data, e.g. to remove less informative Predicates, such as COEXISTS_WITH and ASSOCIATED_WITH.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># Filter out some predicates that are not informative</span>
<span class="kw">pred_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"COEXISTS_WITH"</span>, <span class="st">"ASSOCIATED_WITH"</span>)
<span class="kw">lit_df_filter</span> <span class="op">&lt;-</span> <span class="kw">lit_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(
    <span class="op">!</span><span class="kw">`s1.predicate`</span> <span class="op">%in%</span> <span class="kw">pred_filter</span>,
    <span class="op">!</span><span class="kw">`s2.predicate`</span> <span class="op">%in%</span> <span class="kw">pred_filter</span>
  )
<span class="kw">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="kw">`s1.predicate`</span>) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="kw">n</span>))
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   s1.predicate           n</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt;</span>
<span class="co">#&gt; 1 INTERACTS_WITH       608</span>
<span class="co">#&gt; 2 INHIBITS              68</span>
<span class="co">#&gt; 3 NEG_INTERACTS_WITH    48</span>
<span class="co">#&gt; 4 STIMULATES            28</span>
<span class="co">#&gt; 5 higher_than            9</span>
<span class="co">#&gt; 6 CONVERTS_TO            1</span>
</pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="kw">`s2.predicate`</span>) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="kw">n</span>))
<span class="co">#&gt; # A tibble: 13 x 2</span>
<span class="co">#&gt;    s2.predicate       n</span>
<span class="co">#&gt;    &lt;chr&gt;          &lt;int&gt;</span>
<span class="co">#&gt;  1 CAUSES           154</span>
<span class="co">#&gt;  2 STIMULATES       132</span>
<span class="co">#&gt;  3 PREVENTS         123</span>
<span class="co">#&gt;  4 PREDISPOSES      107</span>
<span class="co">#&gt;  5 AFFECTS           93</span>
<span class="co">#&gt;  6 INTERACTS_WITH    50</span>
<span class="co">#&gt;  7 DISRUPTS          30</span>
<span class="co">#&gt;  8 TREATS            27</span>
<span class="co">#&gt;  9 INHIBITS          24</span>
<span class="co">#&gt; 10 NEG_PREVENTS      16</span>
<span class="co">#&gt; 11 higher_than        3</span>
<span class="co">#&gt; 12 AUGMENTS           2</span>
<span class="co">#&gt; 13 same_as            1</span>
</pre></div>
</div>
</div>
<div id="literature-results" class="section level2">
<h2 class="hasAnchor">
<a href="#literature-results" class="anchor"></a>Literature results</h2>
<p>If we explore the full table in <code>lit_df_filter</code>, we can see lots of links between the two traits, pinned on specific overlapping terms. For example:</p>
<p><code>Aspirin:INHIBITS:Oral anticoagulants</code> from Sleep duration (s1) and <code>anticoagulants:INHIBITS:P2RY12</code> from CHD (s2).</p>
<p>We can summarise the <a href="https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml">SemMedDB semantic type</a> and number of overlapping terms:</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="kw">lit_counts</span> <span class="op">&lt;-</span> <span class="kw">lit_df_filter</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span>(<span class="kw">`st.type`</span>, <span class="kw">`st.name`</span>) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="kw">`st.type`</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span>(<span class="kw">`n`</span>))
<span class="kw">lit_counts</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(n = <span class="fl">30</span>)
<span class="co">#&gt; # A tibble: 21 x 3</span>
<span class="co">#&gt;    st.type st.name                   n</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;                 &lt;int&gt;</span>
<span class="co">#&gt;  1 aapp    Leptin                    6</span>
<span class="co">#&gt;  2 aapp    Somatotropin              5</span>
<span class="co">#&gt;  3 aapp    Adrenergic Receptor       4</span>
<span class="co">#&gt;  4 aapp    apolipoprotein E-4        2</span>
<span class="co">#&gt;  5 aapp    Glutathione               2</span>
<span class="co">#&gt;  6 aapp    Monoamine Oxidase         1</span>
<span class="co">#&gt;  7 gngm    Cytochrome P450          10</span>
<span class="co">#&gt;  8 gngm    Glycoproteins             2</span>
<span class="co">#&gt;  9 horm    Melatonin                28</span>
<span class="co">#&gt; 10 horm    Hormones                  3</span>
<span class="co">#&gt; 11 orch    Ethanol                 672</span>
<span class="co">#&gt; 12 orch    Oral anticoagulants      10</span>
<span class="co">#&gt; 13 orch    Metoprolol                8</span>
<span class="co">#&gt; 14 orch    Propofol                  2</span>
<span class="co">#&gt; 15 orch    Acetaldehyde              1</span>
<span class="co">#&gt; 16 orch    Benzodiazepines           1</span>
<span class="co">#&gt; 17 orch    gamma hydroxybutyrate     1</span>
<span class="co">#&gt; 18 orch    Luteolin                  1</span>
<span class="co">#&gt; 19 orch    Morphine                  1</span>
<span class="co">#&gt; 20 orch    Neostigmine               1</span>
<span class="co">#&gt; 21 orch    Thiopental                1</span>
</pre></div>
<p>Note, the SemMedDB semantic types have been pre-filtered to only include a subset of possibilities.</p>
<p>Further examples of these term IDs and descriptions can be found here - <a href="https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml" class="uri">https://mmtx.nlm.nih.gov/MMTx/semanticTypes.shtml</a></p>
<div id="plotting-the-overlapping-term-frequencies" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-the-overlapping-term-frequencies" class="anchor"></a>Plotting the overlapping term frequencies</h3>
<p>We can also visualise the above table as a bar chart. In this case we will remove Ethanol as it is an outlier.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">lit_counts</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">n</span> <span class="op">&lt;</span> <span class="fl">100</span>) <span class="op">%&gt;%</span>
  {
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="kw">.</span>) <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(x = <span class="kw">`st.name`</span>, y = <span class="kw">`n`</span>, fill = <span class="kw">`st.type`</span>) <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span>() <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span>(
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(label = <span class="kw">`n`</span>),
        position = <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span>(<span class="fl">0.9</span>),
        hjust = <span class="fl">0</span>
      ) <span class="op">+</span>
      <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span>()
  }
</pre></div>
<p><img src="case-3-literature-triangulation_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div id="look-in-detail-at-one-overlapping-term" class="section level3">
<h3 class="hasAnchor">
<a href="#look-in-detail-at-one-overlapping-term" class="anchor"></a>Look in detail at one overlapping term</h3>
<p>Here we look at cases where <code>Leptin</code> is the central overlapping term.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="kw">focus_term</span> <span class="op">&lt;-</span> <span class="st">"Leptin"</span>
<span class="kw">lit_detail</span> <span class="op">&lt;-</span> <span class="kw">lit_df_filter</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">`st.name`</span> <span class="op">==</span> <span class="kw">focus_term</span>)
<span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>()
<span class="co">#&gt; # A tibble: 6 x 18</span>
<span class="co">#&gt;   gwas.trait gwas.id gs1.localCount gs1.pval s1.subject_name s1.predicate s1.id</span>
<span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;            &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;           &lt;chr&gt;        &lt;chr&gt;</span>
<span class="co">#&gt; 1 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; 2 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; 3 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; 4 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; 5 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; 6 Sleep dur… ieu-a-…              5 1.77e-12 ghrelin         INHIBITS     ghre…</span>
<span class="co">#&gt; # … with 11 more variables: s1.object_name &lt;chr&gt;, st.name &lt;chr&gt;, st.type &lt;chr&gt;,</span>
<span class="co">#&gt; #   s2.subject_name &lt;chr&gt;, s2.predicate &lt;chr&gt;, s2.id &lt;chr&gt;,</span>
<span class="co">#&gt; #   s2.object_name &lt;chr&gt;, gs2.localCount &lt;int&gt;, gs2.pval &lt;dbl&gt;,</span>
<span class="co">#&gt; #   assoc_gwas.trait &lt;chr&gt;, assoc_gwas.id &lt;chr&gt;</span>
</pre></div>
<p>We can create a network diagram to visualise these relationships.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">lit_detail</span> <span class="op">&lt;-</span> <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_at</a></span>(<span class="fu"><a href="https://dplyr.tidyverse.org/reference/vars.html">vars</a></span>(<span class="kw">`gwas.trait`</span>, <span class="kw">`assoc_gwas.trait`</span>), <span class="kw">stringr</span>::<span class="kw"><a href="https://stringr.tidyverse.org/reference/case.html">str_to_upper</a></span>)

<span class="co"># add node types: 1 - selected GWAS, 2 - traits from literature, 3 - current focus term connecting 1 and 2</span>
<span class="kw">nodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`gwas.trait`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">1</span>),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`assoc_gwas.trait`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">1</span>),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`s1.subject_name`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">2</span>),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`s2.subject_name`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">2</span>),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`s1.object_name`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">2</span>),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`s2.object_name`</span>) <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>() <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fl">2</span>)
) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(node_type = <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw">node</span> <span class="op">==</span> <span class="kw">focus_term</span>, <span class="fl">3</span>, <span class="kw">node_type</span>)) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>()
<span class="kw">nodes</span>
<span class="co">#&gt; # A tibble: 10 x 2</span>
<span class="co">#&gt;    node                      node_type</span>
<span class="co">#&gt;    &lt;chr&gt;                         &lt;dbl&gt;</span>
<span class="co">#&gt;  1 SLEEP DURATION                    1</span>
<span class="co">#&gt;  2 CORONARY HEART DISEASE            1</span>
<span class="co">#&gt;  3 ghrelin                           2</span>
<span class="co">#&gt;  4 Leptin                            3</span>
<span class="co">#&gt;  5 Coronary Arteriosclerosis         2</span>
<span class="co">#&gt;  6 Coronary heart disease            2</span>
<span class="co">#&gt;  7 Proteome                          2</span>
<span class="co">#&gt;  8 Sleep Apnea, Obstructive          2</span>
<span class="co">#&gt;  9 Adiponectin                       2</span>
<span class="co">#&gt; 10 Insulin                           2</span>
</pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="kw">edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(
  <span class="co"># exposure -&gt; s1 subject</span>
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`gwas.trait`</span>, assoc_node = <span class="kw">`s1.subject_name`</span>) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(),
  <span class="co"># s2 object -&gt; outcome</span>
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(node = <span class="kw">`s2.object_name`</span>, assoc_node = <span class="kw">`assoc_gwas.trait`</span>) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(),
  <span class="co"># s1 subject - s1 predicate -&gt; s1 object</span>
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(
      node = <span class="kw">`s1.subject_name`</span>, assoc_node = <span class="kw">`s1.object_name`</span>,
      label = <span class="kw">`s1.predicate`</span>
    ) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(),
  <span class="co"># s2 subject - s2 predicate -&gt; s2 object</span>
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(
      node = <span class="kw">`s2.subject_name`</span>, assoc_node = <span class="kw">`s2.object_name`</span>,
      label = <span class="kw">`s2.predicate`</span>
    ) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>()
) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>()
<span class="kw">edges</span>
<span class="co">#&gt; # A tibble: 14 x 3</span>
<span class="co">#&gt;    node                      assoc_node                label         </span>
<span class="co">#&gt;    &lt;chr&gt;                     &lt;chr&gt;                     &lt;chr&gt;         </span>
<span class="co">#&gt;  1 SLEEP DURATION            ghrelin                   &lt;NA&gt;          </span>
<span class="co">#&gt;  2 Coronary Arteriosclerosis CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  3 Coronary heart disease    CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  4 Proteome                  CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  5 Sleep Apnea, Obstructive  CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  6 Adiponectin               CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  7 Insulin                   CORONARY HEART DISEASE    &lt;NA&gt;          </span>
<span class="co">#&gt;  8 ghrelin                   Leptin                    INHIBITS      </span>
<span class="co">#&gt;  9 Leptin                    Coronary Arteriosclerosis TREATS        </span>
<span class="co">#&gt; 10 Leptin                    Coronary heart disease    PREDISPOSES   </span>
<span class="co">#&gt; 11 Leptin                    Proteome                  INTERACTS_WITH</span>
<span class="co">#&gt; 12 Leptin                    Sleep Apnea, Obstructive  TREATS        </span>
<span class="co">#&gt; 13 Leptin                    Adiponectin               STIMULATES    </span>
<span class="co">#&gt; 14 Leptin                    Insulin                   INTERACTS_WITH</span>
</pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">plot_network</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">edges</span>, <span class="kw">nodes</span>, <span class="kw">show_edge_labels</span> = <span class="fl">FALSE</span>) {

  <span class="co"># default is to not display edge labels</span>
  <span class="co">if</span> (<span class="op">!</span><span class="kw">show_edge_labels</span>) {
    <span class="kw">edges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">edges</span>, <span class="op">-</span><span class="kw">label</span>)
  }

  <span class="kw">graph</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span>(<span class="kw">edges</span>, directed = <span class="fl">TRUE</span>, vertices = <span class="kw">nodes</span>)
  <span class="kw">graph</span><span class="op">$</span><span class="kw">layout</span> <span class="op">&lt;-</span> <span class="kw">layout_with_kk</span>

  <span class="co"># generate colors based on node type</span>
  <span class="kw">colrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"tomato"</span>, <span class="st">"steelblue"</span>, <span class="st">"gold"</span>)
  <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="kw">graph</span>)<span class="op">$</span><span class="kw">color</span> <span class="op">&lt;-</span> <span class="kw">colrs</span>[<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="kw">graph</span>)<span class="op">$</span><span class="kw">node_type</span>]

  <span class="co"># Configure canvas</span>
  <span class="kw">default_mar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="st">"mar"</span>)
  <span class="kw">new_mar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>)
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mar = <span class="kw">new_mar</span>)

  <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/plot.igraph.html">plot.igraph</a></span>(<span class="kw">graph</span>,
    vertex.size = <span class="fl">13</span>,
    vertex.label.color = <span class="st">"black"</span>,
    vertex.label.family = <span class="st">"Helvetica"</span>,
    vertex.label.cex = <span class="fl">0.8</span>,
    edge.arrow.size = <span class="fl">0.4</span>,
    edge.label.color = <span class="st">"black"</span>,
    edge.label.family = <span class="st">"Helvetica"</span>,
    edge.label.cex = <span class="fl">0.5</span>
  )
  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(mar = <span class="kw">default_mar</span>)
}
</pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="fu">plot_network</span>(<span class="kw">edges</span>, <span class="kw">nodes</span>, show_edge_labels = <span class="fl">TRUE</span>)
</pre></div>
<p><img src="case-3-literature-triangulation_files/figure-html/unnamed-chunk-17-1.png" width="2187.5"></p>
</div>
</div>
<div id="checking-the-source-literature" class="section level2">
<h2 class="hasAnchor">
<a href="#checking-the-source-literature" class="anchor"></a>Checking the source literature</h2>
<p>We can refer back to the articles to check the text that was used to derive the SemMedDB data. This is important due to the imperfect nature of the SemRep annotation process (<a href="https://semrep.nlm.nih.gov/" class="uri">https://semrep.nlm.nih.gov/</a>).</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="kw">get_literature</span> <span class="op">&lt;-</span> <span class="fu">function</span>(<span class="kw">gwas_id</span>, <span class="kw">semmed_triple_id</span>) {
  <span class="kw">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/literature/gwas"</span>
  <span class="kw">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(
    gwas_id = <span class="kw">gwas_id</span>,
    semmed_triple_id = <span class="kw">semmed_triple_id</span>,
    by_gwas_id = <span class="fl">TRUE</span>,
    pval_threshold = <span class="fl">1e-1</span>
  )
  <span class="kw">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span>(route = <span class="kw">endpoint</span>, params = <span class="kw">params</span>, mode = <span class="st">"table"</span>)
  <span class="kw">df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="kw">`triple.id`</span>, <span class="kw">`lit.pubmed_id`</span>)
}

<span class="kw">pub_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span>(
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(gwas_id = <span class="kw">`gwas.id`</span>, semmed_triple_id = <span class="kw">`s1.id`</span>) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>(),
  <span class="kw">lit_detail</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(gwas_id = <span class="kw">`assoc_gwas.id`</span>, semmed_triple_id = <span class="kw">`s2.id`</span>) <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span>()
) <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/transpose.html">transpose</a></span>() <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span>(<span class="fu">function</span>(<span class="kw">x</span>) <span class="fu">get_literature</span>(<span class="kw">x</span><span class="op">$</span><span class="kw">gwas_id</span>, <span class="kw">x</span><span class="op">$</span><span class="kw">semmed_triple_id</span>))
<span class="kw">pub_df</span>
<span class="co">#&gt; # A tibble: 15 x 2</span>
<span class="co">#&gt;    triple.id                                 lit.pubmed_id</span>
<span class="co">#&gt;    &lt;chr&gt;                                     &lt;chr&gt;        </span>
<span class="co">#&gt;  1 ghrelin:INHIBITS:Leptin                   21659802     </span>
<span class="co">#&gt;  2 ghrelin:INHIBITS:Leptin                   22473743     </span>
<span class="co">#&gt;  3 ghrelin:INHIBITS:Leptin                   19955752     </span>
<span class="co">#&gt;  4 ghrelin:INHIBITS:Leptin                   18719052     </span>
<span class="co">#&gt;  5 ghrelin:INHIBITS:Leptin                   30364557     </span>
<span class="co">#&gt;  6 Leptin:TREATS:Coronary Arteriosclerosis   26769430     </span>
<span class="co">#&gt;  7 Leptin:TREATS:Coronary Arteriosclerosis   20691218     </span>
<span class="co">#&gt;  8 Leptin:PREDISPOSES:Coronary heart disease 18723235     </span>
<span class="co">#&gt;  9 Leptin:PREDISPOSES:Coronary heart disease 22412070     </span>
<span class="co">#&gt; 10 Leptin:INTERACTS_WITH:Proteome            26975316     </span>
<span class="co">#&gt; 11 Leptin:TREATS:Sleep Apnea, Obstructive    10449691     </span>
<span class="co">#&gt; 12 Leptin:STIMULATES:Adiponectin             17403719     </span>
<span class="co">#&gt; 13 Leptin:STIMULATES:Adiponectin             21481397     </span>
<span class="co">#&gt; 14 Leptin:INTERACTS_WITH:Insulin             12724058     </span>
<span class="co">#&gt; 15 Leptin:INTERACTS_WITH:Insulin             15648007</span>
</pre></div>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 4.0.2 (2020-06-22)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 16.04.6 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/openblas-base/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/libopenblasp-r0.2.18.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2.1 igraph_1.2.5     ggplot2_3.3.2    glue_1.4.2      </span>
<span class="co">#&gt; [5] purrr_0.3.4      dplyr_1.0.2      magrittr_1.5    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] cpp11_0.2.1        compiler_4.0.2     pillar_1.4.6       tools_4.0.2       </span>
<span class="co">#&gt;  [5] digest_0.6.25      jsonlite_1.7.1     evaluate_0.14      memoise_1.1.0     </span>
<span class="co">#&gt;  [9] lifecycle_0.2.0    tibble_3.0.3       gtable_0.3.0       pkgconfig_2.0.3   </span>
<span class="co">#&gt; [13] rlang_0.4.7        cli_2.0.2          curl_4.3           yaml_2.2.1        </span>
<span class="co">#&gt; [17] pkgdown_1.6.1.9000 xfun_0.17          withr_2.2.0        stringr_1.4.0     </span>
<span class="co">#&gt; [21] httr_1.4.2         knitr_1.29         generics_0.0.2     desc_1.2.0        </span>
<span class="co">#&gt; [25] fs_1.5.0           vctrs_0.3.4        systemfonts_0.3.1  rprojroot_1.3-2   </span>
<span class="co">#&gt; [29] grid_4.0.2         tidyselect_1.1.0   R6_2.4.1           fansi_0.4.1       </span>
<span class="co">#&gt; [33] rmarkdown_2.3      farver_2.0.3       backports_1.1.10   scales_1.1.1      </span>
<span class="co">#&gt; [37] htmltools_0.5.0    ellipsis_0.3.1     assertthat_0.2.1   colorspace_1.4-1  </span>
<span class="co">#&gt; [41] labeling_0.3       ragg_0.3.1         utf8_1.1.4         stringi_1.5.3     </span>
<span class="co">#&gt; [45] munsell_0.5.0      crayon_1.3.4</span>
</pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
