<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started with EpiGraphDB in R • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting started with EpiGraphDB in R">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started with EpiGraphDB in R</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/getting-started-with-epigraphdb-r.Rmd"><code>vignettes/getting-started-with-epigraphdb-r.Rmd</code></a></small>
      <div class="hidden name"><code>getting-started-with-epigraphdb-r.Rmd</code></div>

    </div>

    
    
<p>This article is provided as a brief introductory guide to working with the EpiGraphDB platform through <code>epigraphdb</code> R package. Here we will demonstrate a few basic operations that can be carried out using the platform, but for more advanced methods please refer to the <a href="https://docs.epigraphdb.org/api/api-endpoints/">API endpoint documentation</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co">#install.packages("epigraphdb")</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://mrcieu.github.io/epigraphdb-r/">epigraphdb</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></pre></div>
<div id="part-1-using-epigraphdb-to-obtain-biological-mappings" class="section level2">
<h2 class="hasAnchor">
<a href="#part-1-using-epigraphdb-to-obtain-biological-mappings" class="anchor"></a>Part 1: Using EpiGraphDB to obtain biological mappings</h2>
<p>With EpiGraphDB you can map genetic variants to genes, genes to proteins, proteins to pathways, pathways to diseases and so on, as shown in the network diagram <a href="https://epigraphdb.org/about">here</a>.</p>
<p>In this part, we want to demonstrate how to do basic mappings between biological entities. We are going to map genes to proteins (i.e. their UniProtID), proteins to pathways that they are found in (using Reactome data), and then extract information on the specific pathways identified.</p>
<p>But first, let’s talk about the basic querying syntax. <code>query_epigraphdb</code> is the main querying function in the package; it is used to communicate with EpiGraphDB by specifying API endpoints.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, <span class="co"># supply the endpoint (see below)</span>
                 params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>... <span class="op">=</span> <span class="va">...</span><span class="op">)</span>, <span class="co"># supply the query parameters </span>
                 mode <span class="op">=</span> <span class="st">"table"</span>, <span class="co"># mode</span>
                 method <span class="op">=</span> <span class="st">"POST"</span><span class="op">)</span> <span class="co"># method (ignore this for now)</span></pre></div>
<ul>
<li><p><code>endpoint</code> is the ‘address’ within EpiGraphDB of where you want to query the data from, and there is a specific one for each case. e.g. <code>"/mappings/gene-to-protein"</code> is used to query gene-to-proteins mappings. See the full list of currently available endpoints <a href="https://docs.epigraphdb.org/api/api-endpoints/">here</a></p></li>
<li><p><code>params</code> is where you supply the list of things you want to query about (can be a list of genes, GWAS trait, MR outcome of interest)</p></li>
<li><p><code>mode</code> is the format in which the results are returned. Use “table” for basic cases, for more advanced stuff you may need to use “raw” - more on this later.</p></li>
</ul>
<p><br> In this guide, we are going to use the all-purpose <code>query_epigraphdb</code> function in all basic examples. However, many of the most common queries have been wrapped in <a href="https://mrcieu.github.io/epigraphdb-r/index.html#package-functionalities">specific functions</a> within <code>epigraphdb</code> R package for the ease of use. Those are very helpful, but to help the understanding of the core principles behind using EpiGraphDB, here we present the ways to run queries in a less abstracted way.</p>
<div id="mapping-genes-to-proteins" class="section level3">
<h3 class="hasAnchor">
<a href="#mapping-genes-to-proteins" class="anchor"></a>Mapping genes to proteins</h3>
<p>In this first section, we will take an arbitrary list of genes and query the EpiGraphDB API to find the proteins that they map to.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">

<span class="co"># Let's test this on a few genes</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"BRCA1"</span>, <span class="st">"TNF"</span><span class="op">)</span> 

<span class="co"># Select the endpoint</span>
<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mappings/gene-to-protein"</span>

<span class="co"># Build a query</span>
<span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>,
                params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene_name_list <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>,
                mode <span class="op">=</span> <span class="st">"table"</span>,
                method <span class="op">=</span> <span class="st">"POST"</span><span class="op">)</span>

<span class="co"># Check the output</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">proteins</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 3 x 3</span>
<span class="co">#&gt;   gene.name gene.ensembl_id protein.uniprot_id</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;             </span>
<span class="co">#&gt; 1 TP53      ENSG00000141510 P04637            </span>
<span class="co">#&gt; 2 BRCA1     ENSG00000012048 P38398            </span>
<span class="co">#&gt; 3 TNF       ENSG00000232810 P01375</span></pre></div>
<p>In the above data frame, we see the output from querying EpiGraphDB for the proteins that have been mapped to the genes TP53, BRCA1, and TNF. Our query returned the UniProt and Ensembl IDs for those genes.</p>
</div>
<div id="mapping-proteins-to-pathways" class="section level3">
<h3 class="hasAnchor">
<a href="#mapping-proteins-to-pathways" class="anchor"></a>Mapping proteins to pathways</h3>
<p>Next, to demonstrate the mapping of proteins to pathways, we are going to take one protein from the previous example, P04637, and query EpiGraphDB for all pathways it is known to be involved in.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># Let's see what pathways the protein product of TP53 gene is involved in </span>

<span class="va">proteins_uniprot_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"P04637"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># use I()/(AsIs) function when providing a list of one</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/protein/in-pathway"</span>

<span class="va">pathway_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                  route <span class="op">=</span> <span class="va">endpoint</span>,
                  params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>uniprot_id_list <span class="op">=</span> <span class="va">proteins_uniprot_ids</span><span class="op">)</span>,
                  mode <span class="op">=</span> <span class="st">"table"</span>, 
                  method <span class="op">=</span> <span class="st">"POST"</span><span class="op">)</span>

<span class="co"># Check out how many pathways this protein is found in</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">pathway_df</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 3</span>
<span class="co">#&gt;   uniprot_id pathway_count pathway_reactome_id</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt; &lt;list&gt;             </span>
<span class="co">#&gt; 1 P04637                 5 &lt;chr [5]&gt;</span>

<span class="co"># Get pathways names (Reactome IDs)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">pathway_df</span><span class="op">$</span><span class="va">pathway_reactome_id</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "R-HSA-6785807" "R-HSA-390471"  "R-HSA-5689896" "R-HSA-6804754"</span>
<span class="co">#&gt; [5] "R-HSA-3232118"</span></pre></div>
<p>P04637 is involved in five pathways. Next, let’s get pathway info for one of them.</p>
</div>
<div id="get-pathway-info" class="section level3">
<h3 class="hasAnchor">
<a href="#get-pathway-info" class="anchor"></a>Get pathway info</h3>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co"># Let's see what exactly this pathway is</span>
<span class="va">reactome_id</span> <span class="op">=</span> <span class="st">"R-HSA-6804754"</span>

<span class="va">endpoint</span> <span class="op">=</span> <span class="st">"/meta/nodes/Pathway/search"</span>

<span class="va">pathway_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                    route <span class="op">=</span> <span class="va">endpoint</span>,
                    params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>id <span class="op">=</span> <span class="va">reactome_id</span><span class="op">)</span>,
                    mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>

<span class="co"># Pathway description</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">pathway_info</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 3</span>
<span class="co">#&gt;   node.in_disease node.name                     node.reactome_id</span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;                         &lt;chr&gt;           </span>
<span class="co">#&gt; 1 FALSE           Regulation of TP53 Expression R-HSA-6804754</span></pre></div>
<p>If you are interested in this type of analysis, check out Case studies <a href="https://mrcieu.github.io/epigraphdb-r/articles/case-1-pleiotropy.html">1</a> and <a href="https://mrcieu.github.io/epigraphdb-r/articles/case-2-alt-drug-target.html">2</a> for further details on pathways analysis, PPI, mapping drugs to targets etc.</p>
<p><br><em>Running the above queries using the <a href="https://mrcieu.github.io/epigraphdb-r/index.html#package-functionalities">dedicated wrapped functions</a>:</em></p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu"><a href="../reference/mappings_gene_to_protein.html">mappings_gene_to_protein</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 3 x 3</span>
<span class="co">#&gt;   gene.name gene.ensembl_id protein.uniprot_id</span>
<span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;             </span>
<span class="co">#&gt; 1 TP53      ENSG00000141510 P04637            </span>
<span class="co">#&gt; 2 BRCA1     ENSG00000012048 P38398            </span>
<span class="co">#&gt; 3 TNF       ENSG00000232810 P01375</span>
<span class="fu"><a href="../reference/protein_in_pathway.html">protein_in_pathway</a></span><span class="op">(</span><span class="va">proteins_uniprot_ids</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 3</span>
<span class="co">#&gt;   uniprot_id pathway_count pathway_reactome_id</span>
<span class="co">#&gt;   &lt;chr&gt;              &lt;int&gt; &lt;list&gt;             </span>
<span class="co">#&gt; 1 P04637                 5 &lt;chr [5]&gt;</span></pre></div>
</div>
</div>
<div id="part-2-epidemiological-relationships-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#part-2-epidemiological-relationships-analysis" class="anchor"></a>Part 2: Epidemiological relationships analysis</h2>
<p>In this part we will demonstrate queries that may be relevant in epidemiology research.</p>
<div id="look-up-gwas-studies" class="section level3">
<h3 class="hasAnchor">
<a href="#look-up-gwas-studies" class="anchor"></a>Look up GWAS studies</h3>
<p>First, we want to check what GWAS are available within EpiGraphDB for our trait of interest, e.g. Body mass index. Doing this query is equivalent doing a look-up using <a href="https://dev.epigraphdb.org/explore">EpiGraphDB Web UI</a>. The fuzzy matching is supported, i.e. ‘body mass index’ or ‘Body Mass Index’ will give you the same set of results.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># Let's see what Body mass index GWAS are available in EpiGraphDB</span>
<span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"body mass index"</span> 

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Gwas/search"</span>

<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
              route <span class="op">=</span> <span class="va">endpoint</span>, 
              params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">trait</span><span class="op">)</span>,
              mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>

<span class="co"># show selected columns in the results</span>
<span class="va">results</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.trait</span>, <span class="va">node.id</span>, <span class="va">node.sample_size</span>,
         <span class="va">node.year</span>, <span class="va">node.author</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 10 x 5</span>
<span class="co">#&gt;    node.trait            node.id          node.sample_size node.year node.author</span>
<span class="co">#&gt;    &lt;chr&gt;                 &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 Body mass index       ebi-a-GCST006368 315347           2018      Hoffmann TJ</span>
<span class="co">#&gt;  2 Body mass index       ieu-a-2          339224           2015      Locke AE   </span>
<span class="co">#&gt;  3 Body mass index       ieu-a-785        152893           2015      Locke AE   </span>
<span class="co">#&gt;  4 Body mass index       ieu-a-835        322154           2015      Locke AE   </span>
<span class="co">#&gt;  5 Body mass index       ebi-a-GCST004904 158284           2017      Akiyama M  </span>
<span class="co">#&gt;  6 Body mass index       ieu-a-1089       120286           2016      Wood       </span>
<span class="co">#&gt;  7 Body mass index       ieu-a-974        171977           2015      Locke AE   </span>
<span class="co">#&gt;  8 Body mass index       ieu-a-95         73137            2013      Randall JC </span>
<span class="co">#&gt;  9 Body mass index       ieu-a-94         60586            2013      Randall JC </span>
<span class="co">#&gt; 10 Body mass index (BMI) ukb-b-19953      461460           2018      Ben Elswor…</span></pre></div>
</div>
<div id="explore-mendelian-randomization-studies" class="section level3">
<h3 class="hasAnchor">
<a href="#explore-mendelian-randomization-studies" class="anchor"></a>Explore Mendelian randomization studies</h3>
<p>In these examples, we show how to extract <a href="https://epigraphdb.org/mr">pre-computed MR results</a> for the specified exposure, outcome, or both, traits of interest.</p>
<div id="specify-exposure-trait" class="section level4">
<h4 class="hasAnchor">
<a href="#specify-exposure-trait" class="anchor"></a>Specify exposure trait</h4>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="co"># Look up all MR analyses where a trait was used as exposure</span>
<span class="co"># and find all outcome traits with causal evidence from it.</span>

<span class="va">trait1</span> <span class="op">&lt;-</span> <span class="st">"Body mass index"</span> 
<span class="co"># NB: here, trait name has to specific (not fuzzy):</span>
<span class="co"># use the exact trait name wording as in GWAS `node.trait` (previous example)</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>

<span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, 
                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>exposure_trait <span class="op">=</span> <span class="va">trait1</span>,
                                        pval_threshold <span class="op">=</span> <span class="fl">5e-08</span> <span class="op">)</span>,
                          mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">mr_df</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 633 x 10</span>
<span class="co">#&gt;    exposure.id exposure.trait outcome.id outcome.trait   mr.b   mr.se mr.pval</span>
<span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 ieu-a-2     Body mass ind… ukb-a-74   Non-cancer i… 0.0346 0.00242       0</span>
<span class="co">#&gt;  2 ieu-a-2     Body mass ind… ukb-a-388  Hip circumfe… 0.724  0.0266        0</span>
<span class="co">#&gt;  3 ieu-a-2     Body mass ind… ukb-a-382  Waist circum… 0.656  0.0245        0</span>
<span class="co">#&gt;  4 ieu-a-2     Body mass ind… ukb-a-35   Comparative … 0.137  0.00791       0</span>
<span class="co">#&gt;  5 ieu-a-2     Body mass ind… ukb-a-34   Comparative … 0.366  0.0236        0</span>
<span class="co">#&gt;  6 ieu-a-2     Body mass ind… ukb-a-293  Trunk predic… 0.358  0.0167        0</span>
<span class="co">#&gt;  7 ieu-a-2     Body mass ind… ukb-a-292  Trunk fat-fr… 0.360  0.0177        0</span>
<span class="co">#&gt;  8 ieu-a-2     Body mass ind… ukb-a-291  Trunk fat ma… 0.737  0.0280        0</span>
<span class="co">#&gt;  9 ieu-a-2     Body mass ind… ukb-a-290  Trunk fat pe… 0.527  0.0275        0</span>
<span class="co">#&gt; 10 ieu-a-2     Body mass ind… ukb-a-289  Arm predicte… 0.416  0.0177        0</span>
<span class="co">#&gt; # … with 623 more rows, and 3 more variables: mr.method &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.selection &lt;chr&gt;, mr.moescore &lt;dbl&gt;</span></pre></div>
<p>The returned data frame includes all MR analysis with <code>exposure.trait</code> being “Body mass index”. However, there are several GWAS with this names. If you are interested in a specific GWAS, you will need to filter by <code>exposure.id</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="co"># Show how many MR analyses were done for each Body mass index GWAS</span>
<span class="va">mr_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">exposure.id</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   exposure.id     n</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;int&gt;</span>
<span class="co">#&gt; 1 ieu-a-2       146</span>
<span class="co">#&gt; 2 ieu-a-785      90</span>
<span class="co">#&gt; 3 ieu-a-835     129</span>
<span class="co">#&gt; 4 ieu-a-94       69</span>
<span class="co">#&gt; 5 ieu-a-95       86</span>
<span class="co">#&gt; 6 ieu-a-974     113</span></pre></div>
</div>
<div id="specify-outcome-trait" class="section level4">
<h4 class="hasAnchor">
<a href="#specify-outcome-trait" class="anchor"></a>Specify outcome trait</h4>
<p>Next, we can check all available MR analyses for an outcome trait of interest.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="co"># Look up all MR for a specified outcome trait</span>
<span class="co"># and find all exposure traits with causal evidence on it.</span>

<span class="va">trait2</span> <span class="op">&lt;-</span> <span class="st">"Waist circumference"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>

<span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, 
                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>outcome_trait <span class="op">=</span> <span class="va">trait2</span>,
                                        pval_threshold <span class="op">=</span> <span class="fl">5e-08</span><span class="op">)</span>,
                          mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">mr_df</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 842 x 10</span>
<span class="co">#&gt;    exposure.id exposure.trait outcome.id outcome.trait     mr.b   mr.se mr.pval</span>
<span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 ieu-a-96    Height         ieu-a-67   Waist circum…   0.416  1.81e-2       0</span>
<span class="co">#&gt;  2 ieu-a-54    Hip circumfer… ieu-a-67   Waist circum…   0.461  1.82e-2       0</span>
<span class="co">#&gt;  3 ieu-a-69    Waist circumf… ieu-a-67   Waist circum…   0.625  4.01e-2       0</span>
<span class="co">#&gt;  4 ukb-a-185   Treatment/med… ieu-a-67   Waist circum…  26.8    1.40e+0       0</span>
<span class="co">#&gt;  5 ieu-a-89    Height         ieu-a-67   Waist circum…   0.402  9.41e-3       0</span>
<span class="co">#&gt;  6 ukb-a-485   Types of phys… ieu-a-67   Waist circum…  -1.82   6.62e-2       0</span>
<span class="co">#&gt;  7 ieu-a-67    Waist circumf… ieu-a-69   Waist circum…   0.840  5.03e-2       0</span>
<span class="co">#&gt;  8 ieu-a-70    Waist circumf… ieu-a-69   Waist circum…   0.541  3.41e-2       0</span>
<span class="co">#&gt;  9 ieu-a-87    Extreme waist… ieu-a-69   Waist circum…   0.0950 2.50e-4       0</span>
<span class="co">#&gt; 10 ukb-a-523   Diagnoses - m… ieu-a-69   Waist circum… -12.5    2.97e-1       0</span>
<span class="co">#&gt; # … with 832 more rows, and 3 more variables: mr.method &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.selection &lt;chr&gt;, mr.moescore &lt;dbl&gt;</span></pre></div>
</div>
<div id="specify-both-exposure-and-outcome-traits" class="section level4">
<h4 class="hasAnchor">
<a href="#specify-both-exposure-and-outcome-traits" class="anchor"></a>Specify both exposure and outcome traits</h4>
<p>Finally, we can look up MR causal inference results for a pair of exposure and outcome.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="co"># Look up a specific pair of exposure+outcome</span>
<span class="va">trait1</span> <span class="op">&lt;-</span> <span class="st">"Body mass index"</span>
<span class="va">trait2</span> <span class="op">&lt;-</span> <span class="st">"Coronary heart disease"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>

<span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
              route <span class="op">=</span> <span class="va">endpoint</span>,
              params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>exposure_trait <span class="op">=</span> <span class="va">trait1</span>,
                            outcome_trait <span class="op">=</span> <span class="va">trait2</span><span class="op">)</span>,
              mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">mr_df</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 10</span>
<span class="co">#&gt;   exposure.id exposure.trait outcome.id outcome.trait  mr.b  mr.se  mr.pval</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ieu-a-974   Body mass ind… ieu-a-7    Coronary hea… 0.389 0.0493 3.42e-15</span>
<span class="co">#&gt; 2 ieu-a-2     Body mass ind… ieu-a-7    Coronary hea… 0.397 0.0727 4.79e- 8</span>
<span class="co">#&gt; 3 ieu-a-95    Body mass ind… ieu-a-7    Coronary hea… 0.455 0.0931 1.02e- 6</span>
<span class="co">#&gt; 4 ieu-a-2     Body mass ind… ieu-a-8    Coronary hea… 0.331 0.0684 1.31e- 6</span>
<span class="co">#&gt; 5 ieu-a-835   Body mass ind… ieu-a-7    Coronary hea… 0.360 0.0756 1.94e- 6</span>
<span class="co">#&gt; 6 ieu-a-974   Body mass ind… ieu-a-8    Coronary hea… 0.328 0.0731 6.97e- 6</span>
<span class="co">#&gt; # … with 3 more variables: mr.method &lt;chr&gt;, mr.selection &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.moescore &lt;dbl&gt;</span></pre></div>
<p>To query EpiGraphDB directly by GWAS ID, you will need to use the advanced functionality. See the end of this article.</p>
<p><br><em>Running the above MR query using a <a href="https://mrcieu.github.io/epigraphdb-r/index.html#package-functionalities">dedicated wrapped function</a>:</em></p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu"><a href="../reference/mr.html">mr</a></span><span class="op">(</span>exposure_trait <span class="op">=</span> <span class="va">trait1</span>,
    outcome_trait <span class="op">=</span> <span class="va">trait2</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 10</span>
<span class="co">#&gt;   exposure.id exposure.trait outcome.id outcome.trait  mr.b  mr.se  mr.pval</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;          &lt;chr&gt;      &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;</span>
<span class="co">#&gt; 1 ieu-a-974   Body mass ind… ieu-a-7    Coronary hea… 0.389 0.0493 3.42e-15</span>
<span class="co">#&gt; 2 ieu-a-2     Body mass ind… ieu-a-7    Coronary hea… 0.397 0.0727 4.79e- 8</span>
<span class="co">#&gt; 3 ieu-a-95    Body mass ind… ieu-a-7    Coronary hea… 0.455 0.0931 1.02e- 6</span>
<span class="co">#&gt; 4 ieu-a-2     Body mass ind… ieu-a-8    Coronary hea… 0.331 0.0684 1.31e- 6</span>
<span class="co">#&gt; 5 ieu-a-835   Body mass ind… ieu-a-7    Coronary hea… 0.360 0.0756 1.94e- 6</span>
<span class="co">#&gt; 6 ieu-a-974   Body mass ind… ieu-a-8    Coronary hea… 0.328 0.0731 6.97e- 6</span>
<span class="co">#&gt; # … with 3 more variables: mr.method &lt;chr&gt;, mr.selection &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.moescore &lt;dbl&gt;</span></pre></div>
</div>
</div>
</div>
<div id="part-3--looking-for-literature-evidence" class="section level2">
<h2 class="hasAnchor">
<a href="#part-3--looking-for-literature-evidence" class="anchor"></a>Part 3. Looking for literature evidence</h2>
<p>Accessing information in the literature is a ubiquitous task in research, be it for novel hypothesis generation or as part of evidence triangulation. EpiGraphDB facilitates fast processing of this information by allowing access to a host of literature-mined relationships that have been structured into semantic triples. These take the general form (subject, predicate, object) and have been generated using contemporary natural language processing techniques applied to a massive amount of published biomedical research papers.</p>
<p>In the following section, we will query the API for the relationship between a given gene and a disease outcome.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># Identity all publications where a gene is mentioned with relation to a disease</span>

<span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"IL23R"</span>
<span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"Inflammatory bowel disease"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/gene/literature"</span>


<span class="va">lit_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>, 
                params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>gene_name <span class="op">=</span> <span class="va">gene</span>,
                             object_name <span class="op">=</span> <span class="va">trait</span><span class="op">)</span>, 
                mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>


<span class="co"># Review the found evidence in the literature</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">lit_df</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 4 x 4</span>
<span class="co">#&gt;   pubmed_id  gene.name st.predicate        st.object_name             </span>
<span class="co">#&gt;   &lt;list&gt;     &lt;chr&gt;     &lt;chr&gt;               &lt;chr&gt;                      </span>
<span class="co">#&gt; 1 &lt;chr [2]&gt;  IL23R     NEG_ASSOCIATED_WITH Inflammatory Bowel Diseases</span>
<span class="co">#&gt; 2 &lt;chr [1]&gt;  IL23R     AFFECTS             Inflammatory Bowel Diseases</span>
<span class="co">#&gt; 3 &lt;chr [21]&gt; IL23R     ASSOCIATED_WITH     Inflammatory Bowel Diseases</span>
<span class="co">#&gt; 4 &lt;chr [1]&gt;  IL23R     PREDISPOSES         Inflammatory Bowel Diseases</span></pre></div>
<p>The data frame above shows that IL23R has been mentioned in 25 publications (<code>pubmed_id</code> column) in relation to Inflammatory bowel disease, in four predicates.</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># Get a list of all PubMed IDs</span>
<span class="va">lit_df</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">pubmed_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "17484863" "21155887" "27852544" "19575361" "19496308" "18383521"</span>
<span class="co">#&gt;  [7] "18341487" "19817673" "18088064" "29248579" "18047540" "25159710"</span>
<span class="co">#&gt; [13] "18383363" "19175939" "18698678" "21846945" "20067801" "18368064"</span>
<span class="co">#&gt; [19] "20393462" "19747142" "18164077" "24280935" "23131344"</span></pre></div>
<p>If you are interested in literature mining analysis, and also matching MR results to literature evidence, please refer to more specific examples in case studies <a href="https://mrcieu.github.io/epigraphdb-r/articles/case-3-literature-triangulation.html">3</a> and <a href="https://mrcieu.github.io/epigraphdb-r/articles/case-2-alt-drug-target.html">2</a>.</p>
</div>
<div id="epigraphdb-node-search" class="section level2">
<h2 class="hasAnchor">
<a href="#epigraphdb-node-search" class="anchor"></a>EpiGraphDB node search</h2>
<p>EpiGraphDB stores data as nodes (data types) and edges (relationships between nodes). The available nodes can be viewed through the <code>meta/nodes</code> endpoint. Let’s list the available nodes:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/list"</span>
<span class="va">meta_node_fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span>
<span class="va">meta_node_fields</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Gwas"       "Disease"    "Drug"       "Efo"        "Event"     </span>
<span class="co">#&gt;  [6] "Gene"       "Tissue"     "Literature" "Pathway"    "Protein"   </span>
<span class="co">#&gt; [11] "SemmedTerm" "Variant"</span></pre></div>
<p>The list of nodes is also available in the <a href="https://docs.epigraphdb.org/graph-database/meta-nodes/">documentation</a>, along with the node properties, which are useful for running native Cypher queries. In this section, we want to show how to perform node search for a term of interest. Any node can be searched by specifying it in this query: <code>/meta/nodes/{meta_node}/search</code>.</p>
<p>To demonstrate this, we are going to search for ‘breast cancer’ in various nodes:</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"breast cancer"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Gwas/search"</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>, 
                params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>, 
                mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="va">results</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.trait</span>, <span class="va">node.id</span>, <span class="va">node.sample_size</span>,
         <span class="va">node.year</span>, <span class="va">node.author</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 10 x 5</span>
<span class="co">#&gt;    node.trait                  node.id    node.sample_size node.year node.author</span>
<span class="co">#&gt;    &lt;chr&gt;                       &lt;chr&gt;      &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 Breast cancer               ebi-a-GCS… 89677            2015      Michailido…</span>
<span class="co">#&gt;  2 Breast cancer               ebi-a-GCS… 139274           2017      Michailido…</span>
<span class="co">#&gt;  3 Breast cancer (Combined On… ieu-a-1126 228951           2017      Michailido…</span>
<span class="co">#&gt;  4 Breast cancer (GWAS)        ieu-a-1131 32498            2017      Michailido…</span>
<span class="co">#&gt;  5 Breast cancer (GWAS)        ieu-a-1168 33832            2015      Michailido…</span>
<span class="co">#&gt;  6 Breast cancer (Oncoarray)   ieu-a-1129 106776           2017      Michailido…</span>
<span class="co">#&gt;  7 Breast cancer (Survival)    ieu-a-1165 37954            2015      Guo Q      </span>
<span class="co">#&gt;  8 Breast cancer (iCOGS)       ieu-a-1162 89677            2015      Michailido…</span>
<span class="co">#&gt;  9 Breast cancer (iCOGS)       ieu-a-1130 89677            2017      Michailido…</span>
<span class="co">#&gt; 10 Breast cancer anti-estroge… prot-a-234 3301             2018      Sun BB</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Disease/search"</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>, 
                params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>, 
                mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="va">results</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.label</span>, <span class="va">node.id</span>, <span class="va">node.definition</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 3</span>
<span class="co">#&gt;   node.label             node.id             node.definition                    </span>
<span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;               &lt;chr&gt;                              </span>
<span class="co">#&gt; 1 sporadic breast cancer http://purl.obolib… A carcinoma that arises from the b…</span>
<span class="co">#&gt; 2 progesterone-receptor… http://purl.obolib… NA                                 </span>
<span class="co">#&gt; 3 progesterone-receptor… http://purl.obolib… NA                                 </span>
<span class="co">#&gt; 4 Her2-receptor negativ… http://purl.obolib… NA                                 </span>
<span class="co">#&gt; 5 estrogen-receptor pos… http://purl.obolib… a subtype of breast cancer that is…</span>
<span class="co">#&gt; 6 estrogen-receptor neg… http://purl.obolib… a subtype of breast cancer that is…</span>
<span class="co">#&gt; 7 breast cancer          http://purl.obolib… A primary or metastatic malignant …</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Efo/search"</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>, 
                params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>, 
                mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="va">results</span>
<span class="co">#&gt; # A tibble: 10 x 3</span>
<span class="co">#&gt;    node.type    node.value                       node.id                        </span>
<span class="co">#&gt;    &lt;chr&gt;        &lt;chr&gt;                            &lt;chr&gt;                          </span>
<span class="co">#&gt;  1 typed-liter… breast cancer cell line          http://www.ebi.ac.uk/efo/EFO_0…</span>
<span class="co">#&gt;  2 typed-liter… triple-negative breast cancer    http://www.ebi.ac.uk/efo/EFO_0…</span>
<span class="co">#&gt;  3 typed-liter… Her2-receptor negative breast c… http://purl.obolibrary.org/obo…</span>
<span class="co">#&gt;  4 typed-liter… breast cancer                    http://purl.obolibrary.org/obo…</span>
<span class="co">#&gt;  5 typed-liter… progesterone-receptor negative … http://www.ebi.ac.uk/efo/EFO_0…</span>
<span class="co">#&gt;  6 typed-liter… estrogen-receptor negative brea… http://www.ebi.ac.uk/efo/EFO_1…</span>
<span class="co">#&gt;  7 typed-liter… Hereditary breast cancer         http://www.orpha.net/ORDO/Orph…</span>
<span class="co">#&gt;  8 typed-liter… age at breast cancer diagnosis   http://www.ebi.ac.uk/efo/EFO_0…</span>
<span class="co">#&gt;  9 typed-liter… breast cancer stage              http://www.ebi.ac.uk/efo/EFO_0…</span>
<span class="co">#&gt; 10 typed-liter… family history of breast cancer  http://www.ebi.ac.uk/efo/EFO_0…</span></pre></div>
<p>The queries above can be further expanded using Cypher as will be shown in the next section. To get more information about <code>meta</code> endpoint please see the guidelines on <a href="https://mrcieu.github.io/epigraphdb-r/articles/meta-functionalities.html">meta functionalities</a>.</p>
</div>
<div id="advanced-examples" class="section level2">
<h2 class="hasAnchor">
<a href="#advanced-examples" class="anchor"></a>Advanced examples</h2>
<p>The functionality of <code>epigraphdb</code> R package is currently limited to a certain number of endpoints available via the <code>query_epigraphdb</code> function. If you would like to access one of the endpoints that are currently <a href="https://docs.epigraphdb.org/api/api-endpoints/">not directly available</a> through it, we recommend exploring Neo4j Cypher query language to construct native EpiGraphDB queries.</p>
<p>To get you started, we want to show that all available <code>query_epigraphdb</code> endpoint queries are actually internally Cypher queries. For example, the simple GWAS query we’ve done in Part 2 using “table” mode, can be executed using “raw” mode to expose the exact Cypher query that was run against the database:</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="co"># Running a GWAS query from Part 2</span>
<span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"body mass index"</span>
<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Gwas/search"</span>
<span class="va">response</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
              route <span class="op">=</span> <span class="va">endpoint</span>, 
              params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">trait</span><span class="op">)</span>,
              mode <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span> <span class="co">### asking for raw output </span>

<span class="co"># display the Cypher query</span>
<span class="va">response</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">query</span>
<span class="co">#&gt; [1] "MATCH (node: Gwas) WHERE node.trait =~ \"(?i).*body mass index.*\" RETURN node LIMIT 10;"</span></pre></div>
<p>This is what a native Cypher query looks like:</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="va">query</span> <span class="op">&lt;-</span> <span class="st">"
    MATCH (node: Gwas)                              
    WHERE node.trait =~ \"(?i).*body mass index.*\" 
    RETURN node LIMIT 10;
"</span></pre></div>
<p>This is how you supply a Cypher query to <code>query_epigraphdb</code> function:</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="co"># use Cypher general endpoint </span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/cypher"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span>

<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                route <span class="op">=</span> <span class="va">endpoint</span>, 
                params <span class="op">=</span> <span class="va">params</span>,
                method <span class="op">=</span> <span class="st">"POST"</span>,
                mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>


<span class="co"># The result should be identical to the example in Part 2</span>
<span class="va">results</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.trait</span>, <span class="va">node.id</span>, <span class="va">node.sample_size</span>,
         <span class="va">node.year</span>, <span class="va">node.author</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 10 x 5</span>
<span class="co">#&gt;    node.trait            node.id          node.sample_size node.year node.author</span>
<span class="co">#&gt;    &lt;chr&gt;                 &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt;  1 Body mass index       ebi-a-GCST006368 315347           2018      Hoffmann TJ</span>
<span class="co">#&gt;  2 Body mass index       ieu-a-2          339224           2015      Locke AE   </span>
<span class="co">#&gt;  3 Body mass index       ieu-a-785        152893           2015      Locke AE   </span>
<span class="co">#&gt;  4 Body mass index       ieu-a-835        322154           2015      Locke AE   </span>
<span class="co">#&gt;  5 Body mass index       ebi-a-GCST004904 158284           2017      Akiyama M  </span>
<span class="co">#&gt;  6 Body mass index       ieu-a-1089       120286           2016      Wood       </span>
<span class="co">#&gt;  7 Body mass index       ieu-a-974        171977           2015      Locke AE   </span>
<span class="co">#&gt;  8 Body mass index       ieu-a-95         73137            2013      Randall JC </span>
<span class="co">#&gt;  9 Body mass index       ieu-a-94         60586            2013      Randall JC </span>
<span class="co">#&gt; 10 Body mass index (BMI) ukb-b-19953      461460           2018      Ben Elswor…</span></pre></div>
<p>Next step: let’s modify Cypher query</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># Let's return Body mass index GWAS that were done by Locke AE</span>

<span class="va">query</span> <span class="op">&lt;-</span> <span class="st">"
    MATCH (node: Gwas)                              
    WHERE node.trait =~ \"(?i).*body mass index.*\"  
    AND node.author = \"Locke AE\"
    RETURN node;
"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/cypher"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span>

<span class="va">results_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                        route <span class="op">=</span> <span class="va">endpoint</span>, 
                        params <span class="op">=</span> <span class="va">params</span>,
                        method <span class="op">=</span> <span class="st">"POST"</span>,
                        mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>

<span class="va">results_subset</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.trait</span>, <span class="va">node.id</span>, <span class="va">node.sample_size</span>,
         <span class="va">node.year</span>, <span class="va">node.author</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 4 x 5</span>
<span class="co">#&gt;   node.trait      node.id   node.sample_size node.year node.author</span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt; 1 Body mass index ieu-a-2   339224           2015      Locke AE   </span>
<span class="co">#&gt; 2 Body mass index ieu-a-785 152893           2015      Locke AE   </span>
<span class="co">#&gt; 3 Body mass index ieu-a-835 322154           2015      Locke AE   </span>
<span class="co">#&gt; 4 Body mass index ieu-a-974 171977           2015      Locke AE</span></pre></div>
<p>It is important to note, that not all parameters can be queried using Cypher due to the underlying data types. For example, the query above could not be modified to filter by <code>node.sample_size</code> because in the underlying dataset (<em>currently!</em>) the sample_size column stores data as strings. Please refer to <a href="https://docs.epigraphdb.org/graph-database/meta-nodes/#gwas">data dictionary</a> to explore data types before writing native Cypher queries. In the meantime, such filtering will have to be done post-query in R.</p>
<p><br> Now, let’s try making queries by specifying a GWAS ID.</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="co"># Extract info only for GWAS 'ieu-a-2'</span>
<span class="va">query</span> <span class="op">&lt;-</span> <span class="st">"
    MATCH (node: Gwas)                              
    WHERE node.id = \"ieu-a-2\"
    RETURN node;
"</span>
<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/cypher"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span>
<span class="va">results_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                        route <span class="op">=</span> <span class="va">endpoint</span>, 
                        params <span class="op">=</span> <span class="va">params</span>,
                        method <span class="op">=</span> <span class="st">"POST"</span>,
                        mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>

<span class="va">results_subset</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">node.trait</span>, <span class="va">node.id</span>, <span class="va">node.sample_size</span>,
         <span class="va">node.year</span>, <span class="va">node.author</span><span class="op">)</span> 
<span class="co">#&gt; # A tibble: 1 x 5</span>
<span class="co">#&gt;   node.trait      node.id node.sample_size node.year node.author</span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;   &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;      </span>
<span class="co">#&gt; 1 Body mass index ieu-a-2 339224           2015      Locke AE</span></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="co"># Return MR results only for exposure trait 'ieu-a-2' (body mass index)</span>

<span class="co"># first let's check the MR Cypher query that we run in "table" mode in Part 2</span>
<span class="va">trait1</span> <span class="op">&lt;-</span> <span class="st">"Body mass index"</span> 
<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/mr"</span>
<span class="va">mr_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>route <span class="op">=</span> <span class="va">endpoint</span>, 
                          params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>exposure_trait <span class="op">=</span> <span class="va">trait1</span>,
                                        pval_threshold <span class="op">=</span> <span class="fl">5e-08</span> <span class="op">)</span>,
                          mode <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span>
<span class="va">mr_df</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">query</span>
<span class="co">#&gt; [1] "MATCH (exposure:Gwas)-[mr:MR]-&gt;(outcome:Gwas) WHERE exposure.trait = \"Body mass index\" AND mr.pval &lt; 5e-08 RETURN exposure {.id, .trait}, outcome {.id, .trait}, mr {.b, .se, .pval, .method, .selection, .moescore} ORDER BY mr.pval ;"</span>


<span class="co"># modify the query to only return 'ieu-a-2' GWAS results</span>
<span class="va">query</span><span class="op">&lt;-</span><span class="st">"
  MATCH (exposure:Gwas)-[mr:MR]-&gt;(outcome:Gwas) 
  WHERE exposure.id = \"ieu-a-2\"
  AND mr.pval &lt; 5e-08 
  RETURN exposure {.id, .trait}, outcome {.id, .trait}, mr {.b, .se, .pval, .method, .selection, .moescore} 
  ORDER BY mr.pval ;
"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/cypher"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span>
<span class="va">results_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
                        route <span class="op">=</span> <span class="va">endpoint</span>, 
                        params <span class="op">=</span> <span class="va">params</span>,
                        method <span class="op">=</span> <span class="st">"POST"</span>,
                        mode <span class="op">=</span> <span class="st">"table"</span><span class="op">)</span>
<span class="va">results_subset</span>
<span class="co">#&gt; # A tibble: 146 x 10</span>
<span class="co">#&gt;    exposure.trait exposure.id outcome.trait outcome.id   mr.se   mr.b mr.method</span>
<span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;       &lt;chr&gt;         &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;    </span>
<span class="co">#&gt;  1 Body mass ind… ieu-a-2     Non-cancer i… ukb-a-74   0.00242 0.0346 FE IVW   </span>
<span class="co">#&gt;  2 Body mass ind… ieu-a-2     Hip circumfe… ukb-a-388  0.0266  0.724  Simple m…</span>
<span class="co">#&gt;  3 Body mass ind… ieu-a-2     Waist circum… ukb-a-382  0.0245  0.656  Simple m…</span>
<span class="co">#&gt;  4 Body mass ind… ieu-a-2     Comparative … ukb-a-35   0.00791 0.137  FE IVW   </span>
<span class="co">#&gt;  5 Body mass ind… ieu-a-2     Comparative … ukb-a-34   0.0236  0.366  Simple m…</span>
<span class="co">#&gt;  6 Body mass ind… ieu-a-2     Trunk predic… ukb-a-293  0.0167  0.358  Simple m…</span>
<span class="co">#&gt;  7 Body mass ind… ieu-a-2     Trunk fat-fr… ukb-a-292  0.0177  0.360  Simple m…</span>
<span class="co">#&gt;  8 Body mass ind… ieu-a-2     Trunk fat ma… ukb-a-291  0.0280  0.737  Simple m…</span>
<span class="co">#&gt;  9 Body mass ind… ieu-a-2     Trunk fat pe… ukb-a-290  0.0275  0.527  Weighted…</span>
<span class="co">#&gt; 10 Body mass ind… ieu-a-2     Arm predicte… ukb-a-289  0.0177  0.416  Simple m…</span>
<span class="co">#&gt; # … with 136 more rows, and 3 more variables: mr.selection &lt;chr&gt;,</span>
<span class="co">#&gt; #   mr.pval &lt;dbl&gt;, mr.moescore &lt;dbl&gt;</span>

<span class="co"># check exposures in the results</span>
<span class="va">results_subset</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html">count</a></span><span class="op">(</span><span class="va">exposure.id</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 1 x 2</span>
<span class="co">#&gt;   exposure.id     n</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;int&gt;</span>
<span class="co">#&gt; 1 ieu-a-2       146</span></pre></div>
<p><br> Great! You can now use the basic functionality of the R package and make simple Cypher queries to the API. Next, we recommend to work through the <a href="https://mrcieu.github.io/epigraphdb-r/articles/case-1-pleiotropy.html">case studies</a> and check out the <a href="https://epigraphdb.org/explore">Web UI examples</a> and the <a href="https://epigraphdb.org/gallery">EpiGraphDB Gallery</a>.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
