<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Meta functionalities of the EpiGraphDB platform • epigraphdb</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Meta functionalities of the EpiGraphDB platform">
<meta property="og:description" content="epigraphdb">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epigraphdb</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">Articles</a>
    </li>
    <li>
      <a href="../articles/about.html">About EpiGraphDB</a>
    </li>
    <li>
      <a href="../articles/getting-started-with-epigraphdb-r.html">Getting started with EpiGraphDB in R</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-r-package.html">Using R package</a>
    </li>
    <li>
      <a href="../articles/meta-functionalities.html">Meta functionalities of the EpiGraphDB platform</a>
    </li>
    <li>
      <a href="../articles/case-1-pleiotropy.html">Case study 1: Distinguishing vertical and horizontal pleiotropy for SNP-protein associations</a>
    </li>
    <li>
      <a href="../articles/case-2-alt-drug-target.html">Case study 2: Identification of potential drug targets</a>
    </li>
    <li>
      <a href="../articles/case-3-literature-triangulation.html">Case study 3: Triangulating causal estimates with literature evidence</a>
    </li>
    <li>
      <a href="../articles/using-epigraphdb-api.html">Using API</a>
    </li>
    <li>
      <a href="../articles/options.html">Options</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">CHANGELOG</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MRCIEU/epigraphdb-r">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="meta-functionalities_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Meta functionalities of the EpiGraphDB platform</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MRCIEU/epigraphdb-r/blob/master/vignettes/meta-functionalities.Rmd"><code>vignettes/meta-functionalities.Rmd</code></a></small>
      <div class="hidden name"><code>meta-functionalities.Rmd</code></div>

    </div>

    
    
<p>This RMarkdown document demonstrates how key elements from the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/general-examples/platform-meta-functionalities.ipynb">github notebook for the meta functionalities</a> can be achieved using the R package. For detailed explanations of the case study please refer to the <a href="https://github.com/MRCIEU/epigraphdb/blob/master/general-examples/platform-meta-functionalities.ipynb">notebook on github</a>.</p>
<p>Here we show the following aspects of the EpiGraphDB platform, and how to use the API to get the information:</p>
<ul>
<li>Metadata: meta nodes and meta edges, and the overall schema.</li>
<li>Search for a specific node under the meta node.</li>
<li>Cypher: how to query the database directly using Neo4j Cypher</li>
</ul>
<p>For detailed documentation on the API endpoints please visit:</p>
<ul>
<li>The Swagger interface: <a href="https://api.epigraphdb.org" class="uri">https://api.epigraphdb.org</a>
</li>
<li>The sections regarding API endpoints on the documentation site: <a href="https://docs.epigraphdb.org/api/api-endpoints/" class="uri">https://docs.epigraphdb.org/api/api-endpoints/</a>
</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://magrittr.tidyverse.org">"magrittr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org">"dplyr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="http://purrr.tidyverse.org">"purrr"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:magrittr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     set_names</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://igraph.org">"igraph"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'igraph'</span>
<span class="co">#&gt; The following objects are masked from 'package:purrr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     compose, simplify</span>
<span class="co">#&gt; The following objects are masked from 'package:dplyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     as_data_frame, groups, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://mrcieu.github.io/epigraphdb-r/">"epigraphdb"</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     EpiGraphDB v1.0 (API: https://api.epigraphdb.org)</span>
<span class="co">#&gt; </span></code></pre></div>
<div id="metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#metadata" class="anchor"></a>Metadata</h2>
<p>Here we query for the metadata information using the endpoint <code>GET /meta/schema</code>, which will be used for downstream processing.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/schema"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
  graphviz <span class="op">=</span> <span class="cn">FALSE</span>,
  plot <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>
<span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"raw"</span>
<span class="op">)</span>

<span class="va">metadata</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; List of 3</span>
<span class="co">#&gt;  $ nodes      :List of 12</span>
<span class="co">#&gt;   ..$ Disease         :List of 2</span>
<span class="co">#&gt;   ..$ Pathway         :List of 2</span>
<span class="co">#&gt;   ..$ LiteratureTerm  :List of 2</span>
<span class="co">#&gt;   ..$ Gene            :List of 2</span>
<span class="co">#&gt;   ..$ LiteratureTriple:List of 2</span>
<span class="co">#&gt;   ..$ Literature      :List of 2</span>
<span class="co">#&gt;   ..$ Protein         :List of 2</span>
<span class="co">#&gt;   ..$ Variant         :List of 2</span>
<span class="co">#&gt;   ..$ Efo             :List of 2</span>
<span class="co">#&gt;   ..$ Tissue          :List of 2</span>
<span class="co">#&gt;   ..$ Drug            :List of 2</span>
<span class="co">#&gt;   ..$ Gwas            :List of 2</span>
<span class="co">#&gt;  $ edges      :List of 41</span>
<span class="co">#&gt;   ..$ OPENTARGETS_DRUG_TO_DISEASE :List of 2</span>
<span class="co">#&gt;   ..$ MEDRXIV_SUB                 :List of 2</span>
<span class="co">#&gt;   ..$ GENE_TO_DISEASE             :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_NLP_EFO                :List of 2</span>
<span class="co">#&gt;   ..$ SEMMEDDB_PREDICATE          :List of 2</span>
<span class="co">#&gt;   ..$ STRING_INTERACT_WITH        :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_TO_LITERATURE_TRIPLE   :List of 2</span>
<span class="co">#&gt;   ..$ PRS                         :List of 2</span>
<span class="co">#&gt;   ..$ BIORXIV_PREDICATE           :List of 2</span>
<span class="co">#&gt;   ..$ MONDO_MAP_UMLS              :List of 2</span>
<span class="co">#&gt;   ..$ SEMMEDDB_SUB                :List of 2</span>
<span class="co">#&gt;   ..$ MR_EVE_MR                   :List of 2</span>
<span class="co">#&gt;   ..$ OPENGWAS_TOPHITS            :List of 2</span>
<span class="co">#&gt;   ..$ VARIANT_TO_GENE             :List of 2</span>
<span class="co">#&gt;   ..$ MEDRXIV_PREDICATE           :List of 2</span>
<span class="co">#&gt;   ..$ MEDRXIV_OBJ                 :List of 2</span>
<span class="co">#&gt;   ..$ EXPRESSED_IN                :List of 2</span>
<span class="co">#&gt;   ..$ BIORXIV_OBJ                 :List of 2</span>
<span class="co">#&gt;   ..$ EFO_CHILD_OF                :List of 2</span>
<span class="co">#&gt;   ..$ BIORXIV_TO_LIT              :List of 2</span>
<span class="co">#&gt;   ..$ GEN_COR                     :List of 2</span>
<span class="co">#&gt;   ..$ METAMAP_LITE                :List of 2</span>
<span class="co">#&gt;   ..$ SEMMEDDB_TO_LIT             :List of 2</span>
<span class="co">#&gt;   ..$ PROTEIN_IN_PATHWAY          :List of 2</span>
<span class="co">#&gt;   ..$ MONDO_MAP_EFO               :List of 2</span>
<span class="co">#&gt;   ..$ PATHWAY_CHILD_OF            :List of 2</span>
<span class="co">#&gt;   ..$ TERM_TO_GENE                :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_EFO_EBI                :List of 2</span>
<span class="co">#&gt;   ..$ CPIC                        :List of 2</span>
<span class="co">#&gt;   ..$ XQTL_MULTI_SNP_MR           :List of 2</span>
<span class="co">#&gt;   ..$ XQTL_SINGLE_SNP_MR_SNP_GENE :List of 2</span>
<span class="co">#&gt;   ..$ OBS_COR                     :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_NLP                    :List of 2</span>
<span class="co">#&gt;   ..$ XQTL_SINGLE_SNP_MR_GENE_GWAS:List of 2</span>
<span class="co">#&gt;   ..$ MEDRXIV_TO_LIT              :List of 2</span>
<span class="co">#&gt;   ..$ BIORXIV_SUB                 :List of 2</span>
<span class="co">#&gt;   ..$ OPENTARGETS_DRUG_TO_TARGET  :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_TO_VARIANT             :List of 2</span>
<span class="co">#&gt;   ..$ GWAS_TO_LITERATURE          :List of 2</span>
<span class="co">#&gt;   ..$ GENE_TO_PROTEIN             :List of 2</span>
<span class="co">#&gt;   ..$ SEMMEDDB_OBJ                :List of 2</span>
<span class="co">#&gt;  $ connections:List of 41</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span>
<span class="co">#&gt;   ..$ :List of 4</span></code></pre></div>
<div id="meta-nodes" class="section level3">
<h3 class="hasAnchor">
<a href="#meta-nodes" class="anchor"></a>Meta nodes</h3>
<p>We can extract the specific meta node information as a dataframe from the metadata.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta_node_df</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/pluck.html">pluck</a></span><span class="op">(</span><span class="st">"nodes"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="op">{</span>
    <span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>
    <span class="fu"><a href="https://purrr.tidyverse.org/reference/transpose.html">transpose</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>meta_node <span class="op">=</span> <span class="va">names</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="co"># Hide properties column which does not display well</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">meta_node</span>, <span class="va">count</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="co"># We also need to flatten count</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/flatten.html">flatten_int</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span>
  <span class="op">}</span>

<span class="va">meta_node_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">meta_node</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="va">count</span>, big.mark <span class="op">=</span> <span class="st">","</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 x 2</span>
<span class="co">#&gt;    meta_node        count      </span>
<span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;      </span>
<span class="co">#&gt;  1 Disease          "   38,960"</span>
<span class="co">#&gt;  2 Drug             "    2,697"</span>
<span class="co">#&gt;  3 Efo              "   25,390"</span>
<span class="co">#&gt;  4 Gene             "   57,737"</span>
<span class="co">#&gt;  5 Gwas             "   34,494"</span>
<span class="co">#&gt;  6 Literature       "3,995,672"</span>
<span class="co">#&gt;  7 LiteratureTerm   "  108,905"</span>
<span class="co">#&gt;  8 LiteratureTriple "5,609,945"</span>
<span class="co">#&gt;  9 Pathway          "    2,441"</span>
<span class="co">#&gt; 10 Protein          "   20,280"</span>
<span class="co">#&gt; 11 Tissue           "       54"</span>
<span class="co">#&gt; 12 Variant          "   99,005"</span></code></pre></div>
</div>
<div id="meta-relationships-and-connections" class="section level3">
<h3 class="hasAnchor">
<a href="#meta-relationships-and-connections" class="anchor"></a>Meta relationships and connections</h3>
<p>We can also extract the meta relationship (edge) information, and the connections.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta_rel_df</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://purrr.tidyverse.org/reference/pluck.html">pluck</a></span><span class="op">(</span><span class="st">"edges"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="op">{</span>
    <span class="va">names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>
    <span class="fu"><a href="https://purrr.tidyverse.org/reference/transpose.html">transpose</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>meta_rel <span class="op">=</span> <span class="va">names</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/flatten.html">flatten_int</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">meta_rel</span>, <span class="va">count</span><span class="op">)</span>
  <span class="op">}</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span>
    <span class="va">metadata</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/pluck.html">pluck</a></span><span class="op">(</span><span class="st">"connections"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="op">{</span>
      <span class="fu"><a href="https://purrr.tidyverse.org/reference/transpose.html">transpose</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">%&gt;%</span>
        <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>meta_rel <span class="op">=</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/flatten.html">flatten_chr</a></span><span class="op">(</span><span class="va">rel</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/vars.html">vars</a></span><span class="op">(</span><span class="va">from_node</span>, <span class="va">to_node</span><span class="op">)</span>, <span class="va">flatten_chr</span><span class="op">)</span> <span class="op">%&gt;%</span>
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">meta_rel</span>, <span class="va">from_node</span>, <span class="va">to_node</span><span class="op">)</span>
    <span class="op">}</span>
  <span class="op">)</span>
<span class="co">#&gt; Joining, by = "meta_rel"</span>

<span class="va">meta_rel_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">from_node</span>, <span class="va">to_node</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="va">count</span>, big.mark <span class="op">=</span> <span class="st">","</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 41 x 4</span>
<span class="co">#&gt;    meta_rel                     count        from_node to_node       </span>
<span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;        &lt;chr&gt;     &lt;chr&gt;         </span>
<span class="co">#&gt;  1 MONDO_MAP_EFO                "     2,819" Disease   Efo           </span>
<span class="co">#&gt;  2 MONDO_MAP_UMLS               "     8,247" Disease   LiteratureTerm</span>
<span class="co">#&gt;  3 OPENTARGETS_DRUG_TO_DISEASE  "     2,461" Drug      Disease       </span>
<span class="co">#&gt;  4 CPIC                         "       375" Drug      Gene          </span>
<span class="co">#&gt;  5 OPENTARGETS_DRUG_TO_TARGET   "     6,534" Drug      Gene          </span>
<span class="co">#&gt;  6 EFO_CHILD_OF                 "    43,132" Efo       Efo           </span>
<span class="co">#&gt;  7 GENE_TO_DISEASE              "     5,763" Gene      Disease       </span>
<span class="co">#&gt;  8 XQTL_MULTI_SNP_MR            " 3,015,233" Gene      Gwas          </span>
<span class="co">#&gt;  9 XQTL_SINGLE_SNP_MR_GENE_GWAS " 8,449,779" Gene      Gwas          </span>
<span class="co">#&gt; 10 GENE_TO_PROTEIN              "    19,142" Gene      Protein       </span>
<span class="co">#&gt; # … with 31 more rows</span></code></pre></div>
</div>
<div id="schema-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#schema-plot" class="anchor"></a>Schema plot</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph</span> <span class="op">&lt;-</span> <span class="va">meta_rel_df</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">from_node</span>, <span class="va">to_node</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span><span class="op">(</span>directed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">graph</span><span class="op">$</span><span class="va">layout</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/igraph/man/layout_with_kk.html">layout_with_kk</a></span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
<p><img src="meta-functionalities_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>A detailed version of the <a href="https://api.epigraphdb.org/meta/schema?graphviz=true&amp;plot=true">schema_plot</a> can be obtained from the API.</p>
</div>
</div>
<div id="search-for-specific-node" class="section level2">
<h2 class="hasAnchor">
<a href="#search-for-specific-node" class="anchor"></a>Search for specific node</h2>
<p>Users can use <a href="https://dev.epigraphdb.org/explore">the explorer on the Web UI</a> to search for a specific node by:</p>
<ul>
<li>fuzzy matching by “name” field.</li>
<li>exact matching by “ID” field if you know the its ID (e.g. the ID to a GWAS from IEU GWAS Database).</li>
</ul>
<p>Here we show how these are done at the API level using <code>Gwas</code> nodes as an example.</p>
<p>First we need to know what the “ID” and “name” fields are for the meta nodes using <code>GET /meta/nodes/id-name-schema</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/id-name-schema"</span>
<span class="va">meta_node_fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="cn">NULL</span>, mode <span class="op">=</span> <span class="st">"raw"</span>
<span class="op">)</span>
<span class="va">meta_node_fields</span>
<span class="co">#&gt; $Disease</span>
<span class="co">#&gt; $Disease$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Disease$name</span>
<span class="co">#&gt; [1] "label"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Drug</span>
<span class="co">#&gt; $Drug$id</span>
<span class="co">#&gt; [1] "label"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Drug$name</span>
<span class="co">#&gt; [1] "label"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Efo</span>
<span class="co">#&gt; $Efo$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Efo$name</span>
<span class="co">#&gt; [1] "value"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Gene</span>
<span class="co">#&gt; $Gene$id</span>
<span class="co">#&gt; [1] "ensembl_id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Gene$name</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Gwas</span>
<span class="co">#&gt; $Gwas$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Gwas$name</span>
<span class="co">#&gt; [1] "trait"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Literature</span>
<span class="co">#&gt; $Literature$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Literature$name</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LiteratureTerm</span>
<span class="co">#&gt; $LiteratureTerm$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LiteratureTerm$name</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LiteratureTriple</span>
<span class="co">#&gt; $LiteratureTriple$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $LiteratureTriple$name</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Pathway</span>
<span class="co">#&gt; $Pathway$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Pathway$name</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Protein</span>
<span class="co">#&gt; $Protein$id</span>
<span class="co">#&gt; [1] "uniprot_id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Protein$name</span>
<span class="co">#&gt; [1] "uniprot_id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Tissue</span>
<span class="co">#&gt; $Tissue$id</span>
<span class="co">#&gt; [1] "id"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Tissue$name</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Variant</span>
<span class="co">#&gt; $Variant$id</span>
<span class="co">#&gt; [1] "name"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Variant$name</span>
<span class="co">#&gt; [1] "name"</span></code></pre></div>
</div>
<div id="fuzzy-matching" class="section level2">
<h2 class="hasAnchor">
<a href="#fuzzy-matching" class="anchor"></a>Fuzzy matching</h2>
<p>Here we search for nodes can contain “body mass index” in their traits.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"body mass index"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Gwas/search"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>

<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span>
<span class="op">)</span>
<span class="va">results</span>
<span class="co">#&gt; # A tibble: 10 x 18</span>
<span class="co">#&gt;    node.note        node._name  node.year node.mr node.author node.sex node.pmid</span>
<span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;    &lt;chr&gt;    </span>
<span class="co">#&gt;  1 Dominance model… Body mass … 2016.0    0       Wood        Males a… 26961502…</span>
<span class="co">#&gt;  2 &lt;NA&gt;             Body mass … 2015.0    1       Locke AE    Females  25673413…</span>
<span class="co">#&gt;  3 &lt;NA&gt;             Body mass … 2013.0    1       Randall JC  Females  23754948…</span>
<span class="co">#&gt;  4 &lt;NA&gt;             Body mass … 2017.0    1       Akiyama M   &lt;NA&gt;     28892062…</span>
<span class="co">#&gt;  5 &lt;NA&gt;             Body mass … 2018.0    1       Hoffmann TJ &lt;NA&gt;     30108127…</span>
<span class="co">#&gt;  6 &lt;NA&gt;             Body mass … 2019.0    1       Ishigaki K  Males    28892062…</span>
<span class="co">#&gt;  7 &lt;NA&gt;             Body mass … 2015.0    1       Locke AE    Males a… 25673413…</span>
<span class="co">#&gt;  8 &lt;NA&gt;             Body mass … 2015.0    1       Locke AE    Males a… 25673413…</span>
<span class="co">#&gt;  9 &lt;NA&gt;             Body mass … 2015.0    1       Locke AE    Males    25673413…</span>
<span class="co">#&gt; 10 &lt;NA&gt;             Body mass … 2019.0    1       Ishigaki K  Males a… 28892062…</span>
<span class="co">#&gt; # … with 11 more variables: node.population &lt;chr&gt;, node.sample_size &lt;chr&gt;,</span>
<span class="co">#&gt; #   node.nsnp &lt;chr&gt;, node.build &lt;chr&gt;, node.trait &lt;chr&gt;, node._source &lt;list&gt;,</span>
<span class="co">#&gt; #   node.id &lt;chr&gt;, node._id &lt;chr&gt;, node.subcategory &lt;chr&gt;, node.category &lt;chr&gt;,</span>
<span class="co">#&gt; #   node.sd &lt;chr&gt;</span></code></pre></div>
</div>
<div id="exact-matching" class="section level2">
<h2 class="hasAnchor">
<a href="#exact-matching" class="anchor"></a>Exact matching</h2>
<p>Similarly, we can exact match a specific node by its ID.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ieu-a-2"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/meta/nodes/Gwas/search"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>id <span class="op">=</span> <span class="va">id</span><span class="op">)</span>

<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, mode <span class="op">=</span> <span class="st">"table"</span>
<span class="op">)</span>
<span class="va">results</span>
<span class="co">#&gt; # A tibble: 1 x 17</span>
<span class="co">#&gt;   node._name  node.year node.mr node.author node.sex   node.pmid node.population</span>
<span class="co">#&gt;   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;          </span>
<span class="co">#&gt; 1 Body mass … 2015.0    1       Locke AE    Males and… 25673413… Mixed          </span>
<span class="co">#&gt; # … with 10 more variables: node.sd &lt;chr&gt;, node.sample_size &lt;chr&gt;,</span>
<span class="co">#&gt; #   node.nsnp &lt;chr&gt;, node.build &lt;chr&gt;, node.trait &lt;chr&gt;, node._source &lt;list&gt;,</span>
<span class="co">#&gt; #   node.id &lt;chr&gt;, node._id &lt;chr&gt;, node.subcategory &lt;chr&gt;, node.category &lt;chr&gt;</span></code></pre></div>
</div>
<div id="cypher-advanced" class="section level2">
<h2 class="hasAnchor">
<a href="#cypher-advanced" class="anchor"></a>Cypher (advanced)</h2>
<p>Advanced users that are familiar with Neo4j Cypher can query the database using Cypher directly.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">query</span> <span class="op">&lt;-</span> <span class="st">"
    MATCH (exposure:Gwas)-[mr:MR]-&gt;(outcome:Gwas)
    WHERE exposure.trait = 'Body mass index'
    RETURN exposure, outcome, mr LIMIT 2
"</span>

<span class="va">endpoint</span> <span class="op">&lt;-</span> <span class="st">"/cypher"</span>
<span class="va">params</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query</span><span class="op">)</span>

<span class="co"># NOTE this is a POST request</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_epigraphdb.html">query_epigraphdb</a></span><span class="op">(</span>
  route <span class="op">=</span> <span class="va">endpoint</span>, params <span class="op">=</span> <span class="va">params</span>, method <span class="op">=</span> <span class="st">"POST"</span>,
  mode <span class="op">=</span> <span class="st">"table"</span>
<span class="op">)</span>
<span class="va">results</span>
<span class="co">#&gt; # A tibble: 0 x 0</span></code></pre></div>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a><code>sessionInfo</code>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] epigraphdb_0.2.2 igraph_1.2.6     purrr_0.3.4      dplyr_1.0.6     </span>
<span class="co">#&gt; [5] magrittr_2.0.1  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] pillar_1.6.1      compiler_4.1.0    highr_0.9         tools_4.1.0      </span>
<span class="co">#&gt;  [5] digest_0.6.27     jsonlite_1.7.2    evaluate_0.14     memoise_2.0.0    </span>
<span class="co">#&gt;  [9] lifecycle_1.0.0   tibble_3.1.2      pkgconfig_2.0.3   rlang_0.4.11     </span>
<span class="co">#&gt; [13] cli_2.5.0         rstudioapi_0.13   curl_4.3.1        yaml_2.2.1       </span>
<span class="co">#&gt; [17] pkgdown_1.6.1     xfun_0.23         fastmap_1.1.0     stringr_1.4.0    </span>
<span class="co">#&gt; [21] httr_1.4.2        knitr_1.33        generics_0.1.0    desc_1.3.0       </span>
<span class="co">#&gt; [25] fs_1.5.0          vctrs_0.3.8       systemfonts_1.0.2 rprojroot_2.0.2  </span>
<span class="co">#&gt; [29] tidyselect_1.1.1  glue_1.4.2        R6_2.5.0          textshaping_0.3.4</span>
<span class="co">#&gt; [33] fansi_0.5.0       rmarkdown_2.8     ps_1.6.0          htmltools_0.5.1.1</span>
<span class="co">#&gt; [37] ellipsis_0.3.2    ragg_1.1.2        utf8_1.2.1        stringi_1.6.2    </span>
<span class="co">#&gt; [41] cachem_1.0.5      crayon_1.4.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://yiliu6240.github.io">Yi Liu</a>, Valeriia Haberland, <a href="https://marinalearning.netlify.com">Marina Vabistsevits</a>, <a href="http://www.biocompute.org.uk/">Tom Gaunt</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
